Search.setIndex({"docnames": ["authors", "auto_examples/00_epigenetic_biomarkers/index", "auto_examples/00_epigenetic_biomarkers/plot_custom_imputation_implementation", "auto_examples/00_epigenetic_biomarkers/plot_down_syndrome_model", "auto_examples/00_epigenetic_biomarkers/plot_epigenetic_clocks_on_geo", "auto_examples/00_epigenetic_biomarkers/sg_execution_times", "auto_examples/01_composite_biomarkers/index", "auto_examples/01_composite_biomarkers/plot_nhanes", "auto_examples/01_composite_biomarkers/sg_execution_times", "auto_examples/02_challenge_submissions/index", "auto_examples/02_challenge_submissions/sg_execution_times", "auto_examples/02_challenge_submissions/simple_competiton_submission", "auto_examples/02_challenge_submissions/training_simple_model", "auto_examples/03_deconvolution/index", "auto_examples/03_deconvolution/plot_blood_deconvolution", "auto_examples/03_deconvolution/sg_execution_times", "auto_examples/index", "auto_examples/sg_execution_times", "changes/names", "clocks", "data", "index", "modules/cache", "modules/data_library", "modules/generated/biolearn.cache.LocalFolderCache", "modules/generated/biolearn.cache.NoCache", "modules/generated/biolearn.data_library.DataLibrary", "modules/generated/biolearn.data_library.DataSource", "modules/generated/biolearn.data_library.GeoData", "modules/generated/biolearn.imputation.hybrid_impute", "modules/generated/biolearn.imputation.impute_from_average", "modules/generated/biolearn.imputation.impute_from_standard", "modules/generated/biolearn.load.load_fhs", "modules/generated/biolearn.load.load_nhanes", "modules/generated/biolearn.model_gallery.ModelGallery", "modules/imputation", "modules/index", "modules/load", "modules/model_gallery", "quickstart", "roadmap", "sg_execution_times"], "filenames": ["authors.rst", "auto_examples/00_epigenetic_biomarkers/index.rst", "auto_examples/00_epigenetic_biomarkers/plot_custom_imputation_implementation.rst", "auto_examples/00_epigenetic_biomarkers/plot_down_syndrome_model.rst", "auto_examples/00_epigenetic_biomarkers/plot_epigenetic_clocks_on_geo.rst", "auto_examples/00_epigenetic_biomarkers/sg_execution_times.rst", "auto_examples/01_composite_biomarkers/index.rst", "auto_examples/01_composite_biomarkers/plot_nhanes.rst", "auto_examples/01_composite_biomarkers/sg_execution_times.rst", "auto_examples/02_challenge_submissions/index.rst", "auto_examples/02_challenge_submissions/sg_execution_times.rst", "auto_examples/02_challenge_submissions/simple_competiton_submission.rst", "auto_examples/02_challenge_submissions/training_simple_model.rst", "auto_examples/03_deconvolution/index.rst", "auto_examples/03_deconvolution/plot_blood_deconvolution.rst", "auto_examples/03_deconvolution/sg_execution_times.rst", "auto_examples/index.rst", "auto_examples/sg_execution_times.rst", "changes/names.rst", "clocks.rst", "data.rst", "index.rst", "modules/cache.rst", "modules/data_library.rst", "modules/generated/biolearn.cache.LocalFolderCache.rst", "modules/generated/biolearn.cache.NoCache.rst", "modules/generated/biolearn.data_library.DataLibrary.rst", "modules/generated/biolearn.data_library.DataSource.rst", "modules/generated/biolearn.data_library.GeoData.rst", "modules/generated/biolearn.imputation.hybrid_impute.rst", "modules/generated/biolearn.imputation.impute_from_average.rst", "modules/generated/biolearn.imputation.impute_from_standard.rst", "modules/generated/biolearn.load.load_fhs.rst", "modules/generated/biolearn.load.load_nhanes.rst", "modules/generated/biolearn.model_gallery.ModelGallery.rst", "modules/imputation.rst", "modules/index.rst", "modules/load.rst", "modules/model_gallery.rst", "quickstart.md", "roadmap.rst", "sg_execution_times.rst"], "titles": ["Team", "Epigenetic Clock Examples", "Performing custom imputations", "Down Syndrome Epigenetic Plotting", "\u201cEpigenetic Clocks\u201d in GEO Data", "Computation times", "Composite Biomarkers Examples", "\u201cPhenotypic Ages\u201d in NHANES Data", "Computation times", "Challenge Submission Examples", "Computation times", "Building a competition submission using an existing model", "Training an ElasticNet model", "Deconvolution Examples", "\u201cDeconvolution Example\u201d", "Computation times", "Examples", "Computation times", "&lt;no title&gt;", "Clocks and Other Models", "GEO Data Sources", "Biolearn", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.cache</span></code>: Cache", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.data_library</span></code>: DataLibrary", "biolearn.cache.LocalFolderCache", "biolearn.cache.NoCache", "biolearn.data_library.DataLibrary", "biolearn.data_library.DataSource", "biolearn.data_library.GeoData", "biolearn.imputation.hybrid_impute", "biolearn.imputation.impute_from_average", "biolearn.imputation.impute_from_standard", "biolearn.load.load_fhs", "biolearn.load.load_nhanes", "biolearn.model_gallery.ModelGallery", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.imputation</span></code>: Data Imputation Utilities", "API References", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.load</span></code>: Data Loading Utilities", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.model_gallery</span></code>: ModelGallery", "Quickstart", "Biolearn and the Future of Biomarkers", "Computation times"], "terms": {"thi": [0, 2, 3, 4, 7, 11, 12, 14, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 39, 40], "work": [0, 40], "i": [0, 21, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 36, 39, 40], "made": [0, 40], "avail": [0, 21, 34, 39, 40], "commun": [0, 11], "peopl": 0, "which": [0, 12, 24, 25, 28, 33], "origin": [0, 2], "from": [0, 5, 6, 7, 8, 10, 11, 12, 14, 15, 16, 17, 19, 20, 21, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 35, 39, 40, 41], "biomark": [0, 9, 21, 26, 39], "ag": [0, 6, 8, 9, 12, 16, 19, 20, 21, 33, 39, 40, 41], "consortium": [0, 21], "The": [0, 24, 25, 26, 34, 39, 40], "biolearn": [0, 2, 3, 4, 7, 11, 12, 14, 19, 20, 36], "ar": [0, 2, 12, 21, 26, 27, 28, 29, 33, 34, 39, 40], "mahdi": 0, "moqri": [0, 21], "research": [0, 21, 26, 40], "boa": 0, "execut": [0, 5, 8, 10, 15, 16, 17, 39, 41], "martin": 0, "perez": [0, 21], "guevara": [0, 21], "machin": 0, "learn": [0, 21, 39, 40], "engin": 0, "seth": 0, "paulson": [0, 21], "softwar": 0, "architect": 0, "albert": 0, "ying": [0, 21], "harvard": 0, "universii": 0, "dayoon": 0, "kwon": [0, 21], "univers": 0, "california": 0, "special": 0, "thank": 0, "scientif": 0, "member": 0, "plot": [1, 2, 5, 6, 13, 16, 21, 28, 41], "result": [1, 3, 11, 13, 16], "run": [1, 7, 11, 13, 14, 16, 19, 21, 34, 39], "function": [1, 16, 27, 29, 30, 31, 32, 33, 35, 36, 37], "geo": [1, 5, 16, 26, 27, 28, 40, 41], "data": [1, 5, 6, 8, 14, 16, 21, 24, 26, 27, 28, 29, 32, 33, 34, 39, 40, 41], "down": [1, 5, 12, 16, 19, 28, 41], "syndrom": [1, 5, 16, 19, 28, 41], "perform": [1, 5, 7, 16, 28, 31, 41], "custom": [1, 5, 16, 28, 31, 41], "imput": [1, 4, 5, 16, 28, 34, 41], "go": [2, 3, 4, 7, 11, 12, 14], "end": [2, 3, 4, 7, 11, 12, 14], "download": [2, 3, 4, 7, 11, 12, 14, 16], "full": [2, 3, 4, 7, 11, 12, 14], "exampl": [2, 3, 4, 5, 7, 8, 10, 11, 12, 15, 17, 39, 41], "code": [2, 3, 4, 7, 11, 12, 14, 16, 21, 39, 40], "browser": [2, 3, 4, 7, 11, 12, 14], "via": [2, 3, 4, 7, 11, 12, 14, 40], "binder": [2, 3, 4, 7, 11, 12, 14], "demonstr": [2, 21], "eas": 2, "librari": [2, 20, 21, 26, 34, 39, 40], "data_librari": [2, 3, 4, 11, 12, 14, 39], "import": [2, 3, 4, 7, 11, 12, 14], "datalibrari": [2, 3, 4, 12, 14, 27, 39], "geodata": [2, 11, 12, 27], "comput": [2, 21], "averag": [2, 29, 30, 31, 34, 35], "reference_dataset": 2, "gse40279": [2, 12, 20], "reference_averag": 2, "dnam": [2, 4, 12, 28, 29, 30, 31], "mean": [2, 12], "axi": 2, "1": [2, 5, 7, 8, 12, 14, 15, 40], "second": [2, 3, 4, 7, 14], "target_dataset": 2, "gse51057": [2, 20], "impute_from_standard": 2, "imputed_data": 2, "imputed_dataset": 2, "metadata": [2, 3, 12, 14, 27, 28], "model": [2, 9, 10, 16, 26, 34, 40, 41], "galleri": [2, 3, 4, 7, 11, 12, 14, 16, 19, 41], "compar": [2, 21], "epigenet": [2, 5, 20, 21, 27, 28, 40, 41], "model_galleri": [2, 3, 4, 11, 14], "modelgalleri": [2, 3, 4, 11, 14], "clock_model": 2, "horvathv1": [2, 4, 19], "original_age_predict": 2, "imputed_age_predict": 2, "seaborn": [2, 3, 7, 14], "sn": [2, 3, 4, 7, 14], "panda": [2, 4, 12, 28, 32, 33], "pd": [2, 4, 12, 29, 30, 31], "matplotlib": [2, 3, 7, 12, 14], "pyplot": [2, 3, 7, 12, 14], "plt": [2, 3, 7, 12, 14], "prepar": [2, 40], "comparison_data": 2, "datafram": [2, 4, 12, 28, 29, 30, 31, 32, 33], "creat": [2, 3, 28, 39], "scatter": [2, 4, 12], "figur": [2, 3, 12, 14], "figsiz": [2, 3, 12, 14], "8": [2, 5, 8, 12, 29], "6": [2, 3, 12, 20], "scatterplot": [2, 7, 14], "x": [2, 3, 7, 12, 14], "y": [2, 3, 7, 12, 14], "titl": [2, 3, 12, 14, 20, 27], "v": [2, 12, 21, 39], "xlabel": [2, 7, 12, 14], "ylabel": [2, 7, 12, 14], "grid": [2, 12], "true": [2, 3, 12], "show": [2, 11, 12, 21], "imputationdecor": 2, "defin": [2, 12, 26], "method": [2, 27, 34, 35], "def": 2, "custom_impute_method": 2, "dnam_data": 2, "needed_cpg": 2, "return": [2, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34], "wrap": 2, "decorated_clock": 2, "decorated_clock_predict": 2, "verifi": [2, 39], "direct": 2, "approach": [2, 40], "same_result": 2, "all": [2, 16, 24, 25, 26, 28, 30, 31, 41], "print": [2, 12], "total": [2, 3, 4, 5, 7, 8, 10, 14, 15, 17, 19, 21, 41], "time": [2, 3, 4, 7, 12, 14, 40], "script": [2, 3, 4, 7, 14], "4": 2, "minut": [2, 3, 4, 7, 14, 40], "50": [2, 5], "771": [2, 5], "estim": [2, 3, 4, 7, 11, 12], "memori": [2, 3, 4, 7, 11, 12, 14], "usag": [2, 3, 4, 7, 11, 12, 14], "11709": [2, 5], "mb": [2, 3, 4, 5, 7, 8, 10, 11, 12, 14, 15, 17, 41], "jupyt": [2, 3, 4, 7, 11, 12, 14, 16, 39], "notebook": [2, 3, 4, 7, 11, 12, 14, 16, 39], "plot_custom_imputation_implement": [2, 5, 41], "ipynb": [2, 3, 4, 7, 11, 12, 14], "python": [2, 3, 4, 7, 11, 12, 14, 16, 39], "sourc": [2, 3, 4, 7, 11, 12, 14, 16, 19, 21, 26, 27, 29, 39, 40], "py": [2, 3, 4, 5, 7, 8, 10, 11, 12, 14, 15, 41], "gener": [2, 4, 7, 9, 11, 12, 14, 16, 28, 40], "sphinx": [2, 3, 4, 7, 11, 12, 14, 16], "your": [3, 4, 7, 11, 12, 14, 21, 40], "dna": [3, 4, 14, 19, 20, 26, 29, 30, 31, 35], "how": [3, 11, 21, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 39, 40], "samepl": 3, "can": [3, 11, 19, 21, 26, 28, 34, 39, 40], "distiguish": 3, "test_data": 3, "gse52588": [3, 20], "downsyndrom": [3, 19], "set": [3, 4, 20, 26, 27, 34, 35], "index": [3, 4, 12], "merg": 3, "set_index": 3, "astyp": [3, 14], "str": [3, 24, 25, 26, 28, 29, 30, 31, 34], "inplac": 3, "merged_data": 3, "join": [3, 21], "category_ord": 3, "healthi": [3, 20], "palett": 3, "green": 3, "red": 3, "configur": [3, 27, 40], "10": [3, 12, 14, 15, 21, 39], "ax": [3, 7], "disease_st": 3, "width": 3, "0": [3, 4, 7, 10, 11, 12, 14, 17, 29, 41], "3": [3, 5, 12, 15, 39], "order": 3, "stripplot": 3, "jitter": 3, "hue": 3, "dodg": 3, "fals": 3, "legend": [3, 14], "set_titl": 3, "despin": 3, "25": [3, 5, 40], "117": [3, 5], "1147": 3, "plot_down_syndrome_model": [3, 5, 41], "calcul": [4, 12, 34], "multipl": 4, "them": [4, 14, 16, 26, 40], "against": 4, "chronolog": [4, 21], "gse41169": [4, 20], "blood": [4, 6, 7, 16, 19, 20, 40], "data_sourc": 4, "get": [4, 11, 12, 14, 21, 24, 25, 26, 34, 39], "note": [4, 12], "default": [4, 26, 27, 29, 34], "miss": [4, 25, 27, 28, 29, 30, 31, 34, 35], "To": [4, 39], "chang": [4, 20, 39], "behavior": [4, 39], "paramet": [4, 12, 24, 25, 26, 27, 28, 29, 30, 31, 33, 34], "when": 4, "horvath_result": 4, "predict": [4, 12, 14, 19, 21, 34], "hannum_result": 4, "hannum": [4, 19], "phenoage_result": 4, "phenoag": [4, 19], "actual_ag": 4, "plot_data": 4, "horvath": [4, 21, 39], "relplot": 4, "kind": 4, "axisgrid": 4, "facetgrid": 4, "object": [4, 12, 24, 26, 27, 28, 34], "0x7cc4ec3e0a60": 4, "30": [4, 5], "343": [4, 5], "1705": 4, "plot_epigenetic_clocks_on_geo": [4, 5, 41], "05": [5, 12], "46": 5, "231": 5, "file": [5, 8, 10, 11, 15, 17, 19, 26, 27, 28, 40, 41], "auto_exampl": [5, 8, 10, 15, 17], "00_epigenetic_biomark": [5, 41], "mem": [5, 8, 10, 15, 17, 41], "04": 5, "clock": [5, 6, 21, 27, 28, 39, 41], "00": [5, 8, 10, 15, 17, 41], "1704": 5, "7": [5, 12, 41], "1146": 5, "surviv": [6, 16, 21], "curv": [6, 16], "base": [6, 12, 16, 26, 29, 34], "us": [6, 9, 10, 16, 20, 21, 24, 25, 29, 30, 32, 39, 40, 41], "standard": [6, 16, 34], "test": [6, 16, 40], "phenotyp": [6, 8, 16, 33, 41], "nhane": [6, 8, 16, 21, 33, 41], "exam": 7, "analys": [7, 21, 39], "load_nhan": 7, "year": [7, 19, 33], "df": [7, 12, 32, 33], "years_until_death": 7, "months_until_death": 7, "12": [7, 21], "hematologi": 7, "phenotypic_ag": 7, "": [7, 19, 21, 27, 34, 40], "2": [7, 10, 12], "lifelin": 7, "kaplanmeierfitt": 7, "biologically_old": 7, "subplot": 7, "group": [7, 33], "ix": 7, "t": [7, 12, 14], "e": 7, "is_dead": 7, "kmf": 7, "fit": [7, 12], "label": 7, "plot_survival_funct": 7, "text": [7, 14], "5": [7, 12, 14], "23": 7, "52222222222222": 7, "115": [7, 8], "9": 7, "plot_nhan": [7, 8, 41], "01": [8, 12], "01_composite_biomark": [8, 41], "build": [9, 10, 16, 34, 40, 41], "competit": [9, 10, 16, 34, 41], "an": [9, 10, 16, 20, 21, 24, 25, 26, 27, 28, 29, 31, 32, 33, 34, 40, 41], "exist": [9, 10, 16, 24, 29, 34, 41], "train": [9, 10, 16, 26, 41], "elasticnet": [9, 10, 16, 26, 41], "000": [10, 17, 41], "02_challenge_submiss": [10, 41], "submiss": [10, 34, 41], "simple_competiton_submiss": [10, 11, 41], "training_simple_model": [10, 12, 41], "you": [11, 12, 16, 21, 39, 40], "warmup": [11, 12], "challeng": [11, 21, 40], "link": [11, 12], "here": [11, 12], "http": [11, 12, 21, 39], "www": [11, 12], "synaps": [11, 12, 21], "org": [11, 12, 21], "syn52966292": [11, 12], "wiki": [11, 12], "625231": [11, 12], "downloaded_data_file_path": [11, 12], "add": [11, 12], "path": [11, 12, 24, 26, 27, 28, 40], "challenge_data": [11, 12], "from_methylation_matrix": [11, 12, 28], "score": 11, "best": 11, "so": 11, "let": 11, "save": 11, "csv": [11, 28], "correct": 11, "format": [11, 20, 26, 27, 40], "now": [11, 40], "we": [11, 21, 39, 40], "expect": [11, 39], "rename_axi": 11, "sampleid": [11, 12], "renam": 11, "column": [11, 12, 28, 29, 30, 31, 32, 33], "predictedag": [11, 12], "to_csv": 11, "lin_submiss": 11, "submit": [11, 12], "follow": [11, 16], "instruct": [11, 16, 39], "below": [11, 29], "simpl": [12, 35], "narrow": 12, "what": 12, "site": [12, 28, 29, 30, 31], "correl": 12, "take": [12, 40], "long": 12, "numpi": 12, "np": 12, "sklearn": 12, "linear_model": 12, "linearregress": 12, "extract": [12, 40], "transpos": 12, "valu": [12, 24, 25, 29, 30, 31, 34, 35], "have": [12, 16, 39, 40], "sampl": [12, 20, 28, 29, 30, 31, 40], "row": [12, 28, 29, 30, 31, 32, 33], "cpg": [12, 14, 29, 30, 31], "bootstrap": 12, "featur": [12, 39], "select": [12, 34], "n_bootstrap": 12, "20": 12, "threshold": [12, 29], "store": [12, 20, 24, 25, 26], "count": 12, "each": [12, 28, 32, 33], "deem": 12, "signific": 12, "cpg_count": 12, "zero": 12, "shape": 12, "begin": 12, "iter": 12, "_": [12, 19], "rang": 12, "replac": [12, 29], "sample_idx": 12, "random": 12, "choic": 12, "size": [12, 24], "x_sampl": 12, "y_sampl": 12, "identifi": [12, 26, 28, 40], "magnitud": [12, 40], "coeffici": [12, 19], "significant_cpg": 12, "where": [12, 16, 21, 28, 32, 33, 40], "ab": 12, "coef_": 12, "determin": [12, 29], "stabl": 12, "stable_cpg_sit": 12, "stable_cpg_nam": 12, "tolist": 12, "f": 12, "associ": [12, 20, 28], "more": [12, 21, 35, 39, 40], "than": 12, "60": 12, "model_select": 12, "train_test_split": 12, "top_sites_df": 12, "x_train": 12, "x_test": 12, "y_train": 12, "y_test": 12, "test_siz": 12, "random_st": 12, "42": 12, "alpha": 12, "regular": 12, "strength": 12, "l1_ratio": 12, "mix": 12, "between": [12, 21], "l1": 12, "l2": 12, "lasso": 12, "ridg": 12, "metric": 12, "mean_squared_error": 12, "max_it": 12, "10000": 12, "evalu": 12, "y_pred": 12, "mse": 12, "squar": 12, "error": [12, 24, 39], "edgecolor": 12, "w": 12, "linewidth": 12, "min": 12, "max": 12, "k": [12, 21], "lw": 12, "line": [12, 21, 39], "refer": [12, 14, 20, 21, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 39], "actual": [12, 39], "both": [12, 28, 29, 40], "linestyl": 12, "4f": 12, "absolut": 12, "mae": 12, "pruned_data": 12, "fillna": 12, "challenge_result": 12, "predicted_age_df": 12, "name": [12, 19, 34], "dataset": [13, 16, 20, 29, 30, 31, 34, 35, 40], "known": [13, 16], "cell": [13, 16, 19], "proport": [13, 16, 19], "type": [14, 24, 25, 27, 28, 29, 30, 31, 32, 33, 34], "epic_facs_salas_18": 14, "gse112618": [14, 20], "epic_facs_cell_prop": 14, "deconvolutebloodep": [14, 19], "number": [14, 19, 20, 33], "600": 14, "intersect": 14, "complet": 14, "set_styl": 14, "tick": 14, "neutrophils_proport": 14, "float": [14, 24, 29], "loc": 14, "neutrophil": 14, "monocytes_proport": 14, "monocyt": 14, "nk_proport": 14, "nk_cell": 14, "bcell_proport": 14, "b_cell": 14, "cd4t_proport": 14, "cd4_t_cell": 14, "cd8t_proport": 14, "cd8_t_cell": 14, "nk": 14, "b": 14, "cd4": 14, "cd8": 14, "valid": [14, 20], "976": [14, 15], "plot_blood_deconvolut": [14, 15, 41], "03_deconvolut": [15, 41], "deconvolut": [15, 28, 41], "If": [16, 21, 26, 27, 28, 29, 34, 39, 40], "want": [16, 40], "make": 16, "sure": 16, "directori": [16, 24], "write": [16, 40], "permiss": 16, "copi": [16, 28], "instal": 16, "biolean": 16, "manual": 16, "auto_examples_python": 16, "zip": 16, "auto_examples_jupyt": 16, "n": [17, 21], "A": [17, 24, 25, 26, 27, 28, 32, 33, 34], "support": [19, 20, 21, 26, 33, 34, 40], "easili": [19, 21, 26, 39], "speci": [19, 34], "tissu": [19, 34], "2013": 19, "human": [19, 20, 40], "multi": [19, 21, 40], "paper": [19, 21, 39], "lin": 19, "2016": 19, "2018": 19, "yingcausag": 19, "2022": 19, "yingdamag": 19, "yingadaptag": 19, "horvathv2": 19, "skin": 19, "pedb": 19, "2019": 19, "buccal": 19, "zhang_10": 19, "mortal": [19, 21], "risk": 19, "dunedinpoam38": 19, "2020": 19, "rate": 19, "dunedinpac": [19, 21, 34, 39], "grimagev1": 19, "adjust": 19, "grimagev2": 19, "alcoholmccartnei": 19, "alcohol": 19, "consumpt": 19, "bmi_mccartnei": 19, "bmi": 19, "dnamtl": 19, "adipos": 19, "telomer": 19, "length": 19, "hrsinchphenoag": 19, "knight": 19, "cord": 19, "gestat": 19, "leecontrol": 19, "placenta": 19, "sexestim": 19, "2021": 19, "sex": [19, 20], "leerefinedrobust": 19, "leerobust": 19, "smokingmccartnei": 19, "smoke": 19, "statu": 19, "deconvoluteblood450k": 19, "2024": [19, 21], "methyl": [20, 26, 28, 29, 30, 31, 35, 40], "id": [20, 27], "present": 20, "fac": 20, "optim": [20, 40], "unknown": 20, "ye": 20, "gse110554": 20, "flowsort": 20, "epic": 20, "ba": 20, "49": 20, "genom": [20, 28], "wide": 20, "profil": 20, "reveal": 20, "quantit": 20, "view": 20, "o": 20, "illumina450k": 20, "656": 20, "gse19711": 20, "unit": 20, "kingdom": 20, "ovar": 20, "illumina27k": 20, "540": 20, "No": [20, 34], "methylom": 20, "analysi": 20, "me": 20, "329": 20, "gse42861": 20, "differenti": 20, "rheumatoid": 20, "arthriti": 20, "689": 20, "dutch": 20, "popul": 20, "95": 20, "gse51032": 20, "itali": 20, "hugef": 20, "845": 20, "gse73103": 20, "mani": [20, 21, 39, 40], "obes": 20, "snp": 20, "strongli": 20, "met": 20, "355": 20, "gse69270": 20, "middl": 20, "indi": 20, "184": 20, "gse36054": 20, "children": 20, "192": 20, "gse64495": 20, "sever": [20, 21], "113": 20, "gse30870": 20, "newborn": 20, "nonagenarian": 20, "40": 20, "identif": 20, "signatur": [20, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34], "87": 20, "gse157131": 20, "peripher": 20, "leukocyt": 20, "fro": 20, "illuminaep": 20, "946": 20, "gse132203": 20, "gradi": 20, "trauma": 20, "project": 20, "795": 20, "enabl": [21, 39], "easi": [21, 39, 40], "versatil": [21, 39], "It": [21, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 39], "provid": [21, 27, 35, 39, 40], "tool": [21, 39, 40], "load": [21, 26, 27, 28, 39, 40], "publicli": [21, 39], "like": [21, 27, 39, 40], "gene": [21, 39], "express": [21, 39], "omnibu": [21, 39], "nation": [21, 33, 39], "health": [21, 33, 39, 40], "nutrit": [21, 33, 39], "examin": [21, 33], "survei": [21, 33, 39], "framingham": [21, 32, 39], "heart": [19, 21, 32, 39], "studi": [21, 32, 39], "also": [21, 39, 40], "contain": [21, 27, 28, 29, 31, 39], "implement": [21, 24, 25, 40], "common": [21, 39], "other": [21, 29, 39], "onli": [21, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 39], "few": [21, 39], "read": [21, 39], "about": [21, 27, 32, 33, 39, 40], "our": [21, 39, 40], "develop": [21, 39, 40], "organ": [21, 26, 27, 28], "individu": [21, 32, 33], "especi": 21, "methuselah": 21, "foundat": 21, "volo": 21, "host": 21, "2025": 21, "seri": [21, 29, 31], "platform": 21, "particip": 21, "ask": 21, "morbid": 21, "award": 21, "200k": 21, "pleas": [21, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 39, 40], "cite": 21, "preprint": 21, "m": 21, "emamifar": 21, "mart\u00ed": 21, "nez": 21, "c": 21, "d": 21, "poganik": 21, "j": 21, "r": 21, "gladyshev": 21, "2023": 21, "open": [21, 39, 40], "biorxiv": 21, "doi": 21, "1101": 21, "02": 21, "569722": 21, "quickstart": 21, "start": [21, 39], "roadmap": [21, 40], "see": [21, 40], "head": 21, "contribut": [21, 40], "biolog": [21, 34], "relationship": 21, "variou": [21, 34, 35], "point": 21, "class": [22, 23, 24, 25, 26, 27, 28, 34, 36, 38], "page": [24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34], "document": [24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 39], "explain": [24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34], "user": [24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34], "guid": [24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34], "big": [24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 40], "pictur": [24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34], "max_size_gb": 24, "serial": 24, "local": 24, "basic": 24, "lru": 24, "least": 24, "recent": 24, "cleanup": 24, "maintain": [24, 40], "within": [24, 34], "specifi": [24, 26, 28, 34], "maximum": 24, "limit": 24, "__init__": [24, 25, 26, 27, 28, 34], "initi": [24, 26, 27, 28, 34], "instanc": [24, 26, 27, 28, 34], "filesystem": 24, "storag": 24, "gigabyt": 24, "kei": [24, 25, 27], "retriev": [24, 25, 26, 34], "item": [24, 25, 27, 40], "its": [24, 26, 34], "none": [24, 25, 26, 27, 28, 30, 31, 34], "doe": [24, 25], "occur": 24, "ani": [24, 25, 27, 39, 40], "under": [24, 25], "manag": [24, 26, 34], "stai": 24, "clear": [24, 25], "remov": 24, "content": 24, "noth": 25, "alwai": 25, "simul": 25, "library_fil": 26, "cach": [26, 27, 40], "collect": [26, 34, 40], "respons": 26, "new": [26, 28], "ad": [26, 40], "runtim": 26, "current": [26, 40], "option": [26, 27, 28, 29, 30, 31, 34], "mechan": [26, 27], "adher": [26, 27], "interfac": [26, 27, 39], "modul": [26, 27, 35], "load_sourc": 26, "given": 26, "append": 26, "source_id": 26, "found": [26, 34], "otherwis": 26, "lookup_sourc": 26, "look": 26, "up": [26, 27, 40], "filter": [26, 28, 34], "list": [26, 28, 29, 30, 31], "match": [26, 34], "criteria": [26, 34], "repres": [27, 28, 32, 33], "singl": [27, 40], "encapsul": 27, "detail": [27, 28, 39], "includ": [27, 28, 39, 40], "rais": [27, 29, 34], "valueerror": [27, 29, 34], "requir": [27, 29, 40], "field": [27, 28], "parser": 27, "dure": 27, "pars": [27, 40], "dictionari": [27, 34], "typic": 27, "yaml": 27, "check": [27, 39], "essenti": 27, "attribut": 27, "handl": [27, 34], "dict": [27, 34], "properti": 27, "must": 27, "summari": [27, 28], "nocach": 27, "input": [27, 33, 34], "required_field": 27, "geograph": 27, "focu": 28, "facilit": 28, "access": [28, 40], "differ": [28, 40], "deep": 28, "quality_report": 28, "qualiti": 28, "control": 28, "report": 28, "section": 28, "percentag": [19, 28], "per": 28, "qualityreport": 28, "classmethod": 28, "matrix": 28, "either": 28, "directli": [28, 40], "union": 28, "cpg_sourc": 29, "required_cpg": 29, "extern": [29, 31], "while": 29, "need": [29, 40], "final": 29, "strategi": [29, 34, 35], "fill": [29, 30, 31], "certain": 29, "cpgs_to_imput": [30, 31], "itself": 30, "cpg_averag": 31, "measur": [32, 33], "Not": 33, "come": 33, "two": 33, "pass": 33, "should": 33, "later": 33, "2010": 33, "2012": 33, "model_definit": 34, "produc": 34, "search": 34, "definit": 34, "model_build": 34, "deconvolutionmodel": 34, "bound": 34, "from_definit": 34, "grimagemodel": 34, "linearmethylationmodel": 34, "sexestimationmodel": 34, "imputation_method": 34, "appli": 34, "algorithm": [], "dunedin": 34, "predefin": [], "gold": 34, "sesame_450k": 34, "sesam": 34, "specif": [], "design": [], "450k": 34, "arrai": [], "request": [34, 39], "possibli": 34, "enhanc": 34, "rtype": [], "keyerror": 34, "invalid": 34, "dynam": [], "appropri": [], "argument": [], "allow": 35, "preprocess": [], "accommod": [], "These": 35, "complex": [35, 40], "hybrid": 35, "inform": [], "examim": 39, "implemen": 39, "prereleas": 39, "version": 39, "There": 39, "mai": 39, "bug": 39, "subject": 39, "github": 39, "com": 39, "bio": 39, "homepag": 39, "io": 39, "pip": 39, "wa": 39, "correctli": 39, "out": [39, 40], "some": [39, 40], "team": 39, "ha": [39, 40], "answer": 39, "question": 39, "discuss": 39, "relat": 39, "find": 39, "replic": 39, "setup": 39, "repo": 39, "omic": 40, "revolut": 40, "transform": 40, "biologi": 40, "wealth": 40, "potenti": 40, "care": 40, "could": 40, "offer": 40, "comprehens": 40, "insight": 40, "one": 40, "surrog": 40, "endpoint": 40, "revers": 40, "therapeut": 40, "howev": 40, "obstruct": 40, "address": 40, "issu": 40, "modern": 40, "recogn": 40, "ve": 40, "adopt": 40, "scienc": 40, "invit": 40, "explor": 40, "mission": 40, "primarili": 40, "compris": 40, "thousand": 40, "ambit": 40, "help": 40, "achiev": 40, "larg": 40, "small": 40, "task": 40, "vari": 40, "skill": 40, "level": 40, "knowledg": 40, "right": 40, "improv": 40, "better": 40, "environ": 40, "speed": 40, "10x": 40, "integr": 40, "unifi": 40, "separ": 40, "structur": 40, "know": 40, "process": 40, "partial": 40, "autom": 40, "review": 40, "genet": 40, "diseas": [19, 40], "built": 40, "interest": 40, "share": 40, "reach": 40, "proprietari": 40, "would": 40, "api": 40, "talk": 40, "public": 40, "hear": 40, "privat": 40, "think": 40, "wai": 40, "fund": 40, "financi": 40, "monei": 40, "mainten": 40, "prize": 40, "pot": 40, "further": 40, "expans": 40, "contriubt": 40, "discord": 40, "server": 40, "48": [], "322": [], "hybrid_imput": 34, "educationmccartnei": 19, "educ": 19, "attain": 19, "totalcholesterolmccartnei": 19, "cholesterol": 19, "hdlcholesterolmccartnei": 19, "hdl": 19, "ldlcholesterolmccartnei": 19, "ldl": 19, "remnant": 19, "bodyfatmccartnei": 19, "bodi": 19, "fat": 19, "bmi_re": 19, "prostatecancerkirbi": 19, "2017": 19, "prostat": 19, "cancer": 19, "hepatoxu": 19, "circul": 19, "hepatocellular": 19, "carcinoma": 19, "cvd_westerman": 19, "coronari": 19, "ad_bahado": 19, "singh": 19, "alzheim": 19, "depressionbarbu": 19, "depress": 19}, "objects": {"biolearn": [[22, 0, 0, "-", "cache"], [23, 0, 0, "-", "data_library"], [35, 0, 0, "-", "imputation"], [37, 0, 0, "-", "load"], [38, 0, 0, "-", "model_gallery"]], "biolearn.cache": [[24, 1, 1, "", "LocalFolderCache"], [25, 1, 1, "", "NoCache"]], "biolearn.cache.LocalFolderCache": [[24, 2, 1, "", "__init__"], [24, 2, 1, "", "clear"], [24, 2, 1, "", "get"], [24, 2, 1, "", "store"]], "biolearn.cache.NoCache": [[25, 2, 1, "", "__init__"], [25, 2, 1, "", "clear"], [25, 2, 1, "", "get"], [25, 2, 1, "", "store"]], "biolearn.data_library": [[26, 1, 1, "", "DataLibrary"], [27, 1, 1, "", "DataSource"], [28, 1, 1, "", "GeoData"]], "biolearn.data_library.DataLibrary": [[26, 2, 1, "", "__init__"], [26, 2, 1, "", "get"], [26, 2, 1, "", "load_sources"], [26, 2, 1, "", "lookup_sources"]], "biolearn.data_library.DataSource": [[27, 3, 1, "", "REQUIRED_FIELDS"], [27, 2, 1, "", "__init__"], [27, 2, 1, "", "load"]], "biolearn.data_library.GeoData": [[28, 2, 1, "", "__init__"], [28, 2, 1, "", "copy"], [28, 3, 1, "", "dnam"], [28, 2, 1, "", "from_methylation_matrix"], [28, 3, 1, "", "metadata"], [28, 2, 1, "", "quality_report"]], "biolearn.imputation": [[29, 4, 1, "", "hybrid_impute"], [30, 4, 1, "", "impute_from_average"], [31, 4, 1, "", "impute_from_standard"]], "biolearn.load": [[32, 4, 1, "", "load_fhs"], [33, 4, 1, "", "load_nhanes"]], "biolearn.model_gallery": [[34, 1, 1, "", "ModelGallery"]], "biolearn.model_gallery.ModelGallery": [[34, 2, 1, "", "__init__"], [34, 2, 1, "", "get"], [34, 3, 1, "", "model_builders"], [34, 2, 1, "", "search"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:attribute", "4": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "attribute", "Python attribute"], "4": ["py", "function", "Python function"]}, "titleterms": {"team": 0, "core": 0, "develop": 0, "other": [0, 19], "contributor": 0, "epigenet": [1, 3, 4, 16], "clock": [1, 2, 4, 16, 19], "exampl": [1, 6, 9, 13, 14, 16, 21, 26, 27, 28, 31, 33, 34], "perform": 2, "custom": 2, "imput": [2, 29, 30, 31, 35], "load": [2, 3, 4, 7, 11, 12, 14, 32, 33, 37], "methyl": [2, 3, 4, 14], "data": [2, 3, 4, 7, 11, 12, 20, 35, 37], "from": [2, 3, 4], "geo": [2, 3, 4, 14, 20], "dataset": [2, 4, 14], "us": [2, 3, 4, 11, 12, 14, 26, 27, 28, 31, 33, 34], "refer": [2, 36], "up": [2, 3, 4, 11, 12], "target": 2, "run": [2, 3, 4, 12], "now": [2, 3, 4], "predict": [2, 3, 11], "befor": 2, "after": 2, "visual": 2, "comparison": 2, "ag": [2, 4, 7, 11], "you": 2, "can": 2, "also": 2, "build": [2, 11], "an": [2, 11, 12], "decor": 2, "bundl": 2, "get": [2, 3], "same": 2, "result": [2, 4, 12], "function": 2, "directli": 2, "your": 2, "down": 3, "syndrom": 3, "plot": [3, 4, 7, 12], "first": [3, 4], "some": [3, 4, 12], "librari": [3, 4], "model": [3, 11, 12, 14, 19], "score": 3, "final": [3, 4], "gener": 3, "boxplot": 3, "show": [3, 7], "power": 3, "three": 4, "differ": 4, "produc": 4, "extract": 4, "metadata": 4, "seaborn": 4, "comput": [5, 8, 10, 15, 17, 41], "time": [5, 8, 10, 15, 17, 41], "composit": [6, 16], "biomark": [6, 16, 40], "phenotyp": 7, "nhane": 7, "2010": 7, "calcul": 7, "biolog": 7, "base": 7, "phenotypicag": 7, "relat": 7, "between": 7, "chronolog": 7, "surviv": 7, "curv": 7, "peopl": 7, "acceler": 7, "older": 7, "v": 7, "deceler": 7, "younger": 7, "challeng": [9, 12, 16], "submiss": [9, 11, 12, 16], "competit": [11, 12], "exist": 11, "lin": 11, "train": 12, "elasticnet": 12, "seper": 12, "test": 12, "set": 12, "elast": 12, "net": 12, "see": 12, "how": 12, "good": 12, "our": 12, "i": 12, "through": 12, "save": 12, "output": 12, "file": 12, "deconvolut": [13, 14, 16], "blood": 14, "measur": 14, "epic": 14, "platform": 14, "estim": 14, "cell": 14, "proport": 14, "compar": 14, "known": 14, "fac": 14, "fluoresc": 14, "activ": 14, "sort": 14, "sourc": 20, "biolearn": [21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 37, 38, 39, 40], "featur": 21, "cach": [22, 24, 25], "data_librari": [23, 26, 27, 28], "datalibrari": [23, 26], "localfoldercach": 24, "nocach": 25, "datasourc": 27, "geodata": 28, "hybrid_imput": 29, "impute_from_averag": 30, "impute_from_standard": 31, "load_fh": 32, "load_nhan": 33, "model_galleri": [34, 38], "modelgalleri": [34, 38], "util": [35, 37], "api": 36, "quickstart": 39, "import": 39, "link": 39, "requir": 39, "instal": 39, "discord": 39, "server": 39, "issu": 39, "contribut": 39, "futur": 40, "join": 40, "u": 40}, "envversion": {"sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinxcontrib.bibtex": 9, "sphinx": 60}, "alltitles": {"Team": [[0, "team"]], "Core developers": [[0, "core-developers"]], "Other contributors": [[0, "other-contributors"]], "Epigenetic Clock Examples": [[1, "epigenetic-clock-examples"], [16, "epigenetic-clock-examples"]], "Performing custom imputations": [[2, "performing-custom-imputations"]], "Load methylation data from a GEO dataset to use as a reference for imputation": [[2, "load-methylation-data-from-a-geo-dataset-to-use-as-a-reference-for-imputation"]], "Load up a target dataset and run the imputation": [[2, "load-up-a-target-dataset-and-run-the-imputation"]], "Now run clock predictions on the dataset before and after": [[2, "now-run-clock-predictions-on-the-dataset-before-and-after"]], "Visualize the comparison of age predictions": [[2, "visualize-the-comparison-of-age-predictions"]], "You can also build an imputation decorator to bundle with the clock": [[2, "you-can-also-build-an-imputation-decorator-to-bundle-with-the-clock"]], "You get the same results as running the function directly on your dataset": [[2, "you-get-the-same-results-as-running-the-function-directly-on-your-dataset"]], "Down Syndrome Epigenetic Plotting": [[3, "down-syndrome-epigenetic-plotting"]], "First load up some methylation data from GEO using the data library": [[3, "first-load-up-some-methylation-data-from-geo-using-the-data-library"], [4, "first-load-up-some-methylation-data-from-geo-using-the-data-library"]], "Now run the down syndrome model on the data to get a score": [[3, "now-run-the-down-syndrome-model-on-the-data-to-get-a-score"]], "Finally generate a boxplot to show the predictive power": [[3, "finally-generate-a-boxplot-to-show-the-predictive-power"]], "\u201cEpigenetic Clocks\u201d in GEO Data": [[4, "epigenetic-clocks-in-geo-data"]], "Now run three different clocks on the dataset to produce epigenetic clock ages": [[4, "now-run-three-different-clocks-on-the-dataset-to-produce-epigenetic-clock-ages"]], "Finally extract the age data from the metadata from GEO and plot the results using seaborn": [[4, "finally-extract-the-age-data-from-the-metadata-from-geo-and-plot-the-results-using-seaborn"]], "Computation times": [[5, "computation-times"], [8, "computation-times"], [10, "computation-times"], [15, "computation-times"], [17, "computation-times"], [41, "computation-times"]], "Composite Biomarkers Examples": [[6, "composite-biomarkers-examples"], [16, "composite-biomarkers-examples"]], "\u201cPhenotypic Ages\u201d in NHANES Data": [[7, "phenotypic-ages-in-nhanes-data"]], "Loading NHANES 2010 data": [[7, "loading-nhanes-2010-data"]], "Calculate \u201cbiological age\u201d based on PhenotypicAge": [[7, "calculate-biological-age-based-on-phenotypicage"]], "Show relation between biological age and chronological age": [[7, "show-relation-between-biological-age-and-chronological-age"]], "Plot survival curve for people with accelerated aging (older biological age) vs decelerated aging (younger biological age)": [[7, "plot-survival-curve-for-people-with-accelerated-aging-older-biological-age-vs-decelerated-aging-younger-biological-age"]], "Challenge Submission Examples": [[9, "challenge-submission-examples"], [16, "challenge-submission-examples"]], "Deconvolution Examples": [[13, "deconvolution-examples"], [16, "deconvolution-examples"]], "\u201cDeconvolution Example\u201d": [[14, "deconvolution-example"]], "Load GEO blood methylation dataset measured on EPIC platform": [[14, "load-geo-blood-methylation-dataset-measured-on-epic-platform"]], "Estimate cell proportions using EPIC deconvolution model": [[14, "estimate-cell-proportions-using-epic-deconvolution-model"]], "Compare to known cell proportions measured with FACS (fluorescence-activated cell sorting)": [[14, "compare-to-known-cell-proportions-measured-with-facs-fluorescence-activated-cell-sorting"]], "Examples": [[16, "examples"]], "Biolearn": [[21, "biolearn"], [39, "biolearn"]], "Featured examples": [[21, "featured-examples"]], "biolearn.cache: Cache": [[22, "module-biolearn.cache"]], "biolearn.data_library: DataLibrary": [[23, "module-biolearn.data_library"]], "biolearn.cache.LocalFolderCache": [[24, "biolearn-cache-localfoldercache"]], "biolearn.cache.NoCache": [[25, "biolearn-cache-nocache"]], "biolearn.data_library.DataSource": [[27, "biolearn-data-library-datasource"]], "Examples using biolearn.data_library.DataSource": [[27, "examples-using-biolearn-data-library-datasource"]], "biolearn.data_library.GeoData": [[28, "biolearn-data-library-geodata"]], "Examples using biolearn.data_library.GeoData": [[28, "examples-using-biolearn-data-library-geodata"]], "biolearn.load.load_fhs": [[32, "biolearn-load-load-fhs"]], "biolearn.load.load_nhanes": [[33, "biolearn-load-load-nhanes"]], "Examples using biolearn.load.load_nhanes": [[33, "examples-using-biolearn-load-load-nhanes"]], "biolearn.load: Data Loading Utilities": [[37, "module-biolearn.load"]], "Quickstart": [[39, "quickstart"]], "Important links": [[39, "important-links"]], "Requirements": [[39, "requirements"]], "Install": [[39, "install"]], "Discord server": [[39, "discord-server"]], "Issues": [[39, "issues"]], "Contributing": [[39, "contributing"]], "Biolearn and the Future of Biomarkers": [[40, "biolearn-and-the-future-of-biomarkers"]], "Join Us": [[40, "join-us"]], "biolearn.data_library.DataLibrary": [[26, "biolearn-data-library-datalibrary"]], "Examples using biolearn.data_library.DataLibrary": [[26, "examples-using-biolearn-data-library-datalibrary"]], "biolearn.imputation.hybrid_impute": [[29, "biolearn-imputation-hybrid-impute"]], "biolearn.imputation.impute_from_average": [[30, "biolearn-imputation-impute-from-average"]], "biolearn.imputation.impute_from_standard": [[31, "biolearn-imputation-impute-from-standard"]], "Examples using biolearn.imputation.impute_from_standard": [[31, "examples-using-biolearn-imputation-impute-from-standard"]], "biolearn.model_gallery.ModelGallery": [[34, "biolearn-model-gallery-modelgallery"]], "Examples using biolearn.model_gallery.ModelGallery": [[34, "examples-using-biolearn-model-gallery-modelgallery"]], "biolearn.imputation: Data Imputation Utilities": [[35, "biolearn-imputation-data-imputation-utilities"]], "API References": [[36, "api-references"]], "biolearn.model_gallery: ModelGallery": [[38, "module-biolearn.model_gallery"]], "Building a competition submission using an existing model": [[11, "building-a-competition-submission-using-an-existing-model"]], "Loading up the data for the competition": [[11, "loading-up-the-data-for-the-competition"], [12, "loading-up-the-data-for-the-competition"]], "Use the Lin model to predict the age": [[11, "use-the-lin-model-to-predict-the-age"]], "Training an ElasticNet model": [[12, "training-an-elasticnet-model"]], "Load up some training data": [[12, "load-up-some-training-data"]], "Seperate data into training and test sets": [[12, "seperate-data-into-training-and-test-sets"]], "Train a model using elastic net": [[12, "train-a-model-using-elastic-net"]], "Plot the results to see how good our model is": [[12, "plot-the-results-to-see-how-good-our-model-is"]], "Run the challenge data through the model": [[12, "run-the-challenge-data-through-the-model"]], "Save the results as an output file for submission": [[12, "save-the-results-as-an-output-file-for-submission"]], "Clocks and Other Models": [[19, "clocks-and-other-models"]], "GEO Data Sources": [[20, "geo-data-sources"]]}, "indexentries": {"biolearn.cache": [[22, "module-biolearn.cache"]], "module": [[22, "module-biolearn.cache"], [23, "module-biolearn.data_library"], [35, "module-biolearn.imputation"], [37, "module-biolearn.load"], [38, "module-biolearn.model_gallery"]], "biolearn.data_library": [[23, "module-biolearn.data_library"]], "localfoldercache (class in biolearn.cache)": [[24, "biolearn.cache.LocalFolderCache"]], "__init__() (biolearn.cache.localfoldercache method)": [[24, "biolearn.cache.LocalFolderCache.__init__"]], "clear() (biolearn.cache.localfoldercache method)": [[24, "biolearn.cache.LocalFolderCache.clear"]], "get() (biolearn.cache.localfoldercache method)": [[24, "biolearn.cache.LocalFolderCache.get"]], "store() (biolearn.cache.localfoldercache method)": [[24, "biolearn.cache.LocalFolderCache.store"]], "nocache (class in biolearn.cache)": [[25, "biolearn.cache.NoCache"]], "__init__() (biolearn.cache.nocache method)": [[25, "biolearn.cache.NoCache.__init__"]], "clear() (biolearn.cache.nocache method)": [[25, "biolearn.cache.NoCache.clear"]], "get() (biolearn.cache.nocache method)": [[25, "biolearn.cache.NoCache.get"]], "store() (biolearn.cache.nocache method)": [[25, "biolearn.cache.NoCache.store"]], "datalibrary (class in biolearn.data_library)": [[26, "biolearn.data_library.DataLibrary"]], "__init__() (biolearn.data_library.datalibrary method)": [[26, "biolearn.data_library.DataLibrary.__init__"]], "get() (biolearn.data_library.datalibrary method)": [[26, "biolearn.data_library.DataLibrary.get"]], "load_sources() (biolearn.data_library.datalibrary method)": [[26, "biolearn.data_library.DataLibrary.load_sources"]], "lookup_sources() (biolearn.data_library.datalibrary method)": [[26, "biolearn.data_library.DataLibrary.lookup_sources"]], "datasource (class in biolearn.data_library)": [[27, "biolearn.data_library.DataSource"]], "required_fields (biolearn.data_library.datasource attribute)": [[27, "biolearn.data_library.DataSource.REQUIRED_FIELDS"]], "__init__() (biolearn.data_library.datasource method)": [[27, "biolearn.data_library.DataSource.__init__"]], "load() (biolearn.data_library.datasource method)": [[27, "biolearn.data_library.DataSource.load"]], "geodata (class in biolearn.data_library)": [[28, "biolearn.data_library.GeoData"]], "__init__() (biolearn.data_library.geodata method)": [[28, "biolearn.data_library.GeoData.__init__"]], "copy() (biolearn.data_library.geodata method)": [[28, "biolearn.data_library.GeoData.copy"]], "dnam (biolearn.data_library.geodata attribute)": [[28, "biolearn.data_library.GeoData.dnam"]], "from_methylation_matrix() (biolearn.data_library.geodata class method)": [[28, "biolearn.data_library.GeoData.from_methylation_matrix"]], "metadata (biolearn.data_library.geodata attribute)": [[28, "biolearn.data_library.GeoData.metadata"]], "quality_report() (biolearn.data_library.geodata method)": [[28, "biolearn.data_library.GeoData.quality_report"]], "hybrid_impute() (in module biolearn.imputation)": [[29, "biolearn.imputation.hybrid_impute"]], "impute_from_average() (in module biolearn.imputation)": [[30, "biolearn.imputation.impute_from_average"]], "impute_from_standard() (in module biolearn.imputation)": [[31, "biolearn.imputation.impute_from_standard"]], "load_fhs() (in module biolearn.load)": [[32, "biolearn.load.load_fhs"]], "load_nhanes() (in module biolearn.load)": [[33, "biolearn.load.load_nhanes"]], "modelgallery (class in biolearn.model_gallery)": [[34, "biolearn.model_gallery.ModelGallery"]], "__init__() (biolearn.model_gallery.modelgallery method)": [[34, "biolearn.model_gallery.ModelGallery.__init__"]], "get() (biolearn.model_gallery.modelgallery method)": [[34, "biolearn.model_gallery.ModelGallery.get"]], "model_builders (biolearn.model_gallery.modelgallery attribute)": [[34, "biolearn.model_gallery.ModelGallery.model_builders"]], "search() (biolearn.model_gallery.modelgallery method)": [[34, "biolearn.model_gallery.ModelGallery.search"]], "biolearn.imputation": [[35, "module-biolearn.imputation"]], "biolearn.load": [[37, "module-biolearn.load"]], "biolearn.model_gallery": [[38, "module-biolearn.model_gallery"]]}})