Search.setIndex({"docnames": ["authors", "auto_examples/00_omics_biomarkers/index", "auto_examples/00_omics_biomarkers/plot_custom_imputation_implementation", "auto_examples/00_omics_biomarkers/plot_down_syndrome_model", "auto_examples/00_omics_biomarkers/plot_epigenetic_clocks_on_geo", "auto_examples/00_omics_biomarkers/plot_local_data_load", "auto_examples/00_omics_biomarkers/plot_transcriptomic_clock", "auto_examples/00_omics_biomarkers/sg_execution_times", "auto_examples/01_composite_biomarkers/index", "auto_examples/01_composite_biomarkers/plot_nhanes", "auto_examples/01_composite_biomarkers/sg_execution_times", "auto_examples/02_challenge_submissions/index", "auto_examples/02_challenge_submissions/plot_challenge_data", "auto_examples/02_challenge_submissions/sg_execution_times", "auto_examples/02_challenge_submissions/simple_competiton_submission", "auto_examples/02_challenge_submissions/training_simple_model", "auto_examples/03_deconvolution/index", "auto_examples/03_deconvolution/plot_blood_deconvolution", "auto_examples/03_deconvolution/sg_execution_times", "auto_examples/index", "auto_examples/sg_execution_times", "changes/names", "clocks", "data", "index", "methylation-standard", "model-usage", "modules/cache", "modules/data_library", "modules/generated/biolearn.cache.LocalFolderCache", "modules/generated/biolearn.cache.NoCache", "modules/generated/biolearn.data_library.DataLibrary", "modules/generated/biolearn.data_library.DataSource", "modules/generated/biolearn.data_library.GeoData", "modules/generated/biolearn.imputation.hybrid_impute", "modules/generated/biolearn.imputation.impute_from_average", "modules/generated/biolearn.imputation.impute_from_standard", "modules/generated/biolearn.load.load_fhs", "modules/generated/biolearn.load.load_nhanes", "modules/generated/biolearn.model_gallery.ModelGallery", "modules/generated/biolearn.mortality.calculate_age_adjusted_c_index", "modules/generated/biolearn.mortality.calculate_c_index", "modules/generated/biolearn.mortality.calculate_log_rank_test", "modules/generated/biolearn.mortality.calculate_mortality_hazard_ratios", "modules/generated/biolearn.mortality.plot_hazard_ratios", "modules/generated/biolearn.mortality.run_predictions", "modules/imputation", "modules/index", "modules/load", "modules/model_gallery", "modules/mortality", "quickstart", "roadmap", "sg_execution_times"], "filenames": ["authors.rst", "auto_examples/00_omics_biomarkers/index.rst", "auto_examples/00_omics_biomarkers/plot_custom_imputation_implementation.rst", "auto_examples/00_omics_biomarkers/plot_down_syndrome_model.rst", "auto_examples/00_omics_biomarkers/plot_epigenetic_clocks_on_geo.rst", "auto_examples/00_omics_biomarkers/plot_local_data_load.rst", "auto_examples/00_omics_biomarkers/plot_transcriptomic_clock.rst", "auto_examples/00_omics_biomarkers/sg_execution_times.rst", "auto_examples/01_composite_biomarkers/index.rst", "auto_examples/01_composite_biomarkers/plot_nhanes.rst", "auto_examples/01_composite_biomarkers/sg_execution_times.rst", "auto_examples/02_challenge_submissions/index.rst", "auto_examples/02_challenge_submissions/plot_challenge_data.rst", "auto_examples/02_challenge_submissions/sg_execution_times.rst", "auto_examples/02_challenge_submissions/simple_competiton_submission.rst", "auto_examples/02_challenge_submissions/training_simple_model.rst", "auto_examples/03_deconvolution/index.rst", "auto_examples/03_deconvolution/plot_blood_deconvolution.rst", "auto_examples/03_deconvolution/sg_execution_times.rst", "auto_examples/index.rst", "auto_examples/sg_execution_times.rst", "changes/names.rst", "clocks.rst", "data.rst", "index.rst", "methylation-standard.rst", "model-usage.rst", "modules/cache.rst", "modules/data_library.rst", "modules/generated/biolearn.cache.LocalFolderCache.rst", "modules/generated/biolearn.cache.NoCache.rst", "modules/generated/biolearn.data_library.DataLibrary.rst", "modules/generated/biolearn.data_library.DataSource.rst", "modules/generated/biolearn.data_library.GeoData.rst", "modules/generated/biolearn.imputation.hybrid_impute.rst", "modules/generated/biolearn.imputation.impute_from_average.rst", "modules/generated/biolearn.imputation.impute_from_standard.rst", "modules/generated/biolearn.load.load_fhs.rst", "modules/generated/biolearn.load.load_nhanes.rst", "modules/generated/biolearn.model_gallery.ModelGallery.rst", "modules/generated/biolearn.mortality.calculate_age_adjusted_c_index.rst", "modules/generated/biolearn.mortality.calculate_c_index.rst", "modules/generated/biolearn.mortality.calculate_log_rank_test.rst", "modules/generated/biolearn.mortality.calculate_mortality_hazard_ratios.rst", "modules/generated/biolearn.mortality.plot_hazard_ratios.rst", "modules/generated/biolearn.mortality.run_predictions.rst", "modules/imputation.rst", "modules/index.rst", "modules/load.rst", "modules/model_gallery.rst", "modules/mortality.rst", "quickstart.md", "roadmap.rst", "sg_execution_times.rst"], "titles": ["Team", "Omics Biomarker Examples", "Performing custom imputations", "Down Syndrome Epigenetic Plotting", "\u201cEpigenetic Clocks\u201d in GEO Data", "Local Data Loading", "\u201cTranscriptomic Clock\u201d using GEO Data", "Computation times", "Composite Biomarkers Examples", "\u201cPhenotypic Ages\u201d in NHANES Data", "Computation times", "Challenge Submission Examples", "Exploring the Challenge Data", "Computation times", "Building a competition submission using an existing model", "Training an ElasticNet model", "Deconvolution Examples", "\u201cDeconvolution Example\u201d", "Computation times", "Examples", "Computation times", "&lt;no title&gt;", "Clocks and Other Models", "GEO Data Sources", "Biolearn", "DNA Methylation Array Data Standard V-2410", "Model IP and Usage", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.cache</span></code>: Cache", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.data_library</span></code>: DataLibrary", "biolearn.cache.LocalFolderCache", "biolearn.cache.NoCache", "biolearn.data_library.DataLibrary", "biolearn.data_library.DataSource", "biolearn.data_library.GeoData", "biolearn.imputation.hybrid_impute", "biolearn.imputation.impute_from_average", "biolearn.imputation.impute_from_standard", "biolearn.load.load_fhs", "biolearn.load.load_nhanes", "biolearn.model_gallery.ModelGallery", "biolearn.mortality.calculate_age_adjusted_c_index", "biolearn.mortality.calculate_c_index", "biolearn.mortality.calculate_log_rank_test", "biolearn.mortality.calculate_mortality_hazard_ratios", "biolearn.mortality.plot_hazard_ratios", "biolearn.mortality.run_predictions", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.imputation</span></code>: Data Imputation Functions", "API References", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.load</span></code>: Data Loading Utilities", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.model_gallery</span></code>: ModelGallery", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.mortality</span></code>: Mortality Predictor Evaluation", "Quickstart", "Biolearn and the Future of Biomarkers", "Computation times"], "terms": {"thi": [0, 2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 49, 50, 51, 52], "work": [0, 52], "i": [0, 22, 23, 24, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 47, 51, 52], "made": [0, 52], "avail": [0, 24, 25, 39, 51, 52], "commun": [0, 14], "peopl": 0, "which": [0, 15, 29, 30, 33, 38], "origin": [0, 2], "from": [0, 5, 7, 8, 9, 10, 13, 14, 15, 17, 18, 19, 20, 22, 23, 24, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 40, 41, 42, 43, 44, 45, 46, 51, 52, 53], "biomark": [0, 11, 23, 24, 31, 49, 51], "ag": [0, 8, 10, 11, 12, 15, 19, 22, 23, 24, 25, 38, 40, 41, 42, 43, 51, 52, 53], "consortium": [0, 24], "The": [0, 25, 29, 30, 31, 39, 40, 42, 43, 51, 52], "biolearn": [0, 2, 3, 4, 6, 9, 12, 14, 15, 17, 22, 23, 47], "ar": [0, 2, 15, 23, 24, 25, 26, 31, 32, 33, 34, 38, 39, 40, 41, 42, 43, 45, 51, 52], "mahdi": 0, "moqri": [0, 24], "research": [0, 24, 31, 52], "boa": 0, "execut": [0, 7, 10, 13, 18, 19, 20, 51, 53], "martin": 0, "perez": [0, 24], "guevara": [0, 24], "machin": 0, "learn": [0, 5, 24, 51, 52], "engin": 0, "seth": 0, "paulson": [0, 24], "softwar": 0, "architect": 0, "albert": 0, "ying": [0, 24], "harvard": 0, "universii": 0, "alec": 0, "eam": 0, "dayoon": 0, "kwon": [0, 24], "univers": 0, "california": 0, "mehrnoosh": 0, "emamifar": [0, 24], "northeastern": 0, "ai4lif": 0, "institut": 0, "special": 0, "thank": 0, "scientif": 0, "member": [0, 12], "plot": [1, 2, 7, 8, 12, 16, 19, 24, 44, 53], "result": [1, 3, 14, 16, 19, 40, 41, 42, 43, 45], "run": [1, 5, 9, 14, 16, 17, 19, 22, 24, 39, 45, 51], "base": [1, 8, 15, 19, 31, 34, 39], "model": [1, 2, 6, 11, 13, 19, 31, 39, 40, 42, 43, 45, 49, 52, 53], "function": [1, 19, 32, 34, 35, 36, 37, 38, 40, 41, 42, 43, 44, 45, 47, 48, 50], "includ": [1, 5, 19, 25, 32, 33, 51, 52], "methyl": [1, 5, 19, 23, 31, 33, 34, 35, 36, 46, 52], "transcriptom": [1, 7, 19, 53], "data": [1, 7, 8, 10, 11, 13, 17, 19, 24, 27, 28, 29, 31, 32, 33, 34, 37, 38, 39, 40, 41, 42, 43, 44, 45, 51, 52, 53], "clock": [1, 7, 8, 19, 24, 26, 32, 49, 51, 53], "us": [1, 7, 8, 11, 13, 19, 23, 24, 25, 26, 29, 30, 34, 35, 37, 40, 41, 42, 43, 44, 51, 52, 53], "geo": [1, 7, 19, 25, 31, 32, 52, 53], "epigenet": [1, 2, 7, 19, 23, 24, 32, 52, 53], "down": [1, 7, 15, 19, 22, 53], "syndrom": [1, 7, 19, 22, 53], "perform": [1, 7, 9, 19, 36, 53], "custom": [1, 7, 19, 36, 53], "imput": [1, 4, 5, 7, 19, 39, 53], "go": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17], "end": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17], "download": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 19], "full": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17], "exampl": [2, 3, 4, 5, 6, 7, 9, 10, 12, 13, 14, 15, 18, 20, 51, 53], "code": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 19, 24, 51, 52], "browser": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17], "via": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 52], "binder": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17], "demonstr": [2, 24], "eas": 2, "librari": [2, 23, 24, 27, 31, 39, 51, 52], "data_librari": [2, 3, 4, 5, 6, 12, 14, 15, 17, 51], "import": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17], "datalibrari": [2, 3, 4, 6, 12, 15, 17, 32, 51], "geodata": [2, 5, 14, 15, 32, 43, 45], "comput": [2, 24], "averag": [2, 25, 34, 35, 39, 46], "reference_dataset": 2, "gse40279": [2, 15, 23], "reference_averag": 2, "dnam": [2, 4, 5, 12, 15, 25, 33, 34, 35, 36], "mean": [2, 15], "axi": 2, "1": [2, 5, 6, 9, 10, 12, 15, 17, 18, 25, 43, 52], "second": [2, 3, 4, 5, 6, 9, 12, 17], "target_dataset": 2, "gse51057": [2, 23], "impute_from_standard": 2, "imputed_data": 2, "imputed_dataset": 2, "metadata": [2, 3, 6, 15, 17, 32, 33, 40, 41, 42, 43], "galleri": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 19, 22, 45, 53], "compar": [2, 24], "model_galleri": [2, 3, 4, 6, 14, 17], "modelgalleri": [2, 3, 4, 6, 14, 17], "clock_model": 2, "horvathv1": [2, 4, 12, 22, 26, 45], "original_age_predict": 2, "imputed_age_predict": 2, "seaborn": [2, 3, 9, 12, 17], "sn": [2, 3, 4, 6, 9, 12, 17], "panda": [2, 4, 12, 15, 33, 37, 38], "pd": [2, 4, 12, 15, 34, 35, 36, 40, 41, 42, 43, 44, 45], "matplotlib": [2, 3, 6, 9, 12, 15, 17], "pyplot": [2, 3, 6, 9, 12, 15, 17], "plt": [2, 3, 6, 9, 12, 15, 17], "prepar": [2, 52], "comparison_data": 2, "datafram": [2, 4, 12, 15, 33, 34, 35, 36, 37, 38, 40, 41, 42, 43, 44, 45], "creat": [2, 3, 12, 33, 51], "scatter": [2, 4, 12, 15], "figur": [2, 3, 12, 15, 17], "figsiz": [2, 3, 12, 15, 17], "8": [2, 5, 10, 12, 13, 15, 34], "6": [2, 3, 7, 12, 15, 18, 23, 53], "scatterplot": [2, 9, 17], "x": [2, 3, 6, 9, 12, 15, 17], "y": [2, 3, 6, 9, 12, 15, 17], "titl": [2, 3, 6, 12, 15, 17, 23, 32], "v": [2, 15, 24, 51], "xlabel": [2, 6, 9, 12, 15, 17], "ylabel": [2, 6, 9, 12, 15, 17], "grid": [2, 12, 15], "true": [2, 3, 12, 15], "show": [2, 5, 6, 12, 14, 15, 24], "imputationdecor": 2, "defin": [2, 15, 31], "method": [2, 32, 39, 46], "def": 2, "custom_impute_method": 2, "dnam_data": 2, "needed_cpg": 2, "return": [2, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45], "wrap": 2, "decorated_clock": 2, "decorated_clock_predict": 2, "verifi": [2, 51], "direct": 2, "approach": [2, 52], "same_result": 2, "all": [2, 5, 19, 25, 26, 29, 30, 31, 33, 35, 36, 53], "print": [2, 6, 15], "total": [2, 3, 4, 5, 6, 7, 9, 10, 12, 13, 17, 18, 20, 22, 24, 53], "time": [2, 3, 4, 5, 6, 9, 12, 15, 17, 41, 52], "script": [2, 3, 4, 5, 6, 9, 12, 17], "3": [2, 3, 5, 12, 13, 15, 51], "minut": [2, 3, 4, 5, 6, 9, 12, 17, 52], "9": [2, 9, 12, 18], "803": 2, "estim": [2, 3, 4, 5, 6, 9, 12, 14, 15], "memori": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 23], "usag": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 22], "11895": 2, "mb": [2, 3, 4, 5, 6, 7, 9, 10, 12, 13, 14, 15, 17, 18, 20, 53], "jupyt": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 19, 51], "notebook": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 19, 51], "plot_custom_imputation_implement": [2, 7, 53], "ipynb": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17], "python": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 19, 51], "sourc": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 19, 22, 24, 31, 32, 34, 51, 52], "py": [2, 3, 4, 5, 6, 7, 9, 10, 12, 13, 14, 15, 17, 18, 53], "gener": [2, 4, 5, 6, 9, 11, 12, 14, 15, 17, 19, 33, 52], "sphinx": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 19], "your": [3, 4, 5, 6, 9, 12, 14, 15, 17, 24, 52], "dna": [3, 4, 17, 22, 23, 31, 34, 35, 36, 46], "how": [3, 12, 14, 24, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 51, 52], "samepl": 3, "can": [3, 14, 22, 24, 25, 26, 27, 31, 33, 39, 51, 52], "distiguish": 3, "test_data": 3, "gse52588": [3, 23], "downsyndrom": [3, 22, 26], "set": [3, 4, 5, 23, 31, 32, 39, 46, 50], "index": [3, 4, 15, 40, 41], "merg": 3, "set_index": 3, "astyp": [3, 17], "str": [3, 29, 30, 31, 33, 34, 35, 36, 39], "inplac": 3, "merged_data": 3, "join": [3, 24], "category_ord": 3, "healthi": [3, 23], "palett": 3, "green": 3, "red": 3, "configur": [3, 32, 52], "10": [3, 12, 15, 17, 24, 51, 53], "ax": [3, 9], "disease_st": 3, "width": 3, "0": [3, 4, 5, 6, 7, 9, 12, 13, 14, 15, 17, 20, 25, 34, 43, 53], "order": 3, "stripplot": 3, "jitter": 3, "hue": 3, "dodg": 3, "fals": 3, "legend": [3, 17], "set_titl": 3, "despin": 3, "721": 3, "945": 3, "plot_down_syndrome_model": [3, 7, 53], "calcul": [4, 6, 15, 39, 40, 41, 42, 43], "multipl": [4, 23, 25], "them": [4, 17, 19, 31, 52], "against": 4, "chronolog": [4, 24], "gse41169": [4, 23], "blood": [4, 8, 9, 12, 19, 22, 23, 52], "data_sourc": 4, "get": [4, 6, 12, 14, 15, 17, 24, 29, 30, 31, 39, 51], "note": [4, 5, 15], "default": [4, 5, 31, 32, 34, 39], "miss": [4, 5, 30, 32, 33, 34, 35, 36, 39, 46], "To": [4, 51], "chang": [4, 23, 51], "behavior": [4, 51], "paramet": [4, 15, 29, 30, 31, 32, 33, 34, 35, 36, 38, 39, 40, 41, 42, 43, 44, 45], "when": 4, "horvath_result": 4, "predict": [4, 12, 15, 17, 22, 24, 39, 45], "hannum_result": 4, "hannum": [4, 12, 22, 26], "phenoage_result": 4, "phenoag": [4, 22, 26], "actual_ag": 4, "plot_data": 4, "horvath": [4, 24, 51], "relplot": 4, "kind": 4, "axisgrid": 4, "facetgrid": 4, "object": [4, 15, 29, 31, 32, 33, 39, 40, 41, 42, 43, 45], "0x713789f3e9b0": 4, "2": [4, 9, 12, 15, 23, 25], "616": 4, "1587": 4, "plot_epigenetic_clocks_on_geo": [4, 7, 53], "gse134080": [6, 23], "rna_model": 6, "transcriptomicpredictionmodel": [6, 22, 26], "imputation_method": [6, 39], "none": [6, 12, 25, 29, 30, 31, 32, 33, 35, 36, 39, 44], "scipi": 6, "stat": 6, "pearsonr": 6, "predicted_valu": 6, "true_valu": 6, "correl": [6, 15], "_": [6, 15], "f": [6, 15], "pearson": 6, "coeffici": [6, 15, 22], "set_them": 6, "style": 6, "whitegrid": 6, "regplot": [6, 12], "scatter_kw": 6, "alpha": [6, 15], "5": [6, 7, 9, 15, 17, 25], "7025211774998293": 6, "739": 6, "406": 6, "plot_transcriptomic_clock": [6, 7, 53], "03": [], "14": 12, "879": [], "4": [5, 12], "file": [7, 10, 13, 14, 18, 20, 22, 31, 32, 33, 52, 53], "auto_exampl": [7, 10, 13, 18, 20], "00_omics_biomark": [7, 53], "mem": [7, 10, 13, 18, 20, 53], "09": [], "00": [5, 7, 10, 12, 13, 18, 20, 53], "02": 24, "01": [10, 15], "944": [], "surviv": [8, 19, 24], "curv": [8, 19], "standard": [5, 8, 19, 39], "test": [8, 19, 42, 52], "phenotyp": [8, 10, 19, 38, 53], "nhane": [8, 10, 19, 24, 38, 53], "exam": 9, "analys": [9, 24, 51], "load_nhan": 9, "year": [9, 22, 25, 38, 40, 42, 43], "df": [9, 12, 15, 37, 38], "years_until_death": [9, 40, 41, 42, 43], "months_until_death": 9, "12": [9, 12, 23, 24], "hematologi": 9, "phenotypic_ag": 9, "": [9, 22, 24, 32, 39, 52], "lifelin": 9, "kaplanmeierfitt": 9, "biologically_old": 9, "subplot": 9, "group": [9, 38], "ix": 9, "t": [9, 15, 17], "e": 9, "is_dead": 9, "kmf": 9, "fit": [9, 15], "label": 9, "plot_survival_funct": 9, "text": [9, 17], "23": 9, "52222222222222": 9, "271": [9, 10], "plot_nhan": [9, 10, 53], "01_composite_biomark": [10, 53], "build": [11, 13, 19, 39, 52, 53], "competit": [11, 13, 19, 39, 53], "an": [11, 13, 19, 23, 24, 29, 30, 31, 32, 33, 34, 36, 37, 38, 39, 52, 53], "exist": [11, 13, 19, 25, 29, 34, 39, 53], "explor": [11, 13, 19, 45, 52, 53], "train": [11, 13, 19, 31, 53], "elasticnet": [11, 13, 19, 31, 53], "challenge_data": [12, 14, 15], "boachallengedata": [12, 23], "option": [12, 31, 32, 33, 34, 35, 36, 39], "displai": [12, 23], "max_column": 12, "gsm7866999": 12, "gsm7867305": 12, "gsm7867077": 12, "gsm7867431": 12, "gsm7867126": 12, "gsm7867000": 12, "cg00000029": [5, 12, 25], "720084": 12, "453491": 12, "535564": 12, "487361": 12, "337215": 12, "062426": 12, "cg00000109": [5, 12, 25], "938044": 12, "925149": 12, "949446": 12, "938523": 12, "940187": 12, "942090": 12, "cg00000155": [5, 12, 25], "957093": 12, "959030": 12, "948919": 12, "966912": 12, "968026": 12, "970274": 12, "cg00000158": 12, "965191": 12, "957272": 12, "967278": 12, "971039": 12, "969745": 12, "969660": 12, "cg00000165": 12, "144061": 12, "122099": 12, "167346": 12, "156199": 12, "133038": 12, "641788": 12, "rs9292570": 12, "523412": 12, "980810": 12, "986609": 12, "525876": 12, "537722": 12, "538440": 12, "rs9363764": 12, "960815": 12, "038638": 12, "451884": 12, "978343": 12, "971838": 12, "973633": 12, "rs951295": 12, "971034": 12, "558011": 12, "553397": 12, "485287": 12, "526686": 12, "499980": 12, "rs966367": 12, "455920": 12, "963994": 12, "453094": 12, "455700": 12, "029342": 12, "026907": 12, "rs9839873": 12, "599735": 12, "595571": 12, "630871": 12, "576566": 12, "579195": 12, "973815": 12, "930659": 12, "row": [12, 15, 25, 33, 34, 35, 36, 37, 38, 40, 41, 42, 43], "500": [12, 23], "column": [12, 14, 15, 25, 33, 34, 35, 36, 37, 38, 40, 41, 42, 43, 45], "gsm7867173": 12, "gsm7867127": 12, "gsm7867083": 12, "gsm7867095": 12, "gsm7867115": 12, "gsm7867385": 12, "alamartargetid": 12, "t10319": 12, "13": 12, "387778": 12, "948052": 12, "256551": 12, "062449": 12, "390582": 12, "587867": 12, "t10413": 12, "16": 12, "674015": 12, "15": 12, "992939": 12, "547946": 12, "827118": 12, "872686": 12, "584386": 12, "t10466": 12, "890963": 12, "556167": 12, "528756": 12, "683530": 12, "667331": 12, "483932": 12, "t10563": 12, "933342": 12, "323690": 12, "271350": 12, "993500": 12, "11": 12, "699095": 12, "860206": 12, "t10876": 12, "063506": 12, "429302": 12, "582464": 12, "978461": 12, "654607": 12, "931614": 12, "t8333": 12, "725785": 12, "537733": 12, "848917": 12, "729721": 12, "787099": 12, "055526": 12, "t8334": 12, "503408": 12, "103376": 12, "381374": 12, "070087": 12, "317681": 12, "314627": 12, "t8358": 12, "734120": 12, "238780": 12, "673970": 12, "103683": 12, "949114": 12, "794795": 12, "t8367": 12, "062422": 12, "291734": 12, "604654": 12, "803189": 12, "313347": 12, "830717": 12, "t8399": 12, "599597": 12, "642498": 12, "590723": 12, "073775": 12, "184558": 12, "040475": 12, "374": 12, "503": 12, "util": [5, 12], "get_data_fil": 12, "read_csv": 12, "alamar_refer": 12, "csv": [5, 12, 14, 25, 33], "uniprotid": 12, "target": 12, "proteinnam": 12, "t10034": 12, "o75888": 12, "tnfsf13": 12, "tnf": 12, "superfamili": 12, "o95760": 12, "il33": 12, "interleukin": 12, "33": [5, 12], "t10412": 12, "p01258": 12, "calca": 12, "calcitonin": 12, "cleav": 12, "katacalcin": 12, "t10417": 12, "p01903": 12, "hla": 12, "dra": 12, "major": 12, "histocompat": 12, "complex": [12, 46, 52], "class": [12, 27, 28, 29, 30, 31, 32, 33, 39, 47, 49], "ii": 12, "dr": 12, "t10433": 12, "p02745": 12, "c1qa": 12, "complement": 12, "c1q": 12, "A": [12, 20, 29, 30, 31, 32, 33, 37, 38, 39, 40, 41, 42, 43, 44, 45], "chain": 12, "317": 12, "t8244": 12, "p35247": 12, "sftpd": 12, "pulmonari": 12, "surfact": 12, "associ": [12, 15, 23, 25, 33], "d": [12, 23, 24], "318": 12, "t8246": 12, "p22303": 12, "ach": 12, "acetylcholinesteras": 12, "319": 12, "t8254": 12, "q8n474": 12, "sfrp1": 12, "secret": 12, "frizzl": 12, "relat": [12, 23, 51], "320": 12, "t8355": 12, "p0dji8": 12, "saa1": 12, "serum": 12, "amyloid": 12, "321": 12, "t9441": 12, "o15240": 12, "vgf": 12, "nerv": 12, "growth": 12, "factor": 12, "induc": 12, "322": 12, "ethnic": 12, "race1": 12, "sex": [12, 22, 23, 25], "subject_id": 12, "tissu": [12, 22, 23, 39], "gsm7866964": 12, "31": 12, "35": [7, 12, 53], "non": [12, 26], "hispan": 12, "white": 12, "boa1": 12, "gsm7866965": 12, "79": 12, "45": 12, "asian": 12, "boa2": 12, "gsm7866966": 12, "60": [12, 15], "42": [12, 15, 25], "boa3": 12, "gsm7866967": 12, "59": 12, "24": 12, "boa4": 12, "gsm7866968": 12, "22": 12, "41": 12, "boa5": 12, "p499": 12, "80": 12, "98": 12, "nan": [5, 12, 25], "p500": 12, "89": [12, 25], "p501": 12, "92": [12, 23], "p502": 12, "69": 12, "p503": 12, "68": [12, 23], "18": 12, "651": 12, "7": [12, 15, 25], "mortal": [12, 22, 24], "run_predict": 12, "prediction_dict": 12, "21": [12, 13], "822423": 12, "17": 12, "303025": 12, "56": 12, "266752": 12, "52": 12, "803410": 12, "39": [12, 25], "660930": 12, "36": [5, 12], "211078": 12, "29": [12, 25], "806215": 12, "32": [12, 23], "821235": 12, "gsm7867457": 12, "51": 12, "285108": 12, "49": [12, 23], "308239": 12, "gsm7867136": 12, "37": 12, "539954": 12, "494127": 12, "gsm7867452": 12, "74": 12, "975617": 12, "639608": 12, "083565": 12, "906176": 12, "91": 12, "428559": 12, "870807": 12, "75": 12, "005273": 12, "77": 12, "630528": 12, "assum": [12, 25], "name": [12, 15, 22, 26, 39, 40, 41, 42, 43, 45], "should": [12, 25, 38, 40, 42, 43], "have": [5, 12, 15, 19, 25, 51, 52], "regress": 12, "line": [12, 15, 24, 51], "ci": 12, "304": [12, 13], "14385": 12, "plot_challenge_data": [12, 13, 53], "02_challenge_submiss": [13, 53], "challeng": [13, 14, 23, 24, 45, 52, 53], "14384": 13, "submiss": [13, 39, 53], "simple_competiton_submiss": [13, 14, 53], "000": [7, 13, 20, 53], "training_simple_model": [13, 15, 53], "you": [14, 15, 19, 24, 26, 51, 52], "warmup": [14, 15], "link": [14, 15], "here": [14, 15, 22], "http": [5, 14, 15, 24, 51], "www": [14, 15], "synaps": [14, 15, 24], "org": [14, 15, 24], "syn52966292": [14, 15], "wiki": [14, 15], "625231": [14, 15], "downloaded_data_file_path": [14, 15], "add": [14, 15], "path": [14, 15, 29, 31, 32, 33, 52], "from_methylation_matrix": [5, 14, 15, 33], "score": 14, "best": 14, "so": 14, "let": 14, "save": 14, "correct": [14, 25], "format": [5, 14, 23, 28, 31, 32, 52], "now": [14, 52], "we": [14, 24, 25, 51, 52], "expect": [14, 51], "rename_axi": 14, "sampleid": [14, 15, 25], "renam": 14, "predictedag": [14, 15], "to_csv": 14, "lin_submiss": 14, "submit": [14, 15], "follow": [14, 19, 25, 43], "instruct": [14, 19, 51], "below": [14, 34], "simpl": [15, 46], "narrow": 15, "what": [15, 26], "site": [5, 15, 25, 33, 34, 35, 36], "take": [15, 52], "long": 15, "numpi": 15, "np": 15, "sklearn": 15, "linear_model": 15, "linearregress": 15, "extract": [15, 52], "transpos": 15, "valu": [5, 15, 29, 30, 34, 35, 36, 39, 40, 41, 42, 43, 44, 45, 46], "sampl": [5, 15, 23, 25, 33, 34, 35, 36, 40, 42, 43, 52], "cpg": [15, 25, 34, 35, 36], "bootstrap": 15, "featur": [15, 51], "select": [15, 39], "n_bootstrap": 15, "20": [5, 15], "threshold": [15, 34], "05": 15, "store": [15, 23, 29, 30, 31], "count": [5, 15], "each": [15, 25, 33, 37, 38, 40, 41, 42, 43, 44, 45], "deem": 15, "signific": 15, "cpg_count": 15, "zero": 15, "shape": [15, 23], "begin": 15, "iter": 15, "rang": [15, 25], "replac": [5, 15, 34], "sample_idx": 15, "random": 15, "choic": 15, "size": [15, 29], "x_sampl": 15, "y_sampl": 15, "identifi": [15, 31, 33, 52], "magnitud": [15, 52], "significant_cpg": 15, "where": [15, 19, 24, 33, 37, 38, 45, 52], "ab": 15, "coef_": 15, "determin": [15, 26, 34], "stabl": 15, "stable_cpg_sit": 15, "stable_cpg_nam": 15, "tolist": 15, "more": [15, 24, 25, 46, 51, 52], "than": [15, 25], "model_select": 15, "train_test_split": 15, "top_sites_df": 15, "x_train": 15, "x_test": 15, "y_train": 15, "y_test": 15, "test_siz": 15, "random_st": 15, "regular": 15, "strength": 15, "l1_ratio": 15, "mix": 15, "between": [15, 24, 25, 40, 42, 43], "l1": 15, "l2": 15, "lasso": 15, "ridg": 15, "metric": 15, "mean_squared_error": 15, "max_it": 15, "10000": 15, "evalu": 15, "y_pred": 15, "mse": 15, "squar": 15, "error": [15, 29, 51], "edgecolor": 15, "w": [15, 23], "linewidth": 15, "min": 15, "max": 15, "k": [15, 24], "lw": 15, "refer": [15, 23, 24, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 51], "actual": [15, 51], "both": [15, 23, 33, 34, 52], "linestyl": 15, "4f": 15, "absolut": 15, "mae": 15, "pruned_data": 15, "fillna": 15, "challenge_result": 15, "predicted_age_df": 15, "dataset": [16, 19, 23, 34, 35, 36, 39, 40, 41, 42, 46, 52], "known": [16, 19, 40, 42, 43], "cell": [16, 19, 22, 23], "proport": [16, 19, 22, 43], "type": [17, 23, 29, 30, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 45], "epic_facs_salas_18": 17, "gse112618": [17, 23], "epic_facs_cell_prop": 17, "deconvolutebloodep": [17, 22, 26], "set_styl": 17, "tick": 17, "neutrophils_proport": 17, "float": [17, 29, 34], "loc": 17, "neutrophil": 17, "monocytes_proport": 17, "monocyt": 17, "nk_proport": 17, "nk_cell": 17, "bcell_proport": 17, "b_cell": 17, "cd4t_proport": 17, "cd4_t_cell": 17, "cd8t_proport": 17, "cd8_t_cell": 17, "nk": 17, "b": 17, "cd4": 17, "cd8": 17, "valid": [17, 23], "934": [17, 18], "plot_blood_deconvolut": [17, 18, 53], "03_deconvolut": [18, 53], "deconvolut": [18, 53], "If": [19, 24, 25, 26, 31, 32, 33, 34, 39, 51, 52], "want": [19, 52], "make": 19, "sure": 19, "directori": [19, 29], "write": [19, 52], "permiss": 19, "copi": [19, 33], "instal": 19, "biolean": 19, "manual": 19, "auto_examples_python": 19, "zip": 19, "auto_examples_jupyt": 19, "n": [20, 24], "support": [22, 23, 24, 31, 38, 39, 52], "number": [22, 23, 25, 38], "easili": [22, 24, 31, 51], "speci": [22, 39], "2013": 22, "human": [22, 23, 52], "multi": [22, 24, 52], "paper": [22, 24, 26, 51], "lin": [22, 26], "2016": 22, "2018": 22, "yingcausag": [22, 26], "2022": 22, "yingdamag": [22, 26], "yingadaptag": [22, 26], "horvathv2": [22, 26], "skin": 22, "pedb": [22, 26], "2019": 22, "buccal": 22, "zhang_10": [22, 26], "risk": 22, "dunedinpoam38": [22, 26], "2020": 22, "rate": 22, "dunedinpac": [22, 24, 26, 39, 51], "grimagev1": [22, 26], "adjust": [22, 40, 41, 42], "grimagev2": [22, 26], "alcoholmccartnei": [22, 26], "alcohol": 22, "consumpt": 22, "dnamtl": [22, 26], "adipos": 22, "telomer": 22, "length": [22, 25], "hrsinchphenoag": [22, 26], "knight": [22, 26], "cord": [22, 23], "gestat": 22, "leecontrol": [22, 26], "placenta": 22, "sexestim": [22, 26], "2021": 22, "leerefinedrobust": [22, 26], "leerobust": [22, 26], "smokingmccartnei": [22, 26], "smoke": [22, 23], "statu": [22, 26], "deconvoluteblood450k": [22, 26], "2024": [22, 23, 24], "twelvecelldeconvolutebloodep": [22, 26], "bmi_mccartnei": [22, 26], "bmi": 22, "educationmccartnei": [22, 26], "educ": 22, "attain": 22, "totalcholesterolmccartnei": [22, 26], "cholesterol": 22, "hdlcholesterolmccartnei": [22, 26], "hdl": 22, "ldlcholesterolmccartnei": [22, 26], "ldl": 22, "remnant": 22, "bodyfatmccartnei": [22, 26], "percentag": [22, 33], "bodi": 22, "fat": 22, "bmi_re": [22, 26], "prostatecancerkirbi": [22, 26], "2017": 22, "prostat": 22, "cancer": 22, "hepatoxu": [22, 26], "circul": 22, "hepatocellular": 22, "carcinoma": 22, "cvd_westerman": [22, 26], "coronari": 22, "heart": [22, 24, 37, 51], "diseas": [22, 23, 25, 52], "ad_bahado": [22, 26], "singh": [22, 26], "alzheim": 22, "depressionbarbu": [22, 26], "depress": 22, "2015": 22, "id": [23, 25, 32, 40, 41, 42, 43], "present": 23, "fac": 23, "optim": [23, 52], "illuminaep": 23, "ye": 23, "gse110554": 23, "flowsort": 23, "epic": 23, "ba": 23, "genom": [23, 33], "wide": 23, "profil": 23, "reveal": 23, "quantit": 23, "view": 23, "o": 23, "illumina450k": 23, "656": 23, "gse19711": 23, "unit": 23, "kingdom": 23, "ovar": 23, "illumina27k": 23, "540": 23, "No": [23, 25, 39], "methylom": 23, "analysi": [23, 25], "me": 23, "329": 23, "gse42861": 23, "differenti": 23, "rheumatoid": 23, "arthriti": 23, "689": 23, "dutch": 23, "popul": 23, "95": 23, "gse51032": 23, "itali": 23, "hugef": 23, "845": 23, "gse73103": 23, "mani": [23, 24, 51, 52], "obes": 23, "snp": 23, "strongli": 23, "met": 23, "355": 23, "gse69270": 23, "middl": 23, "indi": 23, "184": 23, "gse36054": 23, "children": 23, "192": 23, "gse64495": 23, "sever": [23, 24], "113": 23, "gse30870": 23, "newborn": 23, "nonagenarian": 23, "40": 23, "identif": 23, "signatur": [23, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45], "87": 23, "gse157131": 23, "peripher": 23, "leukocyt": 23, "fro": 23, "946": 23, "gse132203": 23, "gradi": 23, "trauma": 23, "project": 23, "795": 23, "rnaseq": 23, "whole": 23, "500fg": 23, "cohort": 23, "illuminahiseq2500": 23, "100": 23, "jenag": 23, "deep": [23, 33], "sequenc": 23, "endogen": 23, "mrna": 23, "homo": 23, "sapien": 23, "fo": 23, "62": 23, "gse167998": 23, "bloodextend": 23, "enhanc": [23, 39], "l": 23, "gse180683": 23, "cd4t": 23, "cd8t": 23, "naiv": 23, "76": 23, "gse180970": 23, "umbil": 23, "artifici": 23, "mixtur": 23, "gse182379": 23, "enha": 23, "enabl": [24, 51], "easi": [24, 51, 52], "versatil": [24, 51], "It": [24, 25, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 51], "provid": [22, 24, 27, 28, 32, 44, 46, 49, 50, 51, 52], "tool": [24, 51, 52], "load": [1, 7, 19, 24, 28, 31, 32, 33, 51, 52, 53], "publicli": [24, 51], "like": [24, 25, 32, 51, 52], "gene": [24, 51], "express": [24, 51], "omnibu": [24, 51], "nation": [24, 38, 51], "health": [24, 38, 51, 52], "nutrit": [24, 38, 51], "examin": [24, 38], "survei": [24, 38, 51], "framingham": [24, 37, 51], "studi": [24, 37, 51], "also": [24, 51, 52], "contain": [24, 25, 32, 33, 34, 36, 40, 41, 42, 43, 44, 45, 51], "implement": [24, 27, 28, 29, 30, 49, 52], "common": [23, 24, 25, 51], "other": [24, 25, 26, 34, 51], "onli": [24, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 51], "few": [24, 51], "read": [5, 24, 51], "about": [24, 32, 37, 38, 51, 52], "our": [24, 51, 52], "develop": [24, 51, 52], "organ": [24, 31, 32, 33], "individu": [23, 24, 37, 38], "especi": 24, "methuselah": 24, "foundat": [24, 26], "volo": 24, "host": 24, "2025": 24, "seri": [24, 25, 34, 36], "platform": 24, "particip": 24, "ask": 24, "morbid": 24, "award": 24, "200k": 24, "pleas": [24, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 51, 52], "cite": 24, "preprint": 24, "m": 24, "mart\u00ed": 24, "nez": 24, "c": [24, 40, 41], "poganik": 24, "j": 24, "r": 24, "gladyshev": 24, "2023": 24, "open": [24, 51, 52], "biorxiv": 24, "doi": 24, "1101": 24, "569722": 24, "quickstart": 24, "start": [24, 51], "roadmap": [24, 52], "see": [24, 52], "head": 24, "contribut": [24, 52], "biolog": [24, 39], "relationship": 24, "variou": [24, 39, 46], "point": 24, "interest": [26, 52], "commerci": 26, "tabl": 26, "help": [26, 52], "do": [25, 26], "who": [], "talk": 52, "page": [29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45], "document": [29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 51], "explain": [29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45], "user": [29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45], "guid": [29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45], "big": [29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 52], "pictur": [29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45], "max_size_gb": 29, "serial": 29, "local": [1, 7, 19, 29, 33, 53], "basic": 29, "lru": 29, "least": 29, "recent": 29, "cleanup": 29, "maintain": [29, 52], "within": [29, 39], "specifi": [29, 31, 33, 39, 45], "maximum": 29, "limit": 29, "__init__": [29, 30, 31, 32, 33, 39], "initi": [29, 31, 32, 33, 39], "instanc": [29, 31, 32, 33, 39], "filesystem": 29, "storag": [25, 29], "gigabyt": 29, "kei": [29, 30, 32, 45], "retriev": [29, 30, 31, 39], "item": [29, 30, 32, 52], "its": [29, 31, 39], "doe": [25, 29, 30], "occur": [23, 29], "ani": [25, 29, 30, 32, 51, 52], "under": [29, 30], "manag": [29, 31, 39], "stai": 29, "clear": [29, 30], "remov": [29, 30], "content": [25, 29], "specif": [23, 25, 29, 30], "entri": [25, 29, 30], "noth": 30, "alwai": 30, "simul": 30, "library_fil": 31, "cach": [31, 32, 52], "collect": [31, 39, 40, 42, 43, 45, 52], "respons": 31, "new": [31, 33], "ad": [25, 31, 52], "runtim": 31, "current": [31, 52], "mechan": [31, 32], "adher": [31, 32], "interfac": [31, 32, 51], "modul": [23, 27, 28, 31, 32, 46, 49, 50], "load_sourc": 31, "given": 31, "append": 31, "source_id": 31, "found": [31, 39], "otherwis": [31, 40, 42, 43], "lookup_sourc": 31, "look": 31, "up": [31, 32, 52], "filter": [31, 33, 39], "list": [31, 33, 34, 35, 36, 45], "match": [25, 31, 39], "criteria": [31, 39], "source_definit": 32, "repres": [25, 32, 33, 37, 38], "singl": [25, 32, 52], "encapsul": 32, "detail": [32, 33, 51], "rais": [32, 34, 39], "valueerror": [32, 34, 39], "requir": [32, 34, 52], "field": [32, 33], "parser": 32, "dure": 32, "pars": [32, 52], "dictionari": [32, 39, 45], "typic": 32, "yaml": 32, "check": [32, 51], "essenti": 32, "attribut": 32, "handl": [25, 32, 39], "dict": [32, 39, 45], "properti": 32, "must": [25, 32, 40, 42, 43], "summari": [5, 32, 33], "nocach": 32, "input": [32, 38, 39], "required_field": 32, "geograph": 32, "rtype": 32, "rna": 33, "protein": 33, "save_to_hdf5": [], "filepath": [], "complevel": [], "complib": [], "zlib": [], "hdf5": [], "compress": [], "save_to_csv": [], "base_filepath": [], "separ": [25, 52], "load_from_hdf5": [], "quality_report": [5, 33], "qualiti": [5, 25, 33], "control": [25, 33], "report": [5, 33], "section": 33, "per": 33, "qualityreport": 33, "classmethod": 33, "matrix": 33, "either": 33, "directli": [33, 52], "union": 33, "cpg_sourc": [34, 36], "required_cpg": 34, "extern": [34, 36], "while": [25, 34], "need": 52, "final": [], "strategi": [34, 39, 46], "fill": [34, 35, 36], "certain": 34, "cpgs_to_imput": [35, 36], "itself": 35, "cpg_averag": [], "measur": [25, 37, 38], "Not": 38, "come": 38, "two": [25, 38], "pass": 38, "later": 38, "2010": 38, "2012": 38, "model_definit": 39, "produc": 39, "search": 39, "definit": 39, "model_build": 39, "deconvolutionmodel": 39, "bound": 39, "from_definit": 39, "grimagemodel": 39, "linearmethylationmodel": 39, "linearmodel": 39, "lineartranscriptomicmodel": 39, "sexestimationmodel": 39, "request": [39, 51], "possibli": 39, "keyerror": 39, "invalid": 39, "appli": 39, "dunedin": 39, "hybrid_imput": 39, "gold": [5, 39], "sesame_450k": 39, "sesam": [25, 39], "450k": 39, "These": 46, "allow": 46, "hybrid": 46, "examim": 51, "implemen": 51, "prereleas": 51, "version": 51, "There": 51, "mai": [5, 51], "bug": 51, "subject": [40, 41, 42, 43, 51], "github": [5, 51], "com": 51, "bio": [5, 51], "homepag": 51, "io": [5, 51], "pip": 51, "wa": 51, "correctli": [5, 51], "out": [51, 52], "some": [51, 52], "team": 51, "ha": [25, 51, 52], "answer": 51, "question": 51, "discuss": 51, "find": 51, "replic": 51, "setup": 51, "repo": 51, "omic": 52, "revolut": 52, "transform": 52, "biologi": 52, "wealth": 52, "potenti": 52, "care": 52, "could": 52, "offer": 52, "comprehens": 52, "insight": 52, "one": 52, "surrog": 52, "endpoint": 52, "revers": 52, "therapeut": 52, "howev": [25, 52], "obstruct": 52, "access": [33, 52], "address": 52, "issu": 52, "modern": 52, "recogn": 52, "ve": 52, "adopt": 52, "scienc": 52, "invit": 52, "mission": 52, "primarili": 52, "25": [25, 52], "compris": 52, "thousand": 52, "ambit": 52, "achiev": 52, "larg": 52, "small": 52, "task": 52, "vari": 52, "skill": 52, "level": [23, 25, 52], "knowledg": 52, "right": 52, "improv": 52, "better": 52, "differ": [23, 33, 49, 52], "environ": 52, "speed": 52, "10x": 52, "integr": 52, "unifi": 52, "structur": 52, "know": 52, "process": [25, 52], "partial": 52, "autom": 52, "review": 52, "genet": 52, "built": 52, "share": 52, "reach": 52, "proprietari": 52, "would": 52, "api": 52, "public": 52, "hear": 52, "privat": 52, "think": 52, "wai": 52, "fund": 52, "financi": 52, "monei": 52, "mainten": 52, "prize": 52, "pot": 52, "further": 52, "expans": 52, "contriubt": 52, "discord": 52, "server": 52, "38": 23, "388": [], "contact": [26, 40, 42, 43], "author": 26, "free": 26, "For": 26, "cosmet": 26, "life": 26, "insur": 26, "applic": 26, "ucla": 26, "tdg": 26, "regard": 26, "licens": 26, "focu": 33, "facilit": 33, "describ": [5, 25], "consist": 25, "focus": 25, "transmiss": 25, "without": [25, 41], "mandat": 25, "form": 25, "veri": 25, "commonli": 25, "thei": 25, "problem": 25, "inclus": 25, "even": 25, "As": 25, "grown": 25, "larger": 25, "longer": 25, "instead": 25, "supplementari": 25, "been": 25, "pre": 25, "batch": 25, "normal": 25, "recommend": 25, "illumina": [23, 25], "comma": 25, "same": 25, "correspond": 25, "first": 25, "header": 25, "publish": 25, "exactli": 25, "1000": 25, "beta": 25, "decim": 25, "precis": 25, "round": 25, "three": 25, "place": 25, "integ": 25, "femal": 25, "male": 25, "permit": 25, "unknown": 25, "gsm3074480": [5, 25], "gsm3074481": [5, 25], "gsm3074482": [5, 25], "gsm3074483": [5, 25], "395": [5, 25], "499": [5, 25], "423": [5, 25], "821": [5, 25], "868": [5, 25], "877": [5, 25], "841": [5, 25], "878": [5, 25], "894": [5, 25], "822": 25, "state": 25, "covid": 25, "encod": 25, "soft": 25, "ncbi": 25, "tip": 25, "real": [], "mgb": [], "ftp": [], "nlm": [], "nih": [], "gov": [], "gse246nnn": [], "gse246337": [], "suppl": [], "5fbeta": [], "2ecsv": [], "2egz": [], "cpgsite": [5, 25], "ip": 22, "inform": 22, "predictor": [40, 41, 42, 43, 44], "predictor_result": [40, 41, 42, 43], "dead": [40, 41, 42, 43], "boolean": [40, 41, 42], "indic": [40, 41, 42], "death": [40, 41, 42, 43], "last": [40, 42, 43], "live": [40, 42, 43], "until": 41, "censor": 41, "log": 42, "rank": 42, "statist": 42, "p": [42, 43, 44], "hazard": [43, 44], "ratio": [43, 44], "cox": 43, "aliv": 43, "confid": [43, 44], "interv": [43, 44], "hazard_ratio_data": 44, "forest": 44, "predictors_dict": 45, "output": 45, "ex": 45, "get_test_data_fil": 5, "html": 5, "file_path": 5, "example_data": 5, "420": 5, "890": 5, "With": 5, "high": 5, "deviat": 5, "over": 5, "303": [], "plot_local_data_load": [5, 7, 53], "421": [5, 7, 53], "gse106648": 23, "sclerosi": 23, "279": 23, "gse120307": 23, "34": 23, "gse134429": 23, "inflammatori": 23, "cytokin": 23, "mon": 23, "64": 23, "gse151278": 23, "sam": 23, "70": 23, "gse164056": 23, "social": 23, "anxieti": 23, "143": 23, "gse166611": 23, "trait": 23, "gse20236": 23, "hypermethyl": 23, "prefer": 23, "93": 23, "gse32148": 23, "cr": 23, "48": 23, "gse41037": 23, "effect": 23, "720": 23, "gse49904": 23, "71": 23, "gse50660": 23, "cigarett": 23, "reduc": 23, "464": 23, "gse53740": 23, "384": 23, "gse53840": 23, "hiv": 23, "posit": 23, "men": 23, "120": 23, "gse67705": 23, "chronic": 23, "infect": 23, "patient": 23, "284": 23, "gse67751": 23, "inf": 23, "450": 23, "gse77445": 23, "alter": 23, "cortisol": 23, "stre": 23, "85": 23, "gse85311": 23, "pattern": 23, "exercis": 23, "gse96879": 23, "epigenom": 23, "sustem": 23, "lupu": 23, "erythem": 23, "90": 23, "gse99624": 23, "primari": 23, "osteoporosi": 23, "reflect": 23, "dn": 23}, "objects": {"biolearn": [[27, 0, 0, "-", "cache"], [28, 0, 0, "-", "data_library"], [46, 0, 0, "-", "imputation"], [48, 0, 0, "-", "load"], [49, 0, 0, "-", "model_gallery"], [50, 0, 0, "-", "mortality"]], "biolearn.cache": [[29, 1, 1, "", "LocalFolderCache"], [30, 1, 1, "", "NoCache"]], "biolearn.cache.LocalFolderCache": [[29, 2, 1, "", "__init__"], [29, 2, 1, "", "clear"], [29, 2, 1, "", "get"], [29, 2, 1, "", "remove"], [29, 2, 1, "", "store"]], "biolearn.cache.NoCache": [[30, 2, 1, "", "__init__"], [30, 2, 1, "", "clear"], [30, 2, 1, "", "get"], [30, 2, 1, "", "remove"], [30, 2, 1, "", "store"]], "biolearn.data_library": [[31, 1, 1, "", "DataLibrary"], [32, 1, 1, "", "DataSource"], [33, 1, 1, "", "GeoData"]], "biolearn.data_library.DataLibrary": [[31, 2, 1, "", "__init__"], [31, 2, 1, "", "get"], [31, 2, 1, "", "load_sources"], [31, 2, 1, "", "lookup_sources"]], "biolearn.data_library.DataSource": [[32, 3, 1, "", "REQUIRED_FIELDS"], [32, 2, 1, "", "__init__"], [32, 2, 1, "", "load"]], "biolearn.data_library.GeoData": [[33, 2, 1, "", "__init__"], [33, 2, 1, "", "copy"], [33, 3, 1, "", "dnam"], [33, 2, 1, "", "from_methylation_matrix"], [33, 3, 1, "", "metadata"], [33, 2, 1, "", "quality_report"]], "biolearn.imputation": [[34, 4, 1, "", "hybrid_impute"], [35, 4, 1, "", "impute_from_average"], [36, 4, 1, "", "impute_from_standard"]], "biolearn.load": [[37, 4, 1, "", "load_fhs"], [38, 4, 1, "", "load_nhanes"]], "biolearn.model_gallery": [[39, 1, 1, "", "ModelGallery"]], "biolearn.model_gallery.ModelGallery": [[39, 2, 1, "", "__init__"], [39, 2, 1, "", "get"], [39, 3, 1, "", "model_builders"], [39, 2, 1, "", "search"]], "biolearn.mortality": [[40, 4, 1, "", "calculate_age_adjusted_c_index"], [41, 4, 1, "", "calculate_c_index"], [42, 4, 1, "", "calculate_log_rank_test"], [43, 4, 1, "", "calculate_mortality_hazard_ratios"], [44, 4, 1, "", "plot_hazard_ratios"], [45, 4, 1, "", "run_predictions"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:attribute", "4": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "attribute", "Python attribute"], "4": ["py", "function", "Python function"]}, "titleterms": {"team": 0, "core": 0, "develop": 0, "other": [0, 5, 22], "contributor": 0, "partner": 0, "organ": 0, "omic": [1, 19], "biomark": [1, 8, 19, 52], "exampl": [1, 8, 11, 16, 17, 19, 24, 25, 31, 32, 33, 36, 38, 39, 45], "perform": 2, "custom": 2, "imput": [2, 34, 35, 36, 46], "load": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 37, 38, 48], "methyl": [2, 3, 4, 12, 17, 25], "data": [2, 3, 4, 5, 6, 9, 12, 14, 15, 23, 25, 46, 48], "from": [2, 3, 4, 6, 12], "geo": [2, 3, 4, 6, 17, 23], "dataset": [2, 4, 5, 17], "us": [2, 3, 4, 5, 6, 14, 15, 17, 31, 32, 33, 36, 38, 39, 45], "refer": [2, 12, 25, 47], "up": [2, 3, 4, 5, 6, 12, 14, 15], "target": 2, "run": [2, 3, 4, 6, 12, 15], "now": [2, 3, 4, 5, 6], "clock": [2, 4, 6, 22], "predict": [2, 3, 6, 14], "befor": 2, "after": 2, "visual": 2, "comparison": 2, "ag": [2, 4, 6, 9, 14], "you": [2, 5, 12], "can": [2, 5, 12], "also": [2, 12], "build": [2, 14], "an": [2, 14, 15], "decor": 2, "bundl": 2, "get": [2, 3], "same": 2, "result": [2, 4, 6, 15], "function": [2, 46], "directli": 2, "your": 2, "down": 3, "syndrom": 3, "epigenet": [3, 4], "plot": [3, 4, 6, 9, 15], "first": [3, 4, 6], "some": [3, 4, 6, 12, 15], "librari": [3, 4, 6], "model": [3, 12, 14, 15, 17, 22, 26], "score": 3, "final": [3, 4], "gener": 3, "boxplot": 3, "show": [3, 9], "power": 3, "three": 4, "differ": 4, "produc": 4, "extract": 4, "metadata": [4, 12, 25], "seaborn": [4, 6], "transcriptom": 6, "against": 6, "chronolog": [6, 9], "comput": [7, 10, 13, 18, 20, 53], "time": [7, 10, 13, 18, 20, 53], "composit": [8, 19], "phenotyp": 9, "nhane": 9, "2010": 9, "calcul": 9, "biolog": 9, "base": 9, "phenotypicag": 9, "relat": 9, "between": 9, "surviv": 9, "curv": 9, "peopl": 9, "acceler": 9, "older": 9, "v": [9, 25], "deceler": 9, "younger": 9, "challeng": [11, 12, 15, 19], "submiss": [11, 14, 15, 19], "explor": 12, "competit": [12, 14, 15], "The": 12, "ha": 12, "proteom": 12, "learn": 12, "more": 12, "about": 12, "what": 12, "protein": 12, "identifi": 12, "our": [12, 15], "overlap": 12, "while": 12, "doe": 12, "all": 12, "i": [12, 15, 25], "combin": 12, "easili": 12, "sever": 12, "them": 12, "we": 12, "compar": [12, 17], "output": [12, 15], "two": 12, "exist": 14, "lin": 14, "train": 15, "elasticnet": 15, "seper": 15, "test": 15, "set": 15, "elast": 15, "net": 15, "see": 15, "how": 15, "good": 15, "through": 15, "save": 15, "file": [5, 15, 25], "deconvolut": [16, 17, 19], "blood": 17, "measur": 17, "epic": 17, "platform": 17, "estim": 17, "cell": 17, "proport": 17, "known": 17, "fac": 17, "fluoresc": 17, "activ": 17, "sort": 17, "sourc": 23, "biolearn": [5, 24, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 48, 49, 50, 51, 52], "featur": 24, "ip": 26, "usag": 26, "cach": [27, 29, 30], "data_librari": [28, 31, 32, 33], "datalibrari": [28, 31], "localfoldercach": 29, "nocach": 30, "datasourc": 32, "geodata": 33, "hybrid_imput": 34, "impute_from_averag": 35, "impute_from_standard": 36, "load_fh": 37, "load_nhan": 38, "model_galleri": [39, 49], "modelgalleri": [39, 49], "util": 48, "api": 47, "quickstart": 51, "import": 51, "link": 51, "requir": 51, "instal": 51, "discord": 51, "server": 51, "issu": 51, "contribut": 51, "futur": 52, "join": 52, "u": 52, "dna": 25, "arrai": 25, "standard": 25, "2410": 25, "why": 25, "thi": 25, "need": 25, "preprocess": 25, "matrix": 25, "format": 25, "valu": 25, "field": 25, "addit": 25, "dnam": [], "mortal": [40, 41, 42, 43, 44, 45, 50], "calculate_age_adjusted_c_index": 40, "calculate_c_index": 41, "calculate_log_rank_test": 42, "calculate_mortality_hazard_ratio": 43, "plot_hazard_ratio": 44, "run_predict": 45, "pridictor": [], "evalu": 50, "local": 5, "like": 5, "ani": 5, "predictor": 50}, "envversion": {"sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinxcontrib.bibtex": 9, "sphinx": 60}, "alltitles": {"Team": [[0, "team"]], "Core developers": [[0, "core-developers"]], "Other contributors": [[0, "other-contributors"]], "Partner Organizations": [[0, "partner-organizations"]], "Performing custom imputations": [[2, "performing-custom-imputations"]], "Load methylation data from a GEO dataset to use as a reference for imputation": [[2, "load-methylation-data-from-a-geo-dataset-to-use-as-a-reference-for-imputation"]], "Load up a target dataset and run the imputation": [[2, "load-up-a-target-dataset-and-run-the-imputation"]], "Now run clock predictions on the dataset before and after": [[2, "now-run-clock-predictions-on-the-dataset-before-and-after"]], "Visualize the comparison of age predictions": [[2, "visualize-the-comparison-of-age-predictions"]], "You can also build an imputation decorator to bundle with the clock": [[2, "you-can-also-build-an-imputation-decorator-to-bundle-with-the-clock"]], "You get the same results as running the function directly on your dataset": [[2, "you-get-the-same-results-as-running-the-function-directly-on-your-dataset"]], "Down Syndrome Epigenetic Plotting": [[3, "down-syndrome-epigenetic-plotting"]], "First load up some methylation data from GEO using the data library": [[3, "first-load-up-some-methylation-data-from-geo-using-the-data-library"], [4, "first-load-up-some-methylation-data-from-geo-using-the-data-library"]], "Now run the down syndrome model on the data to get a score": [[3, "now-run-the-down-syndrome-model-on-the-data-to-get-a-score"]], "Finally generate a boxplot to show the predictive power": [[3, "finally-generate-a-boxplot-to-show-the-predictive-power"]], "\u201cEpigenetic Clocks\u201d in GEO Data": [[4, "epigenetic-clocks-in-geo-data"]], "Now run three different clocks on the dataset to produce epigenetic clock ages": [[4, "now-run-three-different-clocks-on-the-dataset-to-produce-epigenetic-clock-ages"]], "Finally extract the age data from the metadata from GEO and plot the results using seaborn": [[4, "finally-extract-the-age-data-from-the-metadata-from-geo-and-plot-the-results-using-seaborn"]], "\u201cTranscriptomic Clock\u201d using GEO Data": [[6, "transcriptomic-clock-using-geo-data"]], "First load up some transcriptomic data from GEO using the data library": [[6, "first-load-up-some-transcriptomic-data-from-geo-using-the-data-library"]], "Now run the transcriptomic clock to predict age": [[6, "now-run-the-transcriptomic-clock-to-predict-age"]], "Plot the results against the chronological age using seaborn": [[6, "plot-the-results-against-the-chronological-age-using-seaborn"]], "Composite Biomarkers Examples": [[8, "composite-biomarkers-examples"], [19, "composite-biomarkers-examples"]], "\u201cPhenotypic Ages\u201d in NHANES Data": [[9, "phenotypic-ages-in-nhanes-data"]], "Loading NHANES 2010 data": [[9, "loading-nhanes-2010-data"]], "Calculate \u201cbiological age\u201d based on PhenotypicAge": [[9, "calculate-biological-age-based-on-phenotypicage"]], "Show relation between biological age and chronological age": [[9, "show-relation-between-biological-age-and-chronological-age"]], "Plot survival curve for people with accelerated aging (older biological age) vs decelerated aging (younger biological age)": [[9, "plot-survival-curve-for-people-with-accelerated-aging-older-biological-age-vs-decelerated-aging-younger-biological-age"]], "Computation times": [[10, "computation-times"], [13, "computation-times"], [18, "computation-times"], [20, "computation-times"], [7, "computation-times"], [53, "computation-times"]], "Challenge Submission Examples": [[11, "challenge-submission-examples"], [19, "challenge-submission-examples"]], "Exploring the Challenge Data": [[12, "exploring-the-challenge-data"]], "Loading up the data for the competition": [[12, "loading-up-the-data-for-the-competition"], [14, "loading-up-the-data-for-the-competition"], [15, "loading-up-the-data-for-the-competition"]], "The challenge data has methylation data": [[12, "the-challenge-data-has-methylation-data"]], "The challenge data also has proteomic data": [[12, "the-challenge-data-also-has-proteomic-data"]], "You can learn more about what the protein identifies in our reference": [[12, "you-can-learn-more-about-what-the-protein-identifies-in-our-reference"]], "Some of the data overlaps while some does not but all the metadata is combined": [[12, "some-of-the-data-overlaps-while-some-does-not-but-all-the-metadata-is-combined"]], "You can easily run several models on them": [[12, "you-can-easily-run-several-models-on-them"]], "We can then compare the output from the two models": [[12, "we-can-then-compare-the-output-from-the-two-models"]], "Deconvolution Examples": [[16, "deconvolution-examples"], [19, "deconvolution-examples"]], "\u201cDeconvolution Example\u201d": [[17, "deconvolution-example"]], "Load GEO blood methylation dataset measured on EPIC platform": [[17, "load-geo-blood-methylation-dataset-measured-on-epic-platform"]], "Estimate cell proportions using EPIC deconvolution model": [[17, "estimate-cell-proportions-using-epic-deconvolution-model"]], "Compare to known cell proportions measured with FACS (fluorescence-activated cell sorting)": [[17, "compare-to-known-cell-proportions-measured-with-facs-fluorescence-activated-cell-sorting"]], "biolearn.cache.LocalFolderCache": [[29, "biolearn-cache-localfoldercache"]], "biolearn.cache.NoCache": [[30, "biolearn-cache-nocache"]], "biolearn.load.load_fhs": [[37, "biolearn-load-load-fhs"]], "biolearn.load.load_nhanes": [[38, "biolearn-load-load-nhanes"]], "Examples using biolearn.load.load_nhanes": [[38, "examples-using-biolearn-load-load-nhanes"]], "biolearn.load: Data Loading Utilities": [[48, "module-biolearn.load"]], "Quickstart": [[51, "quickstart"]], "Biolearn": [[51, "biolearn"], [24, "biolearn"]], "Important links": [[51, "important-links"]], "Requirements": [[51, "requirements"]], "Install": [[51, "install"]], "Discord server": [[51, "discord-server"]], "Issues": [[51, "issues"]], "Contributing": [[51, "contributing"]], "Biolearn and the Future of Biomarkers": [[52, "biolearn-and-the-future-of-biomarkers"]], "Join Us": [[52, "join-us"]], "biolearn.model_gallery.ModelGallery": [[39, "biolearn-model-gallery-modelgallery"]], "Examples using biolearn.model_gallery.ModelGallery": [[39, "examples-using-biolearn-model-gallery-modelgallery"]], "DNA Methylation Array Data Standard V-2410": [[25, "dna-methylation-array-data-standard-v-2410"]], "Why is this standard needed?": [[25, "why-is-this-standard-needed"]], "Preprocessing": [[25, "preprocessing"]], "Methylation Matrix File Standard": [[25, "methylation-matrix-file-standard"]], "File Format": [[25, "file-format"], [25, "id1"]], "Data Values": [[25, "data-values"]], "Metadata File Standard": [[25, "metadata-file-standard"]], "Standard Metadata Fields": [[25, "standard-metadata-fields"]], "Additional Metadata Fields": [[25, "additional-metadata-fields"]], "Example Files": [[25, "example-files"]], "References": [[25, "references"]], "Omics Biomarker Examples": [[1, "omics-biomarker-examples"], [19, "omics-biomarker-examples"]], "Local Data Loading": [[5, "local-data-loading"]], "Load up a local data file": [[5, "load-up-a-local-data-file"]], "You can now use it like any other Biolearn dataset": [[5, "you-can-now-use-it-like-any-other-biolearn-dataset"]], "Building a competition submission using an existing model": [[14, "building-a-competition-submission-using-an-existing-model"]], "Use the Lin model to predict the age": [[14, "use-the-lin-model-to-predict-the-age"]], "Training an ElasticNet model": [[15, "training-an-elasticnet-model"]], "Load up some training data": [[15, "load-up-some-training-data"]], "Seperate data into training and test sets": [[15, "seperate-data-into-training-and-test-sets"]], "Train a model using elastic net": [[15, "train-a-model-using-elastic-net"]], "Plot the results to see how good our model is": [[15, "plot-the-results-to-see-how-good-our-model-is"]], "Run the challenge data through the model": [[15, "run-the-challenge-data-through-the-model"]], "Save the results as an output file for submission": [[15, "save-the-results-as-an-output-file-for-submission"]], "Examples": [[19, "examples"]], "Featured examples": [[24, "featured-examples"]], "biolearn.cache: Cache": [[27, "module-biolearn.cache"]], "biolearn.data_library: DataLibrary": [[28, "module-biolearn.data_library"]], "biolearn.data_library.DataLibrary": [[31, "biolearn-data-library-datalibrary"]], "Examples using biolearn.data_library.DataLibrary": [[31, "examples-using-biolearn-data-library-datalibrary"]], "biolearn.data_library.DataSource": [[32, "biolearn-data-library-datasource"]], "Examples using biolearn.data_library.DataSource": [[32, "examples-using-biolearn-data-library-datasource"]], "biolearn.data_library.GeoData": [[33, "biolearn-data-library-geodata"]], "Examples using biolearn.data_library.GeoData": [[33, "examples-using-biolearn-data-library-geodata"]], "biolearn.imputation.hybrid_impute": [[34, "biolearn-imputation-hybrid-impute"]], "biolearn.imputation.impute_from_average": [[35, "biolearn-imputation-impute-from-average"]], "biolearn.imputation.impute_from_standard": [[36, "biolearn-imputation-impute-from-standard"]], "Examples using biolearn.imputation.impute_from_standard": [[36, "examples-using-biolearn-imputation-impute-from-standard"]], "biolearn.mortality.calculate_age_adjusted_c_index": [[40, "biolearn-mortality-calculate-age-adjusted-c-index"]], "biolearn.mortality.calculate_c_index": [[41, "biolearn-mortality-calculate-c-index"]], "biolearn.mortality.calculate_log_rank_test": [[42, "biolearn-mortality-calculate-log-rank-test"]], "biolearn.mortality.calculate_mortality_hazard_ratios": [[43, "biolearn-mortality-calculate-mortality-hazard-ratios"]], "biolearn.mortality.plot_hazard_ratios": [[44, "biolearn-mortality-plot-hazard-ratios"]], "biolearn.mortality.run_predictions": [[45, "biolearn-mortality-run-predictions"]], "Examples using biolearn.mortality.run_predictions": [[45, "examples-using-biolearn-mortality-run-predictions"]], "biolearn.imputation: Data Imputation Functions": [[46, "biolearn-imputation-data-imputation-functions"]], "API References": [[47, "api-references"]], "biolearn.model_gallery: ModelGallery": [[49, "module-biolearn.model_gallery"]], "biolearn.mortality: Mortality Predictor Evaluation": [[50, "biolearn-mortality-mortality-predictor-evaluation"]], "Clocks and Other Models": [[22, "clocks-and-other-models"]], "GEO Data Sources": [[23, "geo-data-sources"]], "Model IP and Usage": [[26, "model-ip-and-usage"]]}, "indexentries": {"biolearn.cache": [[27, "module-biolearn.cache"]], "module": [[27, "module-biolearn.cache"], [28, "module-biolearn.data_library"], [46, "module-biolearn.imputation"], [48, "module-biolearn.load"], [49, "module-biolearn.model_gallery"], [50, "module-biolearn.mortality"]], "biolearn.data_library": [[28, "module-biolearn.data_library"]], "localfoldercache (class in biolearn.cache)": [[29, "biolearn.cache.LocalFolderCache"]], "__init__() (biolearn.cache.localfoldercache method)": [[29, "biolearn.cache.LocalFolderCache.__init__"]], "clear() (biolearn.cache.localfoldercache method)": [[29, "biolearn.cache.LocalFolderCache.clear"]], "get() (biolearn.cache.localfoldercache method)": [[29, "biolearn.cache.LocalFolderCache.get"]], "remove() (biolearn.cache.localfoldercache method)": [[29, "biolearn.cache.LocalFolderCache.remove"]], "store() (biolearn.cache.localfoldercache method)": [[29, "biolearn.cache.LocalFolderCache.store"]], "nocache (class in biolearn.cache)": [[30, "biolearn.cache.NoCache"]], "__init__() (biolearn.cache.nocache method)": [[30, "biolearn.cache.NoCache.__init__"]], "clear() (biolearn.cache.nocache method)": [[30, "biolearn.cache.NoCache.clear"]], "get() (biolearn.cache.nocache method)": [[30, "biolearn.cache.NoCache.get"]], "remove() (biolearn.cache.nocache method)": [[30, "biolearn.cache.NoCache.remove"]], "store() (biolearn.cache.nocache method)": [[30, "biolearn.cache.NoCache.store"]], "datalibrary (class in biolearn.data_library)": [[31, "biolearn.data_library.DataLibrary"]], "__init__() (biolearn.data_library.datalibrary method)": [[31, "biolearn.data_library.DataLibrary.__init__"]], "get() (biolearn.data_library.datalibrary method)": [[31, "biolearn.data_library.DataLibrary.get"]], "load_sources() (biolearn.data_library.datalibrary method)": [[31, "biolearn.data_library.DataLibrary.load_sources"]], "lookup_sources() (biolearn.data_library.datalibrary method)": [[31, "biolearn.data_library.DataLibrary.lookup_sources"]], "datasource (class in biolearn.data_library)": [[32, "biolearn.data_library.DataSource"]], "required_fields (biolearn.data_library.datasource attribute)": [[32, "biolearn.data_library.DataSource.REQUIRED_FIELDS"]], "__init__() (biolearn.data_library.datasource method)": [[32, "biolearn.data_library.DataSource.__init__"]], "load() (biolearn.data_library.datasource method)": [[32, "biolearn.data_library.DataSource.load"]], "geodata (class in biolearn.data_library)": [[33, "biolearn.data_library.GeoData"]], "__init__() (biolearn.data_library.geodata method)": [[33, "biolearn.data_library.GeoData.__init__"]], "copy() (biolearn.data_library.geodata method)": [[33, "biolearn.data_library.GeoData.copy"]], "dnam (biolearn.data_library.geodata attribute)": [[33, "biolearn.data_library.GeoData.dnam"]], "from_methylation_matrix() (biolearn.data_library.geodata class method)": [[33, "biolearn.data_library.GeoData.from_methylation_matrix"]], "metadata (biolearn.data_library.geodata attribute)": [[33, "biolearn.data_library.GeoData.metadata"]], "quality_report() (biolearn.data_library.geodata method)": [[33, "biolearn.data_library.GeoData.quality_report"]], "hybrid_impute() (in module biolearn.imputation)": [[34, "biolearn.imputation.hybrid_impute"]], "impute_from_average() (in module biolearn.imputation)": [[35, "biolearn.imputation.impute_from_average"]], "impute_from_standard() (in module biolearn.imputation)": [[36, "biolearn.imputation.impute_from_standard"]], "load_fhs() (in module biolearn.load)": [[37, "biolearn.load.load_fhs"]], "load_nhanes() (in module biolearn.load)": [[38, "biolearn.load.load_nhanes"]], "modelgallery (class in biolearn.model_gallery)": [[39, "biolearn.model_gallery.ModelGallery"]], "__init__() (biolearn.model_gallery.modelgallery method)": [[39, "biolearn.model_gallery.ModelGallery.__init__"]], "get() (biolearn.model_gallery.modelgallery method)": [[39, "biolearn.model_gallery.ModelGallery.get"]], "model_builders (biolearn.model_gallery.modelgallery attribute)": [[39, "biolearn.model_gallery.ModelGallery.model_builders"]], "search() (biolearn.model_gallery.modelgallery method)": [[39, "biolearn.model_gallery.ModelGallery.search"]], "calculate_age_adjusted_c_index() (in module biolearn.mortality)": [[40, "biolearn.mortality.calculate_age_adjusted_c_index"]], "calculate_c_index() (in module biolearn.mortality)": [[41, "biolearn.mortality.calculate_c_index"]], "calculate_log_rank_test() (in module biolearn.mortality)": [[42, "biolearn.mortality.calculate_log_rank_test"]], "calculate_mortality_hazard_ratios() (in module biolearn.mortality)": [[43, "biolearn.mortality.calculate_mortality_hazard_ratios"]], "plot_hazard_ratios() (in module biolearn.mortality)": [[44, "biolearn.mortality.plot_hazard_ratios"]], "run_predictions() (in module biolearn.mortality)": [[45, "biolearn.mortality.run_predictions"]], "biolearn.imputation": [[46, "module-biolearn.imputation"]], "biolearn.load": [[48, "module-biolearn.load"]], "biolearn.model_gallery": [[49, "module-biolearn.model_gallery"]], "biolearn.mortality": [[50, "module-biolearn.mortality"]]}})