Search.setIndex({"docnames": ["authors", "auto_examples/00_epigenetic_biomarkers/index", "auto_examples/00_epigenetic_biomarkers/plot_custom_imputation_implementation", "auto_examples/00_epigenetic_biomarkers/plot_epigenetic_clocks_on_geo", "auto_examples/00_epigenetic_biomarkers/sg_execution_times", "auto_examples/01_composite_biomarkers/index", "auto_examples/01_composite_biomarkers/plot_nhanes", "auto_examples/01_composite_biomarkers/sg_execution_times", "auto_examples/index", "changes/names", "clocks", "data", "index", "modules/data_library", "modules/generated/biolearn.data_library.DataLibrary", "modules/generated/biolearn.data_library.DataSource", "modules/generated/biolearn.data_library.GeoData", "modules/generated/biolearn.imputation.biolearn_impute", "modules/generated/biolearn.imputation.hybrid_impute", "modules/generated/biolearn.imputation.impute_from_average", "modules/generated/biolearn.imputation.impute_from_standard", "modules/generated/biolearn.load.load_fhs", "modules/generated/biolearn.load.load_nhanes", "modules/generated/biolearn.model_gallery.ModelGallery", "modules/imputation", "modules/index", "modules/load", "modules/model_gallery", "quickstart"], "filenames": ["authors.rst", "auto_examples/00_epigenetic_biomarkers/index.rst", "auto_examples/00_epigenetic_biomarkers/plot_custom_imputation_implementation.rst", "auto_examples/00_epigenetic_biomarkers/plot_epigenetic_clocks_on_geo.rst", "auto_examples/00_epigenetic_biomarkers/sg_execution_times.rst", "auto_examples/01_composite_biomarkers/index.rst", "auto_examples/01_composite_biomarkers/plot_nhanes.rst", "auto_examples/01_composite_biomarkers/sg_execution_times.rst", "auto_examples/index.rst", "changes/names.rst", "clocks.rst", "data.rst", "index.rst", "modules/data_library.rst", "modules/generated/biolearn.data_library.DataLibrary.rst", "modules/generated/biolearn.data_library.DataSource.rst", "modules/generated/biolearn.data_library.GeoData.rst", "modules/generated/biolearn.imputation.biolearn_impute.rst", "modules/generated/biolearn.imputation.hybrid_impute.rst", "modules/generated/biolearn.imputation.impute_from_average.rst", "modules/generated/biolearn.imputation.impute_from_standard.rst", "modules/generated/biolearn.load.load_fhs.rst", "modules/generated/biolearn.load.load_nhanes.rst", "modules/generated/biolearn.model_gallery.ModelGallery.rst", "modules/imputation.rst", "modules/index.rst", "modules/load.rst", "modules/model_gallery.rst", "quickstart.md"], "titles": ["Team", "Epigenetic Clock Examples", "Performing custom imputations", "\u201cEpigenetic Clocks\u201d in GEO Data", "Computation times", "Composite Biomarkers Examples", "\u201cPhenotypic Ages\u201d in NHANES Data", "Computation times", "Examples", "&lt;no title&gt;", "Clocks and Other Models", "GEO Data Sources", "Biolearn", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.data_library</span></code>: DataLibrary", "biolearn.data_library.DataLibrary", "biolearn.data_library.DataSource", "biolearn.data_library.GeoData", "biolearn.imputation.biolearn_impute", "biolearn.imputation.hybrid_impute", "biolearn.imputation.impute_from_average", "biolearn.imputation.impute_from_standard", "biolearn.load.load_fhs", "biolearn.load.load_nhanes", "biolearn.model_gallery.ModelGallery", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.imputation</span></code>: Data Imputation Utilities", "API References", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.load</span></code>: Data Loading Utilities", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.model_gallery</span></code>: ModelGallery", "Quickstart"], "terms": {"thi": [0, 2, 3, 6, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 28], "work": 0, "i": [0, 12, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 25, 28], "made": 0, "avail": [0, 12, 28], "commun": 0, "peopl": 0, "which": [0, 22], "origin": [0, 2], "from": [0, 5, 6, 8, 10, 11, 12, 14, 15, 18, 19, 20, 21, 22, 24, 28], "biomark": [0, 12, 14, 28], "ag": [0, 5, 7, 8, 10, 11, 12, 22, 28], "consortium": [0, 12], "The": [0, 3, 14, 23, 28], "biolearn": [0, 2, 3, 6, 10, 11, 25], "ar": [0, 2, 14, 15, 18, 22, 23, 28], "mahdi": 0, "moqri": 0, "research": [0, 14], "boa": 0, "execut": [0, 4, 7, 8, 28], "martin": 0, "perez": 0, "guevara": 0, "machin": 0, "learn": [0, 28], "engin": 0, "seth": [0, 3], "paulson": 0, "softwar": 0, "architect": 0, "albert": 0, "ying": 0, "harvard": 0, "universii": 0, "dayoon": 0, "kwon": 0, "univers": 0, "california": 0, "special": 0, "thank": 0, "scientif": [0, 12], "member": 0, "plot": [1, 2, 5, 8, 12], "result": [1, 8], "run": [1, 6, 8, 10, 12, 23, 28], "function": [1, 8, 12, 15, 17, 18, 19, 20, 21, 22, 24, 25, 26], "geo": [1, 4, 8, 14, 15, 16, 23], "data": [1, 4, 5, 7, 8, 12, 14, 15, 16, 18, 21, 22, 23, 28], "perform": [1, 4, 6, 8, 14, 16, 20, 23], "custom": [1, 4, 8, 14, 16, 20, 23], "imput": [1, 3, 4, 8, 14, 16, 23], "go": [2, 3, 6], "end": [2, 3, 6], "download": [2, 3, 6, 8], "full": [2, 3, 6], "exampl": [2, 3, 6, 25, 28], "code": [2, 3, 6, 8, 12, 28], "browser": [2, 3, 6], "via": [2, 3, 6], "binder": [2, 3, 6], "demonstr": [2, 12], "eas": 2, "librari": [2, 14, 15, 28], "data_librari": [2, 3, 28], "import": [2, 3, 6], "datalibrari": [2, 3, 15, 28], "comput": [2, 12], "averag": [2, 18, 19, 20, 23, 24], "reference_dataset": 2, "gse40279": [2, 11], "reference_averag": 2, "dnam": [2, 3, 16, 17, 18, 19, 20], "mean": 2, "axi": 2, "1": [2, 4, 6], "second": [2, 3, 6], "target_dataset": 2, "gse51057": [2, 11], "impute_from_standard": 2, "imputed_data": 2, "model": [2, 23], "galleri": [2, 3, 6, 8, 10], "compar": [2, 12], "epigenet": [2, 4, 11, 12, 14, 15, 16, 23], "model_galleri": [2, 3], "modelgalleri": [2, 3], "clock_model": 2, "horvathv1": [2, 3, 10], "original_age_predict": 2, "imputed_age_predict": 2, "seaborn": [2, 6], "sn": [2, 3, 6], "panda": [2, 3, 16, 21, 22], "pd": [2, 3, 17, 18, 19, 20], "matplotlib": [2, 6], "pyplot": [2, 6], "plt": [2, 6], "prepar": 2, "comparison_data": 2, "datafram": [2, 3, 16, 17, 18, 19, 20, 21, 22], "creat": [2, 28], "scatter": [2, 3], "figur": [2, 3], "figsiz": 2, "8": [2, 3, 18], "6": 2, "scatterplot": [2, 6], "x": [2, 6], "y": [2, 6], "titl": [2, 11], "v": [2, 28], "xlabel": [2, 6], "ylabel": [2, 6], "grid": 2, "true": 2, "show": [2, 12], "imputationdecor": [2, 23], "defin": [2, 14], "method": [2, 23, 24], "def": 2, "custom_impute_method": 2, "dnam_data": 2, "needed_cpg": 2, "return": [2, 14, 15, 17, 18, 19, 20, 21, 22, 23], "wrap": [2, 23], "decorated_clock": 2, "decorated_clock_predict": 2, "verifi": [2, 28], "direct": 2, "approach": 2, "same_result": 2, "all": [2, 8, 19, 20], "print": 2, "total": [2, 3, 4, 6, 7], "time": [2, 3, 6], "script": [2, 3, 6], "minut": [2, 3, 6], "35": [2, 4], "182": [2, 4], "estim": [2, 3, 6], "memori": [2, 3, 6], "usag": [2, 3, 6, 25], "12106": 2, "mb": [2, 3, 4, 6, 7], "python": [2, 3, 6, 8, 28], "sourc": [2, 3, 6, 8, 10, 12, 14, 15, 18, 28], "plot_custom_imputation_implement": [2, 4], "py": [2, 3, 4, 6, 7], "jupyt": [2, 3, 6, 8, 28], "notebook": [2, 3, 6, 8, 28], "ipynb": [2, 3, 6], "gener": [2, 3, 6, 8], "sphinx": [2, 3, 6, 8], "your": [3, 6], "dna": [3, 11, 14, 17, 18, 19, 20, 24], "calcul": 3, "multipl": 3, "them": [3, 8, 14], "against": 3, "chronolog": [3, 12], "gse41169": [3, 11], "blood": [3, 5, 6, 8, 10, 11], "data_sourc": 3, "get": [3, 12, 14, 23, 28], "ha": [3, 28], "well": 3, "each": [3, 21, 22], "subject": [3, 28], "methylation_data": 3, "note": 3, "default": [3, 14, 18, 23], "miss": [3, 15, 17, 18, 19, 20, 24], "To": [3, 28], "chang": [3, 11, 28], "behavior": [3, 28], "set": [3, 11, 14, 24], "paramet": [3, 14, 15, 16, 17, 18, 19, 20, 22, 23], "when": 3, "horvath_result": 3, "predict": [3, 10, 23], "hannum_result": 3, "hannum": [3, 10], "phenoage_result": 3, "phenoag": [3, 10], "actual_ag": 3, "plot_data": 3, "horvath": [3, 12, 28], "index": 3, "relplot": 3, "kind": 3, "home": 3, "longev": 3, "venv": 3, "lib": 3, "python3": 3, "10": [3, 28], "site": [3, 16, 17, 18, 19, 20], "packag": 3, "axisgrid": 3, "123": 3, "userwarn": 3, "layout": 3, "tight": 3, "self": 3, "_figur": 3, "tight_layout": 3, "arg": 3, "kwarg": 3, "facetgrid": 3, "object": [3, 23], "0x7fbd6ae55330": 3, "0": [3, 6, 7, 18], "498": [3, 4], "1769": [3, 4], "plot_epigenetic_clocks_on_geo": [3, 4], "01": [4, 7], "43": 4, "680": 4, "auto_examples_00_epigenetic_biomark": 4, "file": [4, 7, 10, 14, 15], "12105": 4, "9": [4, 6, 7], "clock": [4, 5, 12, 14, 15, 16, 23, 28], "00": [4, 7], "08": 4, "surviv": [5, 8, 12], "curv": [5, 8], "base": [5, 8, 14, 18, 23], "us": [5, 8, 11, 12, 17, 18, 19, 21, 28], "standard": [5, 8, 17], "test": [5, 8], "phenotyp": [5, 7, 8, 22], "nhane": [5, 7, 8, 12, 22], "exam": 6, "analys": [6, 12, 28], "load_nhan": 6, "year": [6, 10, 22], "df": [6, 21, 22], "years_until_death": 6, "months_until_death": 6, "12": 6, "hematologi": 6, "phenotypic_ag": 6, "": [6, 15], "2": 6, "ax": 6, "lifelin": 6, "kaplanmeierfitt": 6, "biologically_old": 6, "subplot": 6, "group": [6, 22], "ix": 6, "t": 6, "e": 6, "is_dead": 6, "kmf": 6, "fit": 6, "label": 6, "plot_survival_funct": 6, "text": 6, "5": 6, "23": 6, "52222222222222": 6, "129": [6, 7], "plot_nhan": [6, 7], "auto_examples_01_composite_biomark": 7, "If": [8, 14, 15, 18, 23, 28], "you": [8, 28], "want": 8, "make": 8, "sure": 8, "directori": 8, "where": [8, 16, 21, 22], "have": [8, 28], "write": 8, "permiss": 8, "copi": 8, "instal": 8, "biolean": 8, "manual": 8, "follow": 8, "instruct": [8, 28], "auto_examples_python": 8, "zip": 8, "auto_examples_jupyt": 8, "support": [10, 11, 12, 14, 22, 23], "number": [10, 11, 22], "can": [10, 12, 14, 23, 28], "easili": [10, 12, 14, 28], "name": [10, 23], "speci": [10, 23], "tissu": [10, 23], "coeffici": 10, "2013": 10, "human": [10, 11], "multi": 10, "paper": 10, "lin": 10, "2016": 10, "2018": 10, "horvathv2": 10, "skin": 10, "pedb": 10, "2019": 10, "buccal": 10, "zhang_10": 10, "mortal": 10, "risk": 10, "dunedinpoam38": 10, "2020": 10, "rate": 10, "dunedinpac": [10, 12, 28], "2022": 10, "alcoholmccartnei": 10, "alcohol": 10, "consumpt": 10, "bmi_mccartnei": 10, "bmi": 10, "dnamtl": 10, "adipos": 10, "telomer": 10, "length": 10, "hrsinchphenoag": 10, "knight": 10, "cord": 10, "gestat": 10, "leecontrol": 10, "placenta": 10, "leerefinedrobust": 10, "leerobust": 10, "smokingmccartnei": 10, "smoke": 10, "statu": 10, "methyl": [11, 14, 16, 17, 18, 19, 20, 24], "id": [11, 15], "format": [11, 14], "sampl": [11, 16, 17, 18, 19, 20], "present": 11, "sex": 11, "genom": [11, 16], "wide": 11, "profil": 11, "reveal": 11, "quantit": 11, "view": 11, "o": 11, "illumina450k": 11, "656": 11, "ye": 11, "gse19711": 11, "unit": 11, "kingdom": 11, "ovar": 11, "illumina27k": 11, "540": 11, "No": 11, "methylom": 11, "analysi": 11, "associ": [11, 16], "me": 11, "329": 11, "gse42861": 11, "differenti": 11, "rheumatoid": 11, "arthriti": 11, "689": 11, "dutch": 11, "popul": 11, "95": 11, "gse51032": 11, "epic": 11, "itali": 11, "hugef": 11, "845": 11, "gse73103": 11, "mani": [11, 12, 28], "obes": 11, "snp": 11, "strongli": 11, "met": 11, "355": 11, "gse69270": 11, "middl": 11, "indi": 11, "184": 11, "gse36054": 11, "healthi": 11, "children": 11, "192": 11, "gse64495": 11, "sever": [11, 12], "113": 11, "gse30870": 11, "newborn": 11, "nonagenarian": 11, "40": 11, "enabl": [12, 28], "easi": [12, 28], "versatil": [12, 28], "It": [12, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 28], "provid": [12, 24, 28], "tool": [12, 28], "load": [12, 14, 15, 28], "publicli": [12, 28], "like": [12, 28], "gene": [12, 28], "express": [12, 28], "omnibu": [12, 28], "nation": [12, 22, 28], "health": [12, 22, 28], "nutrit": [12, 22, 28], "examim": [12, 28], "survei": [12, 22, 28], "framingham": [12, 21, 28], "heart": [12, 21, 28], "studi": [12, 21, 28], "also": [12, 28], "contain": [12, 15, 18, 20, 28], "refer": [12, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 28], "implemen": [12, 28], "common": [12, 28], "other": [12, 18, 28], "onli": [12, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 28], "few": [12, 28], "line": [12, 28], "financi": 12, "methuselah": 12, "foundat": 12, "respect": 12, "quickstart": 12, "start": [12, 28], "discov": 12, "read": 12, "2024": 12, "join": 12, "challeng": 12, "build": 12, "better": 12, "how": [12, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 28], "biolog": [12, 23], "relationship": 12, "between": 12, "variou": [12, 23, 24], "point": 12, "class": [13, 14, 15, 16, 23, 25, 27], "page": [14, 15, 16, 17, 18, 19, 20, 21, 22, 23], "document": [14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 28], "explain": [14, 15, 16, 17, 18, 19, 20, 21, 22, 23], "signatur": [14, 15, 16, 17, 18, 19, 20, 21, 22, 23], "pleas": [14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 25, 28], "user": [14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 25], "guid": [14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 25], "big": [14, 15, 16, 17, 18, 19, 20, 21, 22, 23], "pictur": [14, 15, 16, 17, 18, 19, 20, 21, 22, 23], "library_fil": 14, "none": [14, 19, 20, 23], "manag": [14, 23], "collect": [14, 23], "respons": 14, "store": 14, "retriev": [14, 23], "new": 14, "ad": 14, "runtim": 14, "current": 14, "__init__": [14, 15, 16, 23], "initi": [14, 15, 16, 23], "instanc": [14, 15, 16, 23], "str": [14, 18, 19, 20, 23], "option": [14, 18, 19, 20, 23], "path": [14, 15], "load_sourc": 14, "given": 14, "append": 14, "source_id": 14, "its": [14, 23], "identifi": 14, "found": [14, 23], "otherwis": 14, "lookup_sourc": 14, "organ": [14, 16], "look": 14, "up": 14, "specifi": [14, 23], "filter": [14, 23], "A": [14, 15, 16, 21, 22, 23], "list": [14, 18, 19, 20], "match": [14, 23], "criteria": [14, 23], "repres": [15, 16, 21, 22], "singl": 15, "encapsul": 15, "detail": [15, 28], "includ": [15, 28], "metadata": [15, 16], "about": [15, 21, 22], "rais": [15, 18, 23], "valueerror": [15, 18, 23], "ani": [15, 28], "requir": [15, 18], "field": [15, 16], "parser": 15, "dure": 15, "intend": 15, "yaml": 15, "dict": [15, 23], "dictionari": [15, 23], "properti": 15, "required_field": 15, "kei": 15, "item": 15, "an": [15, 18, 20, 21, 22, 23], "geodata": 15, "pars": 15, "geograph": 15, "type": [15, 16, 17, 18, 19, 20, 21, 22, 23], "focu": 16, "facilit": 16, "access": 16, "row": [16, 17, 18, 19, 20, 21, 22], "differ": 16, "column": [16, 17, 18, 19, 20, 21, 22], "valu": [17, 18, 19, 20, 23, 24], "dataset": [17, 18, 19, 20, 24], "cpg": [17, 18, 19, 20], "fill": [17, 18, 19, 20], "cpg_sourc": 18, "required_cpg": 18, "threshold": 18, "below": 18, "replac": 18, "extern": [18, 20], "while": 18, "exist": 18, "seri": [18, 20], "need": 18, "final": 18, "float": 18, "determin": 18, "strategi": [18, 24], "certain": 18, "both": 18, "cpgs_to_imput": [19, 20], "itself": 19, "cpg_averag": 20, "individu": [21, 22], "measur": [21, 22], "examin": 22, "Not": 22, "come": 22, "two": 22, "pass": [22, 23], "should": 22, "later": 22, "input": 22, "2010": 22, "2012": 22, "model_definit": 23, "produc": 23, "search": 23, "definit": 23, "imputation_method": 23, "possibl": 23, "dunedin": 23, "sesame_450k": 23, "request": [23, 28], "possibli": 23, "keyerror": 23, "invalid": 23, "modul": 24, "These": 24, "allow": 24, "simpl": 24, "more": [24, 25], "complex": 24, "hybrid": 24, "inform": 25, "prereleas": 28, "version": 28, "There": 28, "mai": 28, "bug": 28, "interfac": 28, "http": 28, "github": 28, "com": 28, "bio": 28, "homepag": 28, "io": 28, "3": 28, "pip": 28, "wa": 28, "correctli": 28, "open": 28, "error": 28, "check": 28, "out": 28, "some": 28, "team": 28, "answer": 28, "question": 28, "discuss": 28, "featur": 28, "relat": 28, "find": 28, "we": 28, "replic": 28, "expect": 28, "actual": 28, "develop": 28, "setup": 28, "repo": 28}, "objects": {"biolearn": [[13, 0, 0, "-", "data_library"], [24, 0, 0, "-", "imputation"], [26, 0, 0, "-", "load"], [27, 0, 0, "-", "model_gallery"]], "biolearn.data_library": [[14, 1, 1, "", "DataLibrary"], [15, 1, 1, "", "DataSource"], [16, 1, 1, "", "GeoData"]], "biolearn.data_library.DataLibrary": [[14, 2, 1, "", "__init__"], [14, 2, 1, "", "get"], [14, 2, 1, "", "load_sources"], [14, 2, 1, "", "lookup_sources"]], "biolearn.data_library.DataSource": [[15, 3, 1, "", "REQUIRED_FIELDS"], [15, 2, 1, "", "__init__"], [15, 2, 1, "", "load"]], "biolearn.data_library.GeoData": [[16, 2, 1, "", "__init__"], [16, 3, 1, "", "dnam"], [16, 3, 1, "", "metadata"]], "biolearn.imputation": [[17, 4, 1, "", "biolearn_impute"], [18, 4, 1, "", "hybrid_impute"], [19, 4, 1, "", "impute_from_average"], [20, 4, 1, "", "impute_from_standard"]], "biolearn.load": [[21, 4, 1, "", "load_fhs"], [22, 4, 1, "", "load_nhanes"]], "biolearn.model_gallery": [[23, 1, 1, "", "ModelGallery"]], "biolearn.model_gallery.ModelGallery": [[23, 2, 1, "", "__init__"], [23, 2, 1, "", "get"], [23, 2, 1, "", "search"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:attribute", "4": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "attribute", "Python attribute"], "4": ["py", "function", "Python function"]}, "titleterms": {"team": 0, "core": 0, "develop": 0, "other": [0, 10], "contributor": 0, "epigenet": [1, 3, 8], "clock": [1, 2, 3, 8, 10], "exampl": [1, 5, 8, 12, 14, 15, 16, 20, 22, 23], "perform": 2, "custom": 2, "imput": [2, 17, 18, 19, 20, 24], "load": [2, 3, 6, 21, 22, 26], "methyl": [2, 3], "data": [2, 3, 6, 11, 24, 26], "from": [2, 3], "geo": [2, 3, 11], "dataset": [2, 3], "us": [2, 3, 14, 15, 16, 20, 22, 23], "refer": [2, 25], "up": [2, 3], "target": 2, "run": [2, 3], "now": [2, 3], "predict": 2, "befor": 2, "after": 2, "visual": 2, "comparison": 2, "ag": [2, 3, 6], "you": 2, "can": 2, "also": 2, "build": 2, "an": 2, "decor": 2, "bundl": 2, "get": 2, "same": 2, "result": [2, 3], "function": 2, "directli": 2, "your": 2, "first": 3, "some": 3, "librari": 3, "three": 3, "differ": 3, "produc": 3, "final": 3, "exctract": 3, "metadata": 3, "plot": [3, 6], "seaborn": 3, "comput": [4, 7], "time": [4, 7], "composit": [5, 8], "biomark": [5, 8], "phenotyp": 6, "nhane": 6, "2010": 6, "calcul": 6, "biolog": 6, "base": 6, "phenotypicag": 6, "show": 6, "relat": 6, "between": 6, "chronolog": 6, "surviv": 6, "curv": 6, "peopl": 6, "acceler": 6, "older": 6, "v": 6, "deceler": 6, "younger": 6, "model": 10, "sourc": 11, "biolearn": [12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 26, 27, 28], "featur": 12, "data_librari": [13, 14, 15, 16], "datalibrari": [13, 14], "datasourc": 15, "geodata": 16, "biolearn_imput": 17, "hybrid_imput": 18, "impute_from_averag": 19, "impute_from_standard": 20, "load_fh": 21, "load_nhan": 22, "model_galleri": [23, 27], "modelgalleri": [23, 27], "util": [24, 26], "api": 25, "quickstart": 28, "import": 28, "link": 28, "requir": 28, "instal": 28, "discord": 28, "server": 28, "issu": 28, "contribut": 28}, "envversion": {"sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinxcontrib.bibtex": 9, "sphinx": 60}, "alltitles": {"Team": [[0, "team"]], "Core developers": [[0, "core-developers"]], "Other contributors": [[0, "other-contributors"]], "Epigenetic Clock Examples": [[1, "epigenetic-clock-examples"], [8, "epigenetic-clock-examples"]], "Performing custom imputations": [[2, "performing-custom-imputations"]], "Load methylation data from a GEO dataset to use as a reference for imputation": [[2, "load-methylation-data-from-a-geo-dataset-to-use-as-a-reference-for-imputation"]], "Load up a target dataset and run the imputation": [[2, "load-up-a-target-dataset-and-run-the-imputation"]], "Now run clock predictions on the dataset before and after": [[2, "now-run-clock-predictions-on-the-dataset-before-and-after"]], "Visualize the comparison of age predictions": [[2, "visualize-the-comparison-of-age-predictions"]], "You can also build an imputation decorator to bundle with the clock": [[2, "you-can-also-build-an-imputation-decorator-to-bundle-with-the-clock"]], "You get the same results as running the function directly on your dataset": [[2, "you-get-the-same-results-as-running-the-function-directly-on-your-dataset"]], "\u201cEpigenetic Clocks\u201d in GEO Data": [[3, "epigenetic-clocks-in-geo-data"]], "First load up some methylation data from GEO using the data library": [[3, "first-load-up-some-methylation-data-from-geo-using-the-data-library"]], "Now run three different clocks on the dataset to produce epigenetic clock ages": [[3, "now-run-three-different-clocks-on-the-dataset-to-produce-epigenetic-clock-ages"]], "Finally exctract the age data from the metadata from GEO and plot the results using seaborn": [[3, "finally-exctract-the-age-data-from-the-metadata-from-geo-and-plot-the-results-using-seaborn"]], "Computation times": [[4, "computation-times"], [7, "computation-times"]], "Composite Biomarkers Examples": [[5, "composite-biomarkers-examples"], [8, "composite-biomarkers-examples"]], "\u201cPhenotypic Ages\u201d in NHANES Data": [[6, "phenotypic-ages-in-nhanes-data"]], "Loading NHANES 2010 data": [[6, "loading-nhanes-2010-data"]], "Calculate \u201cbiological age\u201d based on PhenotypicAge": [[6, "calculate-biological-age-based-on-phenotypicage"]], "Show relation between biological age and chronological age": [[6, "show-relation-between-biological-age-and-chronological-age"]], "Plot survival curve for people with accelerated aging (older biological age) vs decelerated aging (younger biological age)": [[6, "plot-survival-curve-for-people-with-accelerated-aging-older-biological-age-vs-decelerated-aging-younger-biological-age"]], "Examples": [[8, "examples"]], "Biolearn": [[12, "biolearn"], [28, "biolearn"]], "Featured examples": [[12, "featured-examples"]], "biolearn.data_library: DataLibrary": [[13, "module-biolearn.data_library"]], "biolearn.data_library.DataLibrary": [[14, "biolearn-data-library-datalibrary"]], "Examples using biolearn.data_library.DataLibrary": [[14, "examples-using-biolearn-data-library-datalibrary"]], "biolearn.data_library.DataSource": [[15, "biolearn-data-library-datasource"]], "Examples using biolearn.data_library.DataSource": [[15, "examples-using-biolearn-data-library-datasource"]], "biolearn.data_library.GeoData": [[16, "biolearn-data-library-geodata"]], "Examples using biolearn.data_library.GeoData": [[16, "examples-using-biolearn-data-library-geodata"]], "biolearn.imputation.biolearn_impute": [[17, "biolearn-imputation-biolearn-impute"]], "biolearn.imputation.hybrid_impute": [[18, "biolearn-imputation-hybrid-impute"]], "biolearn.imputation.impute_from_average": [[19, "biolearn-imputation-impute-from-average"]], "biolearn.imputation.impute_from_standard": [[20, "biolearn-imputation-impute-from-standard"]], "Examples using biolearn.imputation.impute_from_standard": [[20, "examples-using-biolearn-imputation-impute-from-standard"]], "biolearn.load.load_fhs": [[21, "biolearn-load-load-fhs"]], "biolearn.load.load_nhanes": [[22, "biolearn-load-load-nhanes"]], "Examples using biolearn.load.load_nhanes": [[22, "examples-using-biolearn-load-load-nhanes"]], "biolearn.model_gallery.ModelGallery": [[23, "biolearn-model-gallery-modelgallery"]], "Examples using biolearn.model_gallery.ModelGallery": [[23, "examples-using-biolearn-model-gallery-modelgallery"]], "biolearn.imputation: Data Imputation Utilities": [[24, "biolearn-imputation-data-imputation-utilities"]], "API References": [[25, "api-references"]], "biolearn.load: Data Loading Utilities": [[26, "module-biolearn.load"]], "biolearn.model_gallery: ModelGallery": [[27, "biolearn-model-gallery-modelgallery"]], "Quickstart": [[28, "quickstart"]], "Important links": [[28, "important-links"]], "Requirements": [[28, "requirements"]], "Install": [[28, "install"]], "Discord server": [[28, "discord-server"]], "Issues": [[28, "issues"]], "Contributing": [[28, "contributing"]], "Clocks and Other Models": [[10, "clocks-and-other-models"]], "GEO Data Sources": [[11, "geo-data-sources"]]}, "indexentries": {"biolearn.data_library": [[13, "module-biolearn.data_library"]], "module": [[13, "module-biolearn.data_library"], [24, "module-biolearn.imputation"], [26, "module-biolearn.load"], [27, "module-biolearn.model_gallery"]], "datalibrary (class in biolearn.data_library)": [[14, "biolearn.data_library.DataLibrary"]], "__init__() (biolearn.data_library.datalibrary method)": [[14, "biolearn.data_library.DataLibrary.__init__"]], "get() (biolearn.data_library.datalibrary method)": [[14, "biolearn.data_library.DataLibrary.get"]], "load_sources() (biolearn.data_library.datalibrary method)": [[14, "biolearn.data_library.DataLibrary.load_sources"]], "lookup_sources() (biolearn.data_library.datalibrary method)": [[14, "biolearn.data_library.DataLibrary.lookup_sources"]], "datasource (class in biolearn.data_library)": [[15, "biolearn.data_library.DataSource"]], "required_fields (biolearn.data_library.datasource attribute)": [[15, "biolearn.data_library.DataSource.REQUIRED_FIELDS"]], "__init__() (biolearn.data_library.datasource method)": [[15, "biolearn.data_library.DataSource.__init__"]], "load() (biolearn.data_library.datasource method)": [[15, "biolearn.data_library.DataSource.load"]], "geodata (class in biolearn.data_library)": [[16, "biolearn.data_library.GeoData"]], "__init__() (biolearn.data_library.geodata method)": [[16, "biolearn.data_library.GeoData.__init__"]], "dnam (biolearn.data_library.geodata attribute)": [[16, "biolearn.data_library.GeoData.dnam"]], "metadata (biolearn.data_library.geodata attribute)": [[16, "biolearn.data_library.GeoData.metadata"]], "biolearn_impute() (in module biolearn.imputation)": [[17, "biolearn.imputation.biolearn_impute"]], "hybrid_impute() (in module biolearn.imputation)": [[18, "biolearn.imputation.hybrid_impute"]], "impute_from_average() (in module biolearn.imputation)": [[19, "biolearn.imputation.impute_from_average"]], "impute_from_standard() (in module biolearn.imputation)": [[20, "biolearn.imputation.impute_from_standard"]], "load_fhs() (in module biolearn.load)": [[21, "biolearn.load.load_fhs"]], "load_nhanes() (in module biolearn.load)": [[22, "biolearn.load.load_nhanes"]], "modelgallery (class in biolearn.model_gallery)": [[23, "biolearn.model_gallery.ModelGallery"]], "__init__() (biolearn.model_gallery.modelgallery method)": [[23, "biolearn.model_gallery.ModelGallery.__init__"]], "get() (biolearn.model_gallery.modelgallery method)": [[23, "biolearn.model_gallery.ModelGallery.get"]], "search() (biolearn.model_gallery.modelgallery method)": [[23, "biolearn.model_gallery.ModelGallery.search"]], "biolearn.imputation": [[24, "module-biolearn.imputation"]], "biolearn.load": [[26, "module-biolearn.load"]], "biolearn.model_gallery": [[27, "module-biolearn.model_gallery"]]}})