Search.setIndex({"docnames": ["authors", "auto_examples/00_omics_biomarkers/index", "auto_examples/00_omics_biomarkers/plot_custom_imputation_implementation", "auto_examples/00_omics_biomarkers/plot_down_syndrome_model", "auto_examples/00_omics_biomarkers/plot_epigenetic_clocks_on_geo", "auto_examples/00_omics_biomarkers/plot_transcriptomic_clock", "auto_examples/00_omics_biomarkers/sg_execution_times", "auto_examples/01_composite_biomarkers/index", "auto_examples/01_composite_biomarkers/plot_nhanes", "auto_examples/01_composite_biomarkers/sg_execution_times", "auto_examples/02_challenge_submissions/index", "auto_examples/02_challenge_submissions/plot_challenge_data", "auto_examples/02_challenge_submissions/sg_execution_times", "auto_examples/02_challenge_submissions/simple_competiton_submission", "auto_examples/02_challenge_submissions/training_simple_model", "auto_examples/03_deconvolution/index", "auto_examples/03_deconvolution/plot_blood_deconvolution", "auto_examples/03_deconvolution/sg_execution_times", "auto_examples/index", "auto_examples/sg_execution_times", "changes/names", "clocks", "data", "index", "methylation-standard", "model-usage", "modules/cache", "modules/data_library", "modules/generated/biolearn.cache.LocalFolderCache", "modules/generated/biolearn.cache.NoCache", "modules/generated/biolearn.data_library.DataLibrary", "modules/generated/biolearn.data_library.DataSource", "modules/generated/biolearn.data_library.GeoData", "modules/generated/biolearn.imputation.hybrid_impute", "modules/generated/biolearn.imputation.impute_from_average", "modules/generated/biolearn.imputation.impute_from_standard", "modules/generated/biolearn.load.load_fhs", "modules/generated/biolearn.load.load_nhanes", "modules/generated/biolearn.model_gallery.ModelGallery", "modules/imputation", "modules/index", "modules/load", "modules/model_gallery", "quickstart", "roadmap", "sg_execution_times"], "filenames": ["authors.rst", "auto_examples/00_omics_biomarkers/index.rst", "auto_examples/00_omics_biomarkers/plot_custom_imputation_implementation.rst", "auto_examples/00_omics_biomarkers/plot_down_syndrome_model.rst", "auto_examples/00_omics_biomarkers/plot_epigenetic_clocks_on_geo.rst", "auto_examples/00_omics_biomarkers/plot_transcriptomic_clock.rst", "auto_examples/00_omics_biomarkers/sg_execution_times.rst", "auto_examples/01_composite_biomarkers/index.rst", "auto_examples/01_composite_biomarkers/plot_nhanes.rst", "auto_examples/01_composite_biomarkers/sg_execution_times.rst", "auto_examples/02_challenge_submissions/index.rst", "auto_examples/02_challenge_submissions/plot_challenge_data.rst", "auto_examples/02_challenge_submissions/sg_execution_times.rst", "auto_examples/02_challenge_submissions/simple_competiton_submission.rst", "auto_examples/02_challenge_submissions/training_simple_model.rst", "auto_examples/03_deconvolution/index.rst", "auto_examples/03_deconvolution/plot_blood_deconvolution.rst", "auto_examples/03_deconvolution/sg_execution_times.rst", "auto_examples/index.rst", "auto_examples/sg_execution_times.rst", "changes/names.rst", "clocks.rst", "data.rst", "index.rst", "methylation-standard.rst", "model-usage.rst", "modules/cache.rst", "modules/data_library.rst", "modules/generated/biolearn.cache.LocalFolderCache.rst", "modules/generated/biolearn.cache.NoCache.rst", "modules/generated/biolearn.data_library.DataLibrary.rst", "modules/generated/biolearn.data_library.DataSource.rst", "modules/generated/biolearn.data_library.GeoData.rst", "modules/generated/biolearn.imputation.hybrid_impute.rst", "modules/generated/biolearn.imputation.impute_from_average.rst", "modules/generated/biolearn.imputation.impute_from_standard.rst", "modules/generated/biolearn.load.load_fhs.rst", "modules/generated/biolearn.load.load_nhanes.rst", "modules/generated/biolearn.model_gallery.ModelGallery.rst", "modules/imputation.rst", "modules/index.rst", "modules/load.rst", "modules/model_gallery.rst", "quickstart.md", "roadmap.rst", "sg_execution_times.rst"], "titles": ["Team", "Omics Biomarker Examples", "Performing custom imputations", "Down Syndrome Epigenetic Plotting", "\u201cEpigenetic Clocks\u201d in GEO Data", "\u201cTranscriptomic Clock\u201d using GEO Data", "Computation times", "Composite Biomarkers Examples", "\u201cPhenotypic Ages\u201d in NHANES Data", "Computation times", "Challenge Submission Examples", "Exploring the Challenge Data", "Computation times", "Building a competition submission using an existing model", "Training an ElasticNet model", "Deconvolution Examples", "\u201cDeconvolution Example\u201d", "Computation times", "Examples", "Computation times", "&lt;no title&gt;", "Clocks and Other Models", "GEO Data Sources", "Biolearn", "DNA Methylation Array Data Standard V-2410", "Model IP and Usage", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.cache</span></code>: Cache", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.data_library</span></code>: DataLibrary", "biolearn.cache.LocalFolderCache", "biolearn.cache.NoCache", "biolearn.data_library.DataLibrary", "biolearn.data_library.DataSource", "biolearn.data_library.GeoData", "biolearn.imputation.hybrid_impute", "biolearn.imputation.impute_from_average", "biolearn.imputation.impute_from_standard", "biolearn.load.load_fhs", "biolearn.load.load_nhanes", "biolearn.model_gallery.ModelGallery", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.imputation</span></code>: Data Imputation Utilities", "API References", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.load</span></code>: Data Loading Utilities", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.model_gallery</span></code>: ModelGallery", "Quickstart", "Biolearn and the Future of Biomarkers", "Computation times"], "terms": {"thi": [0, 2, 3, 4, 5, 8, 11, 13, 14, 16, 25, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 43, 44], "work": [0, 44], "i": [0, 23, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 40, 43, 44], "made": [0, 44], "avail": [0, 23, 24, 38, 43, 44], "commun": [0, 13], "peopl": 0, "which": [0, 14, 28, 29, 32, 37], "origin": [0, 2], "from": [0, 6, 7, 8, 9, 12, 13, 14, 16, 17, 18, 19, 21, 22, 23, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 39, 43, 44, 45], "biomark": [0, 10, 22, 23, 30, 43], "ag": [0, 7, 9, 10, 11, 14, 18, 21, 22, 23, 24, 37, 43, 44, 45], "consortium": [0, 23], "The": [0, 24, 28, 29, 30, 38, 43, 44], "biolearn": [0, 2, 3, 4, 5, 8, 11, 13, 14, 16, 21, 22, 40], "ar": [0, 2, 14, 23, 24, 25, 30, 31, 32, 33, 37, 38, 43, 44], "mahdi": 0, "moqri": [0, 23], "research": [0, 23, 30, 44], "boa": 0, "execut": [0, 6, 9, 12, 17, 18, 19, 43, 45], "martin": 0, "perez": [0, 23], "guevara": [0, 23], "machin": 0, "learn": [0, 23, 43, 44], "engin": 0, "seth": 0, "paulson": [0, 23], "softwar": 0, "architect": 0, "albert": 0, "ying": [0, 23], "harvard": 0, "universii": 0, "alec": 0, "eam": 0, "dayoon": 0, "kwon": [0, 23], "univers": 0, "california": 0, "mehrnoosh": 0, "emamifar": [0, 23], "northeastern": 0, "ai4lif": 0, "institut": 0, "special": 0, "thank": 0, "scientif": 0, "member": [0, 11], "plot": [1, 2, 6, 7, 11, 15, 18, 23, 32, 45], "result": [1, 3, 13, 15, 18], "run": [1, 8, 13, 15, 16, 18, 21, 23, 38, 43], "base": [1, 7, 14, 18, 30, 33, 38], "model": [1, 2, 5, 10, 12, 18, 30, 38, 44, 45], "function": [1, 18, 31, 33, 34, 35, 36, 37, 39, 40, 41], "includ": [1, 18, 24, 31, 32, 43, 44], "methyl": [1, 18, 22, 30, 32, 33, 34, 35, 39, 44], "transcriptom": [1, 6, 18, 32, 45], "data": [1, 6, 7, 9, 10, 12, 16, 18, 23, 28, 30, 31, 32, 33, 36, 37, 38, 43, 44, 45], "clock": [1, 6, 7, 18, 23, 25, 31, 32, 43, 45], "us": [1, 6, 7, 10, 12, 18, 22, 23, 24, 25, 28, 29, 33, 34, 36, 43, 44, 45], "geo": [1, 6, 18, 24, 30, 31, 32, 44, 45], "epigenet": [1, 2, 6, 18, 22, 23, 31, 32, 44, 45], "down": [1, 6, 14, 18, 21, 32, 45], "syndrom": [1, 6, 18, 21, 32, 45], "perform": [1, 6, 8, 18, 32, 35, 45], "custom": [1, 6, 18, 32, 35, 45], "imput": [1, 4, 6, 18, 32, 38, 45], "go": [2, 3, 4, 5, 8, 11, 13, 14, 16], "end": [2, 3, 4, 5, 8, 11, 13, 14, 16], "download": [2, 3, 4, 5, 8, 11, 13, 14, 16, 18], "full": [2, 3, 4, 5, 8, 11, 13, 14, 16], "exampl": [2, 3, 4, 5, 6, 8, 9, 11, 12, 13, 14, 17, 19, 43, 45], "code": [2, 3, 4, 5, 8, 11, 13, 14, 16, 18, 23, 43, 44], "browser": [2, 3, 4, 5, 8, 11, 13, 14, 16], "via": [2, 3, 4, 5, 8, 11, 13, 14, 16, 44], "binder": [2, 3, 4, 5, 8, 11, 13, 14, 16], "demonstr": [2, 23], "eas": 2, "librari": [2, 22, 23, 30, 38, 43, 44], "data_librari": [2, 3, 4, 5, 11, 13, 14, 16, 43], "import": [2, 3, 4, 5, 8, 11, 13, 14, 16], "datalibrari": [2, 3, 4, 5, 11, 14, 16, 31, 43], "geodata": [2, 13, 14, 31], "comput": [2, 23], "averag": [2, 24, 33, 34, 35, 38, 39], "reference_dataset": 2, "gse40279": [2, 14, 22], "reference_averag": 2, "dnam": [2, 4, 11, 14, 24, 32, 33, 34, 35], "mean": [2, 14], "axi": 2, "1": [2, 5, 6, 8, 9, 11, 14, 16, 17, 24, 44, 45], "second": [2, 3, 4, 5, 8, 11, 16], "target_dataset": 2, "gse51057": [2, 22], "impute_from_standard": 2, "imputed_data": 2, "imputed_dataset": 2, "metadata": [2, 3, 5, 14, 16, 31, 32], "galleri": [2, 3, 4, 5, 8, 11, 13, 14, 16, 18, 21, 45], "compar": [2, 23], "model_galleri": [2, 3, 4, 5, 13, 16], "modelgalleri": [2, 3, 4, 5, 13, 16], "clock_model": 2, "horvathv1": [2, 4, 11, 21, 25], "original_age_predict": 2, "imputed_age_predict": 2, "seaborn": [2, 3, 8, 11, 16], "sn": [2, 3, 4, 5, 8, 11, 16], "panda": [2, 4, 11, 14, 32, 36, 37], "pd": [2, 4, 11, 14, 33, 34, 35], "matplotlib": [2, 3, 5, 8, 11, 14, 16], "pyplot": [2, 3, 5, 8, 11, 14, 16], "plt": [2, 3, 5, 8, 11, 14, 16], "prepar": [2, 44], "comparison_data": 2, "datafram": [2, 4, 11, 14, 32, 33, 34, 35, 36, 37], "creat": [2, 3, 11, 32, 43], "scatter": [2, 4, 11, 14], "figur": [2, 3, 11, 14, 16], "figsiz": [2, 3, 11, 14, 16], "8": [2, 9, 11, 12, 14, 33, 45], "6": [2, 3, 11, 14, 17, 22, 45], "scatterplot": [2, 8, 16], "x": [2, 3, 5, 8, 11, 14, 16], "y": [2, 3, 5, 8, 11, 14, 16], "titl": [2, 3, 5, 11, 14, 16, 22, 31], "v": [2, 14, 23, 43], "xlabel": [2, 5, 8, 11, 14, 16], "ylabel": [2, 5, 8, 11, 14, 16], "grid": [2, 11, 14], "true": [2, 3, 11, 14], "show": [2, 5, 11, 13, 14, 23], "imputationdecor": 2, "defin": [2, 14, 30], "method": [2, 31, 38, 39], "def": 2, "custom_impute_method": 2, "dnam_data": 2, "needed_cpg": 2, "return": [2, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38], "wrap": 2, "decorated_clock": 2, "decorated_clock_predict": 2, "verifi": [2, 43], "direct": 2, "approach": [2, 44], "same_result": 2, "all": [2, 18, 24, 25, 28, 29, 30, 32, 34, 35, 45], "print": [2, 5, 14], "total": [2, 3, 4, 5, 6, 8, 9, 11, 12, 16, 17, 19, 21, 23, 45], "time": [2, 3, 4, 5, 8, 11, 14, 16, 44], "script": [2, 3, 4, 5, 8, 11, 16], "3": [2, 3, 6, 11, 12, 14, 43, 45], "minut": [2, 3, 4, 5, 8, 11, 16, 44], "9": [2, 6, 8, 11, 17, 45], "803": [2, 6, 45], "estim": [2, 3, 4, 5, 8, 11, 13, 14], "memori": [2, 3, 4, 5, 8, 11, 13, 14, 16, 22], "usag": [2, 3, 4, 5, 8, 11, 13, 14, 16], "11895": [2, 6, 45], "mb": [2, 3, 4, 5, 6, 8, 9, 11, 12, 13, 14, 16, 17, 19, 45], "jupyt": [2, 3, 4, 5, 8, 11, 13, 14, 16, 18, 43], "notebook": [2, 3, 4, 5, 8, 11, 13, 14, 16, 18, 43], "plot_custom_imputation_implement": [2, 6, 45], "ipynb": [2, 3, 4, 5, 8, 11, 13, 14, 16], "python": [2, 3, 4, 5, 8, 11, 13, 14, 16, 18, 43], "sourc": [2, 3, 4, 5, 8, 11, 13, 14, 16, 18, 21, 23, 30, 31, 33, 43, 44], "py": [2, 3, 4, 5, 6, 8, 9, 11, 12, 13, 14, 16, 17, 45], "gener": [2, 4, 5, 8, 10, 11, 13, 14, 16, 18, 32, 44], "sphinx": [2, 3, 4, 5, 8, 11, 13, 14, 16, 18], "your": [3, 4, 5, 8, 11, 13, 14, 16, 23, 44], "dna": [3, 4, 16, 21, 22, 30, 33, 34, 35, 39], "how": [3, 11, 13, 23, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 43, 44], "samepl": 3, "can": [3, 13, 21, 23, 24, 25, 30, 32, 38, 43, 44], "distiguish": 3, "test_data": 3, "gse52588": [3, 22], "downsyndrom": [3, 21, 25], "set": [3, 4, 22, 30, 31, 38, 39], "index": [3, 4, 14], "merg": 3, "set_index": 3, "astyp": [3, 16], "str": [3, 28, 29, 30, 32, 33, 34, 35, 38], "inplac": 3, "merged_data": 3, "join": [3, 23], "category_ord": 3, "healthi": [3, 22], "palett": 3, "green": 3, "red": 3, "configur": [3, 31, 44], "10": [3, 11, 14, 16, 23, 43], "ax": [3, 8], "disease_st": 3, "width": 3, "0": [3, 4, 5, 8, 11, 12, 13, 14, 16, 19, 24, 33, 45], "order": 3, "stripplot": 3, "jitter": 3, "hue": 3, "dodg": 3, "fals": 3, "legend": [3, 16], "set_titl": 3, "despin": 3, "721": [3, 6, 45], "945": 3, "plot_down_syndrome_model": [3, 6, 45], "calcul": [4, 5, 14, 38], "multipl": [4, 24], "them": [4, 16, 18, 30, 44], "against": 4, "chronolog": [4, 23], "gse41169": [4, 22], "blood": [4, 7, 8, 11, 18, 21, 22, 44], "data_sourc": 4, "get": [4, 5, 11, 13, 14, 16, 23, 28, 29, 30, 38, 43], "note": [4, 14], "default": [4, 30, 31, 33, 38], "miss": [4, 29, 31, 32, 33, 34, 35, 38, 39], "To": [4, 43], "chang": [4, 22, 43], "behavior": [4, 43], "paramet": [4, 14, 28, 29, 30, 31, 32, 33, 34, 35, 37, 38], "when": 4, "horvath_result": 4, "predict": [4, 11, 14, 16, 21, 23, 38], "hannum_result": 4, "hannum": [4, 11, 21, 25], "phenoage_result": 4, "phenoag": [4, 21, 25], "actual_ag": 4, "plot_data": 4, "horvath": [4, 23, 43], "relplot": 4, "kind": 4, "axisgrid": 4, "facetgrid": 4, "object": [4, 14, 28, 30, 31, 32, 38], "0x713789f3e9b0": 4, "2": [4, 8, 11, 14, 24], "616": [4, 6, 45], "1587": [4, 6, 45], "plot_epigenetic_clocks_on_geo": [4, 6, 45], "gse134080": [5, 22], "rna_model": 5, "transcriptomicpredictionmodel": [5, 21, 25], "imputation_method": [5, 38], "none": [5, 11, 24, 28, 29, 30, 31, 32, 34, 35, 38], "scipi": 5, "stat": 5, "pearsonr": 5, "predicted_valu": 5, "true_valu": 5, "correl": [5, 14], "_": [5, 14], "f": [5, 14], "pearson": 5, "coeffici": [5, 14, 21], "set_them": 5, "style": 5, "whitegrid": 5, "regplot": [5, 11], "scatter_kw": 5, "alpha": [5, 14], "5": [5, 8, 14, 16, 24], "7025211774998293": 5, "739": [5, 6, 45], "406": [5, 6, 45], "plot_transcriptomic_clock": [5, 6, 45], "03": [6, 45], "14": [6, 11], "879": 6, "4": [6, 11, 45], "file": [6, 9, 12, 13, 17, 19, 21, 30, 31, 32, 44, 45], "auto_exampl": [6, 9, 12, 17, 19], "00_omics_biomark": [6, 45], "mem": [6, 9, 12, 17, 19, 45], "09": [6, 45], "00": [6, 9, 11, 12, 17, 19, 45], "02": [6, 23, 45], "01": [6, 9, 14, 45], "944": [6, 45], "surviv": [7, 18, 23], "curv": [7, 18], "standard": [7, 18, 38], "test": [7, 18, 44], "phenotyp": [7, 9, 18, 37, 45], "nhane": [7, 9, 18, 23, 37, 45], "exam": 8, "analys": [8, 23, 43], "load_nhan": 8, "year": [8, 21, 24, 37], "df": [8, 11, 14, 36, 37], "years_until_death": 8, "months_until_death": 8, "12": [8, 11, 22, 23], "hematologi": 8, "phenotypic_ag": 8, "": [8, 21, 23, 31, 38, 44], "lifelin": 8, "kaplanmeierfitt": 8, "biologically_old": 8, "subplot": 8, "group": [8, 37], "ix": 8, "t": [8, 14, 16], "e": 8, "is_dead": 8, "kmf": 8, "fit": [8, 14], "label": 8, "plot_survival_funct": 8, "text": [8, 16], "23": 8, "52222222222222": 8, "271": [8, 9, 45], "plot_nhan": [8, 9, 45], "01_composite_biomark": [9, 45], "build": [10, 12, 18, 38, 44, 45], "competit": [10, 12, 18, 38, 45], "an": [10, 12, 18, 22, 23, 28, 29, 30, 31, 32, 33, 35, 36, 37, 38, 44, 45], "exist": [10, 12, 18, 24, 28, 33, 38, 45], "explor": [10, 12, 18, 32, 44, 45], "train": [10, 12, 18, 30, 45], "elasticnet": [10, 12, 18, 30, 45], "challenge_data": [11, 13, 14], "boachallengedata": [11, 22], "option": [11, 30, 31, 32, 33, 34, 35, 38], "displai": 11, "max_column": 11, "gsm7866999": 11, "gsm7867305": 11, "gsm7867077": 11, "gsm7867431": 11, "gsm7867126": 11, "gsm7867000": 11, "cg00000029": [11, 24], "720084": 11, "453491": 11, "535564": 11, "487361": 11, "337215": 11, "062426": 11, "cg00000109": [11, 24], "938044": 11, "925149": 11, "949446": 11, "938523": 11, "940187": 11, "942090": 11, "cg00000155": [11, 24], "957093": 11, "959030": 11, "948919": 11, "966912": 11, "968026": 11, "970274": 11, "cg00000158": 11, "965191": 11, "957272": 11, "967278": 11, "971039": 11, "969745": 11, "969660": 11, "cg00000165": 11, "144061": 11, "122099": 11, "167346": 11, "156199": 11, "133038": 11, "641788": 11, "rs9292570": 11, "523412": 11, "980810": 11, "986609": 11, "525876": 11, "537722": 11, "538440": 11, "rs9363764": 11, "960815": 11, "038638": 11, "451884": 11, "978343": 11, "971838": 11, "973633": 11, "rs951295": 11, "971034": 11, "558011": 11, "553397": 11, "485287": 11, "526686": 11, "499980": 11, "rs966367": 11, "455920": 11, "963994": 11, "453094": 11, "455700": 11, "029342": 11, "026907": 11, "rs9839873": 11, "599735": 11, "595571": 11, "630871": 11, "576566": 11, "579195": 11, "973815": 11, "930659": 11, "row": [11, 14, 24, 32, 33, 34, 35, 36, 37], "500": [11, 22], "column": [11, 13, 14, 24, 32, 33, 34, 35, 36, 37], "gsm7867173": 11, "gsm7867127": 11, "gsm7867083": 11, "gsm7867095": 11, "gsm7867115": 11, "gsm7867385": 11, "alamartargetid": 11, "t10319": 11, "13": 11, "387778": 11, "948052": 11, "256551": 11, "062449": 11, "390582": 11, "587867": 11, "t10413": 11, "16": 11, "674015": 11, "15": 11, "992939": 11, "547946": 11, "827118": 11, "872686": 11, "584386": 11, "t10466": 11, "890963": 11, "556167": 11, "528756": 11, "683530": 11, "667331": 11, "483932": 11, "t10563": 11, "933342": 11, "323690": 11, "271350": 11, "993500": 11, "11": 11, "699095": 11, "860206": 11, "t10876": 11, "063506": 11, "429302": 11, "582464": 11, "978461": 11, "654607": 11, "931614": 11, "t8333": 11, "725785": 11, "537733": 11, "848917": 11, "729721": 11, "787099": 11, "055526": 11, "t8334": 11, "503408": 11, "103376": 11, "381374": 11, "070087": 11, "317681": 11, "314627": 11, "t8358": 11, "734120": 11, "238780": 11, "673970": 11, "103683": 11, "949114": 11, "794795": 11, "t8367": 11, "062422": 11, "291734": 11, "604654": 11, "803189": 11, "313347": 11, "830717": 11, "t8399": 11, "599597": 11, "642498": 11, "590723": 11, "073775": 11, "184558": 11, "040475": 11, "374": 11, "503": 11, "util": 11, "get_data_fil": 11, "read_csv": 11, "alamar_refer": 11, "csv": [11, 13, 24, 32], "uniprotid": 11, "target": 11, "proteinnam": 11, "t10034": 11, "o75888": 11, "tnfsf13": 11, "tnf": 11, "superfamili": 11, "o95760": 11, "il33": 11, "interleukin": 11, "33": 11, "t10412": 11, "p01258": 11, "calca": 11, "calcitonin": 11, "cleav": 11, "katacalcin": 11, "t10417": 11, "p01903": 11, "hla": 11, "dra": 11, "major": 11, "histocompat": 11, "complex": [11, 39, 44], "class": [11, 26, 27, 28, 29, 30, 31, 32, 38, 40, 42], "ii": 11, "dr": 11, "t10433": 11, "p02745": 11, "c1qa": 11, "complement": 11, "c1q": 11, "A": [11, 19, 28, 29, 30, 31, 32, 36, 37, 38], "chain": 11, "317": 11, "t8244": 11, "p35247": 11, "sftpd": 11, "pulmonari": 11, "surfact": 11, "associ": [11, 14, 22, 24, 32], "d": [11, 23], "318": 11, "t8246": 11, "p22303": 11, "ach": 11, "acetylcholinesteras": 11, "319": 11, "t8254": 11, "q8n474": 11, "sfrp1": 11, "secret": 11, "frizzl": 11, "relat": [11, 43], "320": 11, "t8355": 11, "p0dji8": 11, "saa1": 11, "serum": 11, "amyloid": 11, "321": 11, "t9441": 11, "o15240": 11, "vgf": 11, "nerv": 11, "growth": 11, "factor": 11, "induc": 11, "322": 11, "ethnic": 11, "race1": 11, "sex": [11, 21, 22, 24], "subject_id": 11, "tissu": [11, 21, 38], "gsm7866964": 11, "31": 11, "35": 11, "non": [11, 25], "hispan": 11, "white": 11, "boa1": 11, "gsm7866965": 11, "79": 11, "45": 11, "asian": 11, "boa2": 11, "gsm7866966": 11, "60": [11, 14], "42": [11, 14, 24], "boa3": 11, "gsm7866967": 11, "59": 11, "24": 11, "boa4": 11, "gsm7866968": 11, "22": 11, "41": 11, "boa5": 11, "p499": 11, "80": 11, "98": 11, "nan": [11, 24], "p500": 11, "89": [11, 24], "p501": 11, "92": 11, "p502": 11, "69": 11, "p503": 11, "68": [11, 22], "18": 11, "651": 11, "7": [11, 14, 24], "mortal": [11, 21, 23], "run_predict": 11, "prediction_dict": 11, "21": [11, 12, 45], "822423": 11, "17": 11, "303025": 11, "56": 11, "266752": 11, "52": 11, "803410": 11, "39": [11, 24], "660930": 11, "36": 11, "211078": 11, "29": [11, 24], "806215": 11, "32": 11, "821235": 11, "gsm7867457": 11, "51": 11, "285108": 11, "49": [11, 22], "308239": 11, "gsm7867136": 11, "37": 11, "539954": 11, "494127": 11, "gsm7867452": 11, "74": 11, "975617": 11, "639608": 11, "083565": 11, "906176": 11, "91": 11, "428559": 11, "870807": 11, "75": 11, "005273": 11, "77": 11, "630528": 11, "assum": [11, 24], "name": [11, 14, 21, 25, 38], "should": [11, 24, 37], "have": [11, 14, 18, 24, 43, 44], "regress": 11, "line": [11, 14, 23, 43], "ci": 11, "304": [11, 12, 45], "14385": 11, "plot_challenge_data": [11, 12, 45], "02_challenge_submiss": [12, 45], "challeng": [12, 13, 22, 23, 32, 44, 45], "14384": [12, 45], "submiss": [12, 38, 45], "simple_competiton_submiss": [12, 13, 45], "000": [12, 19, 45], "training_simple_model": [12, 14, 45], "you": [13, 14, 18, 23, 25, 43, 44], "warmup": [13, 14], "link": [13, 14], "here": [13, 14], "http": [13, 14, 23, 43], "www": [13, 14], "synaps": [13, 14, 23], "org": [13, 14, 23], "syn52966292": [13, 14], "wiki": [13, 14], "625231": [13, 14], "downloaded_data_file_path": [13, 14], "add": [13, 14], "path": [13, 14, 28, 30, 31, 32, 44], "from_methylation_matrix": [13, 14, 32], "score": 13, "best": 13, "so": 13, "let": 13, "save": 13, "correct": [13, 24], "format": [13, 22, 30, 31, 44], "now": [13, 44], "we": [13, 23, 24, 43, 44], "expect": [13, 43], "rename_axi": 13, "sampleid": [13, 14, 24], "renam": 13, "predictedag": [13, 14], "to_csv": 13, "lin_submiss": 13, "submit": [13, 14], "follow": [13, 18, 24], "instruct": [13, 18, 43], "below": [13, 33], "simpl": [14, 39], "narrow": 14, "what": [14, 25], "site": [14, 24, 32, 33, 34, 35], "take": [14, 44], "long": 14, "numpi": 14, "np": 14, "sklearn": 14, "linear_model": 14, "linearregress": 14, "extract": [14, 44], "transpos": 14, "valu": [14, 28, 29, 33, 34, 35, 38, 39], "sampl": [14, 22, 24, 32, 33, 34, 35, 44], "cpg": [14, 24, 33, 34, 35], "bootstrap": 14, "featur": [14, 43], "select": [14, 38], "n_bootstrap": 14, "20": 14, "threshold": [14, 33], "05": 14, "store": [14, 22, 28, 29, 30], "count": 14, "each": [14, 24, 32, 36, 37], "deem": 14, "signific": 14, "cpg_count": 14, "zero": 14, "shape": 14, "begin": 14, "iter": 14, "rang": [14, 24], "replac": [14, 33], "sample_idx": 14, "random": 14, "choic": 14, "size": [14, 28], "x_sampl": 14, "y_sampl": 14, "identifi": [14, 30, 32, 44], "magnitud": [14, 44], "significant_cpg": 14, "where": [14, 18, 23, 32, 36, 37, 44], "ab": 14, "coef_": 14, "determin": [14, 25, 33], "stabl": 14, "stable_cpg_sit": 14, "stable_cpg_nam": 14, "tolist": 14, "more": [14, 23, 24, 39, 43, 44], "than": [14, 24], "model_select": 14, "train_test_split": 14, "top_sites_df": 14, "x_train": 14, "x_test": 14, "y_train": 14, "y_test": 14, "test_siz": 14, "random_st": 14, "regular": 14, "strength": 14, "l1_ratio": 14, "mix": 14, "between": [14, 23, 24], "l1": 14, "l2": 14, "lasso": 14, "ridg": 14, "metric": 14, "mean_squared_error": 14, "max_it": 14, "10000": 14, "evalu": 14, "y_pred": 14, "mse": 14, "squar": 14, "error": [14, 28, 43], "edgecolor": 14, "w": 14, "linewidth": 14, "min": 14, "max": 14, "k": [14, 23], "lw": 14, "refer": [14, 22, 23, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 43], "actual": [14, 43], "both": [14, 32, 33, 44], "linestyl": 14, "4f": 14, "absolut": 14, "mae": 14, "pruned_data": 14, "fillna": 14, "challenge_result": 14, "predicted_age_df": 14, "dataset": [15, 18, 22, 33, 34, 35, 38, 39, 44], "known": [15, 18], "cell": [15, 18, 21, 22], "proport": [15, 18, 21], "type": [16, 22, 28, 29, 32, 33, 34, 35, 36, 37, 38], "epic_facs_salas_18": 16, "gse112618": [16, 22], "epic_facs_cell_prop": 16, "deconvolutebloodep": [16, 21, 25], "set_styl": 16, "tick": 16, "neutrophils_proport": 16, "float": [16, 28, 33], "loc": 16, "neutrophil": 16, "monocytes_proport": 16, "monocyt": 16, "nk_proport": 16, "nk_cell": 16, "bcell_proport": 16, "b_cell": 16, "cd4t_proport": 16, "cd4_t_cell": 16, "cd8t_proport": 16, "cd8_t_cell": 16, "nk": 16, "b": 16, "cd4": 16, "cd8": 16, "valid": [16, 22], "934": [16, 17, 45], "plot_blood_deconvolut": [16, 17, 45], "03_deconvolut": [17, 45], "deconvolut": [17, 32, 45], "If": [18, 23, 24, 25, 30, 31, 32, 33, 38, 43, 44], "want": [18, 44], "make": 18, "sure": 18, "directori": [18, 28], "write": [18, 44], "permiss": 18, "copi": [18, 32], "instal": 18, "biolean": 18, "manual": 18, "auto_examples_python": 18, "zip": 18, "auto_examples_jupyt": 18, "n": [19, 23], "support": [21, 22, 23, 30, 37, 38, 44], "number": [21, 22, 24, 37], "easili": [21, 23, 30, 43], "speci": [21, 38], "2013": 21, "human": [21, 22, 44], "multi": [21, 23, 44], "paper": [21, 23, 25, 43], "lin": [21, 25], "2016": 21, "2018": 21, "yingcausag": [21, 25], "2022": 21, "yingdamag": [21, 25], "yingadaptag": [21, 25], "horvathv2": [21, 25], "skin": 21, "pedb": [21, 25], "2019": 21, "buccal": 21, "zhang_10": [21, 25], "risk": 21, "dunedinpoam38": [21, 25], "2020": 21, "rate": 21, "dunedinpac": [21, 23, 25, 38, 43], "grimagev1": [21, 25], "adjust": 21, "grimagev2": [21, 25], "alcoholmccartnei": [21, 25], "alcohol": 21, "consumpt": 21, "dnamtl": [21, 25], "adipos": 21, "telomer": 21, "length": [21, 24], "hrsinchphenoag": [21, 25], "knight": [21, 25], "cord": [21, 22], "gestat": 21, "leecontrol": [21, 25], "placenta": 21, "sexestim": [21, 25], "2021": 21, "leerefinedrobust": [21, 25], "leerobust": [21, 25], "smokingmccartnei": [21, 25], "smoke": 21, "statu": [21, 25], "deconvoluteblood450k": [21, 25], "2024": [21, 22, 23], "twelvecelldeconvolutebloodep": [21, 25], "bmi_mccartnei": [21, 25], "bmi": 21, "educationmccartnei": [21, 25], "educ": 21, "attain": 21, "totalcholesterolmccartnei": [21, 25], "cholesterol": 21, "hdlcholesterolmccartnei": [21, 25], "hdl": 21, "ldlcholesterolmccartnei": [21, 25], "ldl": 21, "remnant": 21, "bodyfatmccartnei": [21, 25], "percentag": [21, 32], "bodi": 21, "fat": 21, "bmi_re": [21, 25], "prostatecancerkirbi": [21, 25], "2017": 21, "prostat": 21, "cancer": 21, "hepatoxu": [21, 25], "circul": 21, "hepatocellular": 21, "carcinoma": 21, "cvd_westerman": [21, 25], "coronari": 21, "heart": [21, 23, 36, 43], "diseas": [21, 24, 44], "ad_bahado": [21, 25], "singh": [21, 25], "alzheim": 21, "depressionbarbu": [21, 25], "depress": 21, "2015": 21, "id": [22, 24, 31], "present": 22, "fac": 22, "optim": [22, 44], "illuminaep": 22, "ye": 22, "gse110554": 22, "flowsort": 22, "epic": 22, "ba": 22, "genom": [22, 32], "wide": 22, "profil": 22, "reveal": 22, "quantit": 22, "view": 22, "o": 22, "illumina450k": 22, "656": 22, "gse19711": 22, "unit": 22, "kingdom": 22, "ovar": 22, "illumina27k": 22, "540": 22, "No": [22, 24, 38], "methylom": 22, "analysi": [22, 24], "me": 22, "329": 22, "gse42861": 22, "differenti": 22, "rheumatoid": 22, "arthriti": 22, "689": 22, "dutch": 22, "popul": 22, "95": 22, "gse51032": 22, "itali": 22, "hugef": 22, "845": 22, "gse73103": 22, "mani": [22, 23, 43, 44], "obes": 22, "snp": 22, "strongli": 22, "met": 22, "355": 22, "gse69270": 22, "middl": 22, "indi": 22, "184": 22, "gse36054": 22, "children": 22, "192": 22, "gse64495": 22, "sever": [22, 23], "113": 22, "gse30870": 22, "newborn": 22, "nonagenarian": 22, "40": 22, "identif": 22, "signatur": [22, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38], "87": 22, "gse157131": 22, "peripher": 22, "leukocyt": 22, "fro": 22, "946": 22, "gse132203": 22, "gradi": 22, "trauma": 22, "project": 22, "795": 22, "rnaseq": 22, "whole": 22, "500fg": 22, "cohort": 22, "illuminahiseq2500": 22, "100": 22, "jenag": 22, "deep": [22, 32], "sequenc": 22, "endogen": 22, "mrna": 22, "homo": 22, "sapien": 22, "fo": 22, "62": 22, "gse167998": 22, "bloodextend": 22, "enhanc": [22, 38], "l": 22, "gse180683": 22, "cd4t": 22, "cd8t": 22, "naiv": 22, "76": 22, "gse180970": 22, "umbil": 22, "artifici": 22, "mixtur": 22, "gse182379": 22, "enha": 22, "enabl": [23, 43], "easi": [23, 43, 44], "versatil": [23, 43], "It": [23, 24, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 43], "provid": [23, 31, 39, 43, 44], "tool": [23, 43, 44], "load": [23, 30, 31, 32, 43, 44], "publicli": [23, 43], "like": [23, 24, 31, 43, 44], "gene": [23, 43], "express": [23, 43], "omnibu": [23, 43], "nation": [23, 37, 43], "health": [23, 37, 43, 44], "nutrit": [23, 37, 43], "examin": [23, 37], "survei": [23, 37, 43], "framingham": [23, 36, 43], "studi": [23, 36, 43], "also": [23, 43, 44], "contain": [23, 24, 31, 32, 33, 35, 43], "implement": [23, 28, 29, 44], "common": [23, 24, 43], "other": [23, 24, 25, 33, 43], "onli": [23, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 43], "few": [23, 43], "read": [23, 43], "about": [23, 31, 36, 37, 43, 44], "our": [23, 43, 44], "develop": [23, 43, 44], "organ": [23, 30, 31, 32], "individu": [23, 36, 37], "especi": 23, "methuselah": 23, "foundat": [23, 25], "volo": 23, "host": 23, "2025": 23, "seri": [23, 24, 33, 35], "platform": 23, "particip": 23, "ask": 23, "morbid": 23, "award": 23, "200k": 23, "pleas": [23, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 43, 44], "cite": 23, "preprint": 23, "m": 23, "mart\u00ed": 23, "nez": 23, "c": 23, "poganik": 23, "j": 23, "r": 23, "gladyshev": 23, "2023": 23, "open": [23, 43, 44], "biorxiv": 23, "doi": 23, "1101": 23, "569722": 23, "quickstart": 23, "start": [23, 43], "roadmap": [23, 44], "see": [23, 44], "head": 23, "contribut": [23, 44], "biolog": [23, 38], "relationship": 23, "variou": [23, 38, 39], "point": 23, "interest": [25, 44], "commerci": 25, "tabl": 25, "help": [25, 44], "do": [24, 25], "who": [], "talk": 44, "page": [28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38], "document": [28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 43], "explain": [28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38], "user": [28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38], "guid": [28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38], "big": [28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 44], "pictur": [28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38], "max_size_gb": 28, "serial": 28, "local": 28, "basic": 28, "lru": 28, "least": 28, "recent": 28, "cleanup": 28, "maintain": [28, 44], "within": [28, 38], "specifi": [28, 30, 32, 38], "maximum": 28, "limit": 28, "__init__": [28, 29, 30, 31, 32, 38], "initi": [28, 30, 31, 32, 38], "instanc": [28, 30, 31, 32, 38], "filesystem": 28, "storag": [24, 28], "gigabyt": 28, "kei": [28, 29, 31], "retriev": [28, 29, 30, 38], "item": [28, 29, 31, 44], "its": [28, 30, 38], "doe": [24, 28, 29], "occur": 28, "ani": [24, 28, 29, 31, 43, 44], "under": [28, 29], "manag": [28, 30, 38], "stai": 28, "clear": [28, 29], "remov": [28, 29], "content": [24, 28], "specif": [24, 28, 29], "entri": [24, 28, 29], "noth": 29, "alwai": 29, "simul": 29, "library_fil": 30, "cach": [30, 31, 44], "collect": [30, 38, 44], "respons": 30, "new": [30, 32], "ad": [24, 30, 44], "runtim": 30, "current": [30, 44], "mechan": [30, 31], "adher": [30, 31], "interfac": [30, 31, 43], "modul": [30, 31, 39], "load_sourc": 30, "given": 30, "append": 30, "source_id": 30, "found": [30, 38], "otherwis": 30, "lookup_sourc": 30, "look": 30, "up": [30, 31, 44], "filter": [30, 32, 38], "list": [30, 32, 33, 34, 35], "match": [24, 30, 38], "criteria": [30, 38], "source_definit": 31, "repres": [24, 31, 32, 36, 37], "singl": [24, 31, 44], "encapsul": 31, "detail": [31, 32, 43], "rais": [31, 33, 38], "valueerror": [31, 33, 38], "requir": [31, 33, 44], "field": [31, 32], "parser": 31, "dure": 31, "pars": [31, 44], "dictionari": [31, 38], "typic": 31, "yaml": 31, "check": [31, 43], "essenti": 31, "attribut": 31, "handl": [24, 31, 38], "dict": [31, 38], "properti": 31, "must": [24, 31], "summari": [31, 32], "nocach": 31, "input": [31, 37, 38], "required_field": 31, "geograph": 31, "rtype": 31, "rna": 32, "protein": 32, "save_to_hdf5": [], "filepath": [], "complevel": [], "complib": [], "zlib": [], "hdf5": [], "compress": [], "save_to_csv": [], "base_filepath": [], "separ": [24, 44], "load_from_hdf5": [], "quality_report": 32, "qualiti": [24, 32], "control": [24, 32], "report": 32, "section": 32, "per": 32, "qualityreport": 32, "classmethod": 32, "matrix": 32, "either": 32, "directli": [32, 44], "union": 32, "cpg_sourc": 33, "required_cpg": 33, "extern": [33, 35], "while": [24, 33], "need": [33, 44], "final": 33, "strategi": [33, 38, 39], "fill": [33, 34, 35], "certain": 33, "cpgs_to_imput": [34, 35], "itself": 34, "cpg_averag": 35, "measur": [24, 36, 37], "Not": 37, "come": 37, "two": [24, 37], "pass": 37, "later": 37, "2010": 37, "2012": 37, "model_definit": 38, "produc": 38, "search": 38, "definit": 38, "model_build": 38, "deconvolutionmodel": 38, "bound": 38, "from_definit": 38, "grimagemodel": 38, "linearmethylationmodel": 38, "linearmodel": 38, "lineartranscriptomicmodel": 38, "sexestimationmodel": 38, "request": [38, 43], "possibli": 38, "keyerror": 38, "invalid": 38, "appli": 38, "dunedin": 38, "hybrid_imput": 38, "gold": 38, "sesame_450k": 38, "sesam": [24, 38], "450k": 38, "These": 39, "allow": 39, "hybrid": 39, "examim": 43, "implemen": 43, "prereleas": 43, "version": 43, "There": 43, "mai": 43, "bug": 43, "subject": 43, "github": 43, "com": 43, "bio": 43, "homepag": 43, "io": 43, "pip": 43, "wa": 43, "correctli": 43, "out": [43, 44], "some": [43, 44], "team": 43, "ha": [24, 43, 44], "answer": 43, "question": 43, "discuss": 43, "find": 43, "replic": 43, "setup": 43, "repo": 43, "omic": 44, "revolut": 44, "transform": 44, "biologi": 44, "wealth": 44, "potenti": 44, "care": 44, "could": 44, "offer": 44, "comprehens": 44, "insight": 44, "one": 44, "surrog": 44, "endpoint": 44, "revers": 44, "therapeut": 44, "howev": [24, 44], "obstruct": 44, "access": [32, 44], "address": 44, "issu": 44, "modern": 44, "recogn": 44, "ve": 44, "adopt": 44, "scienc": 44, "invit": 44, "mission": 44, "primarili": 44, "25": [24, 44], "compris": 44, "thousand": 44, "ambit": 44, "achiev": 44, "larg": 44, "small": 44, "task": 44, "vari": 44, "skill": 44, "level": [24, 44], "knowledg": 44, "right": 44, "improv": 44, "better": 44, "differ": [32, 44], "environ": 44, "speed": 44, "10x": 44, "integr": 44, "unifi": 44, "structur": 44, "know": 44, "process": [24, 44], "partial": 44, "autom": 44, "review": 44, "genet": 44, "built": 44, "share": 44, "reach": 44, "proprietari": 44, "would": 44, "api": 44, "public": 44, "hear": 44, "privat": 44, "think": 44, "wai": 44, "fund": 44, "financi": 44, "monei": 44, "mainten": 44, "prize": 44, "pot": 44, "further": 44, "expans": 44, "contriubt": 44, "discord": 44, "server": 44, "38": 45, "388": 45, "contact": 25, "author": 25, "free": 25, "For": 25, "cosmet": 25, "life": 25, "insur": 25, "applic": 25, "ucla": 25, "tdg": 25, "regard": 25, "licens": 25, "focu": 32, "facilit": 32, "describ": 24, "consist": 24, "focus": 24, "transmiss": 24, "without": 24, "mandat": 24, "form": 24, "veri": 24, "commonli": 24, "thei": 24, "problem": 24, "inclus": 24, "even": 24, "As": 24, "grown": 24, "larger": 24, "longer": 24, "instead": 24, "supplementari": 24, "been": 24, "pre": 24, "batch": 24, "normal": 24, "recommend": 24, "illumina": 24, "comma": 24, "same": 24, "correspond": 24, "first": 24, "header": 24, "publish": 24, "exactli": 24, "1000": 24, "beta": 24, "decim": 24, "precis": 24, "round": 24, "three": 24, "place": 24, "integ": 24, "femal": 24, "male": 24, "permit": 24, "unknown": 24, "gsm3074480": 24, "gsm3074481": 24, "gsm3074482": 24, "gsm3074483": 24, "395": 24, "499": 24, "423": 24, "821": 24, "868": 24, "877": 24, "841": 24, "878": 24, "894": 24, "822": 24, "state": 24, "covid": 24, "encod": 24, "soft": 24, "ncbi": 24, "tip": 24, "real": [], "mgb": [], "ftp": [], "nlm": [], "nih": [], "gov": [], "gse246nnn": [], "gse246337": [], "suppl": [], "5fbeta": [], "2ecsv": [], "2egz": [], "cpgsite": 24}, "objects": {"biolearn": [[26, 0, 0, "-", "cache"], [27, 0, 0, "-", "data_library"], [39, 0, 0, "-", "imputation"], [41, 0, 0, "-", "load"], [42, 0, 0, "-", "model_gallery"]], "biolearn.cache": [[28, 1, 1, "", "LocalFolderCache"], [29, 1, 1, "", "NoCache"]], "biolearn.cache.LocalFolderCache": [[28, 2, 1, "", "__init__"], [28, 2, 1, "", "clear"], [28, 2, 1, "", "get"], [28, 2, 1, "", "remove"], [28, 2, 1, "", "store"]], "biolearn.cache.NoCache": [[29, 2, 1, "", "__init__"], [29, 2, 1, "", "clear"], [29, 2, 1, "", "get"], [29, 2, 1, "", "remove"], [29, 2, 1, "", "store"]], "biolearn.data_library": [[30, 1, 1, "", "DataLibrary"], [31, 1, 1, "", "DataSource"], [32, 1, 1, "", "GeoData"]], "biolearn.data_library.DataLibrary": [[30, 2, 1, "", "__init__"], [30, 2, 1, "", "get"], [30, 2, 1, "", "load_sources"], [30, 2, 1, "", "lookup_sources"]], "biolearn.data_library.DataSource": [[31, 3, 1, "", "REQUIRED_FIELDS"], [31, 2, 1, "", "__init__"], [31, 2, 1, "", "load"]], "biolearn.data_library.GeoData": [[32, 2, 1, "", "__init__"], [32, 2, 1, "", "copy"], [32, 3, 1, "", "dnam"], [32, 2, 1, "", "from_methylation_matrix"], [32, 3, 1, "", "metadata"], [32, 2, 1, "", "quality_report"]], "biolearn.imputation": [[33, 4, 1, "", "hybrid_impute"], [34, 4, 1, "", "impute_from_average"], [35, 4, 1, "", "impute_from_standard"]], "biolearn.load": [[36, 4, 1, "", "load_fhs"], [37, 4, 1, "", "load_nhanes"]], "biolearn.model_gallery": [[38, 1, 1, "", "ModelGallery"]], "biolearn.model_gallery.ModelGallery": [[38, 2, 1, "", "__init__"], [38, 2, 1, "", "get"], [38, 3, 1, "", "model_builders"], [38, 2, 1, "", "search"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:attribute", "4": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "attribute", "Python attribute"], "4": ["py", "function", "Python function"]}, "titleterms": {"team": 0, "core": 0, "develop": 0, "other": [0, 21], "contributor": 0, "partner": 0, "organ": 0, "omic": [1, 18], "biomark": [1, 7, 18, 44], "exampl": [1, 7, 10, 15, 16, 18, 23, 24, 30, 31, 32, 35, 37, 38], "perform": 2, "custom": 2, "imput": [2, 33, 34, 35, 39], "load": [2, 3, 4, 5, 8, 11, 13, 14, 16, 36, 37, 41], "methyl": [2, 3, 4, 11, 16, 24], "data": [2, 3, 4, 5, 8, 11, 13, 14, 22, 24, 39, 41], "from": [2, 3, 4, 5, 11], "geo": [2, 3, 4, 5, 16, 22], "dataset": [2, 4, 16], "us": [2, 3, 4, 5, 13, 14, 16, 30, 31, 32, 35, 37, 38], "refer": [2, 11, 24, 40], "up": [2, 3, 4, 5, 11, 13, 14], "target": 2, "run": [2, 3, 4, 5, 11, 14], "now": [2, 3, 4, 5], "clock": [2, 4, 5, 21], "predict": [2, 3, 5, 13], "befor": 2, "after": 2, "visual": 2, "comparison": 2, "ag": [2, 4, 5, 8, 13], "you": [2, 11], "can": [2, 11], "also": [2, 11], "build": [2, 13], "an": [2, 13, 14], "decor": 2, "bundl": 2, "get": [2, 3], "same": 2, "result": [2, 4, 5, 14], "function": 2, "directli": 2, "your": 2, "down": 3, "syndrom": 3, "epigenet": [3, 4], "plot": [3, 4, 5, 8, 14], "first": [3, 4, 5], "some": [3, 4, 5, 11, 14], "librari": [3, 4, 5], "model": [3, 11, 13, 14, 16, 21, 25], "score": 3, "final": [3, 4], "gener": 3, "boxplot": 3, "show": [3, 8], "power": 3, "three": 4, "differ": 4, "produc": 4, "extract": 4, "metadata": [4, 11, 24], "seaborn": [4, 5], "transcriptom": 5, "against": 5, "chronolog": [5, 8], "comput": [6, 9, 12, 17, 19, 45], "time": [6, 9, 12, 17, 19, 45], "composit": [7, 18], "phenotyp": 8, "nhane": 8, "2010": 8, "calcul": 8, "biolog": 8, "base": 8, "phenotypicag": 8, "relat": 8, "between": 8, "surviv": 8, "curv": 8, "peopl": 8, "acceler": 8, "older": 8, "v": [8, 24], "deceler": 8, "younger": 8, "challeng": [10, 11, 14, 18], "submiss": [10, 13, 14, 18], "explor": 11, "competit": [11, 13, 14], "The": 11, "ha": 11, "proteom": 11, "learn": 11, "more": 11, "about": 11, "what": 11, "protein": 11, "identifi": 11, "our": [11, 14], "overlap": 11, "while": 11, "doe": 11, "all": 11, "i": [11, 14, 24], "combin": 11, "easili": 11, "sever": 11, "them": 11, "we": 11, "compar": [11, 16], "output": [11, 14], "two": 11, "exist": 13, "lin": 13, "train": 14, "elasticnet": 14, "seper": 14, "test": 14, "set": 14, "elast": 14, "net": 14, "see": 14, "how": 14, "good": 14, "through": 14, "save": 14, "file": [14, 24], "deconvolut": [15, 16, 18], "blood": 16, "measur": 16, "epic": 16, "platform": 16, "estim": 16, "cell": 16, "proport": 16, "known": 16, "fac": 16, "fluoresc": 16, "activ": 16, "sort": 16, "sourc": 22, "biolearn": [23, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 41, 42, 43, 44], "featur": 23, "ip": 25, "usag": 25, "cach": [26, 28, 29], "data_librari": [27, 30, 31, 32], "datalibrari": [27, 30], "localfoldercach": 28, "nocach": 29, "datasourc": 31, "geodata": 32, "hybrid_imput": 33, "impute_from_averag": 34, "impute_from_standard": 35, "load_fh": 36, "load_nhan": 37, "model_galleri": [38, 42], "modelgalleri": [38, 42], "util": [39, 41], "api": 40, "quickstart": 43, "import": 43, "link": 43, "requir": 43, "instal": 43, "discord": 43, "server": 43, "issu": 43, "contribut": 43, "futur": 44, "join": 44, "u": 44, "dna": 24, "arrai": 24, "standard": 24, "2410": 24, "why": 24, "thi": 24, "need": 24, "preprocess": 24, "matrix": 24, "format": 24, "valu": 24, "field": 24, "addit": 24, "dnam": []}, "envversion": {"sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinxcontrib.bibtex": 9, "sphinx": 60}, "alltitles": {"Team": [[0, "team"]], "Core developers": [[0, "core-developers"]], "Other contributors": [[0, "other-contributors"]], "Partner Organizations": [[0, "partner-organizations"]], "Omics Biomarker Examples": [[1, "omics-biomarker-examples"], [18, "omics-biomarker-examples"]], "Performing custom imputations": [[2, "performing-custom-imputations"]], "Load methylation data from a GEO dataset to use as a reference for imputation": [[2, "load-methylation-data-from-a-geo-dataset-to-use-as-a-reference-for-imputation"]], "Load up a target dataset and run the imputation": [[2, "load-up-a-target-dataset-and-run-the-imputation"]], "Now run clock predictions on the dataset before and after": [[2, "now-run-clock-predictions-on-the-dataset-before-and-after"]], "Visualize the comparison of age predictions": [[2, "visualize-the-comparison-of-age-predictions"]], "You can also build an imputation decorator to bundle with the clock": [[2, "you-can-also-build-an-imputation-decorator-to-bundle-with-the-clock"]], "You get the same results as running the function directly on your dataset": [[2, "you-get-the-same-results-as-running-the-function-directly-on-your-dataset"]], "Down Syndrome Epigenetic Plotting": [[3, "down-syndrome-epigenetic-plotting"]], "First load up some methylation data from GEO using the data library": [[3, "first-load-up-some-methylation-data-from-geo-using-the-data-library"], [4, "first-load-up-some-methylation-data-from-geo-using-the-data-library"]], "Now run the down syndrome model on the data to get a score": [[3, "now-run-the-down-syndrome-model-on-the-data-to-get-a-score"]], "Finally generate a boxplot to show the predictive power": [[3, "finally-generate-a-boxplot-to-show-the-predictive-power"]], "\u201cEpigenetic Clocks\u201d in GEO Data": [[4, "epigenetic-clocks-in-geo-data"]], "Now run three different clocks on the dataset to produce epigenetic clock ages": [[4, "now-run-three-different-clocks-on-the-dataset-to-produce-epigenetic-clock-ages"]], "Finally extract the age data from the metadata from GEO and plot the results using seaborn": [[4, "finally-extract-the-age-data-from-the-metadata-from-geo-and-plot-the-results-using-seaborn"]], "\u201cTranscriptomic Clock\u201d using GEO Data": [[5, "transcriptomic-clock-using-geo-data"]], "First load up some transcriptomic data from GEO using the data library": [[5, "first-load-up-some-transcriptomic-data-from-geo-using-the-data-library"]], "Now run the transcriptomic clock to predict age": [[5, "now-run-the-transcriptomic-clock-to-predict-age"]], "Plot the results against the chronological age using seaborn": [[5, "plot-the-results-against-the-chronological-age-using-seaborn"]], "Computation times": [[6, "computation-times"], [9, "computation-times"], [12, "computation-times"], [17, "computation-times"], [19, "computation-times"], [45, "computation-times"]], "Composite Biomarkers Examples": [[7, "composite-biomarkers-examples"], [18, "composite-biomarkers-examples"]], "\u201cPhenotypic Ages\u201d in NHANES Data": [[8, "phenotypic-ages-in-nhanes-data"]], "Loading NHANES 2010 data": [[8, "loading-nhanes-2010-data"]], "Calculate \u201cbiological age\u201d based on PhenotypicAge": [[8, "calculate-biological-age-based-on-phenotypicage"]], "Show relation between biological age and chronological age": [[8, "show-relation-between-biological-age-and-chronological-age"]], "Plot survival curve for people with accelerated aging (older biological age) vs decelerated aging (younger biological age)": [[8, "plot-survival-curve-for-people-with-accelerated-aging-older-biological-age-vs-decelerated-aging-younger-biological-age"]], "Challenge Submission Examples": [[10, "challenge-submission-examples"], [18, "challenge-submission-examples"]], "Exploring the Challenge Data": [[11, "exploring-the-challenge-data"]], "Loading up the data for the competition": [[11, "loading-up-the-data-for-the-competition"], [13, "loading-up-the-data-for-the-competition"], [14, "loading-up-the-data-for-the-competition"]], "The challenge data has methylation data": [[11, "the-challenge-data-has-methylation-data"]], "The challenge data also has proteomic data": [[11, "the-challenge-data-also-has-proteomic-data"]], "You can learn more about what the protein identifies in our reference": [[11, "you-can-learn-more-about-what-the-protein-identifies-in-our-reference"]], "Some of the data overlaps while some does not but all the metadata is combined": [[11, "some-of-the-data-overlaps-while-some-does-not-but-all-the-metadata-is-combined"]], "You can easily run several models on them": [[11, "you-can-easily-run-several-models-on-them"]], "We can then compare the output from the two models": [[11, "we-can-then-compare-the-output-from-the-two-models"]], "Deconvolution Examples": [[15, "deconvolution-examples"], [18, "deconvolution-examples"]], "\u201cDeconvolution Example\u201d": [[16, "deconvolution-example"]], "Load GEO blood methylation dataset measured on EPIC platform": [[16, "load-geo-blood-methylation-dataset-measured-on-epic-platform"]], "Estimate cell proportions using EPIC deconvolution model": [[16, "estimate-cell-proportions-using-epic-deconvolution-model"]], "Compare to known cell proportions measured with FACS (fluorescence-activated cell sorting)": [[16, "compare-to-known-cell-proportions-measured-with-facs-fluorescence-activated-cell-sorting"]], "Examples": [[18, "examples"]], "Biolearn": [[23, "biolearn"], [43, "biolearn"]], "Featured examples": [[23, "featured-examples"]], "biolearn.cache: Cache": [[26, "module-biolearn.cache"]], "biolearn.cache.LocalFolderCache": [[28, "biolearn-cache-localfoldercache"]], "biolearn.cache.NoCache": [[29, "biolearn-cache-nocache"]], "biolearn.imputation.hybrid_impute": [[33, "biolearn-imputation-hybrid-impute"]], "biolearn.imputation.impute_from_average": [[34, "biolearn-imputation-impute-from-average"]], "biolearn.imputation.impute_from_standard": [[35, "biolearn-imputation-impute-from-standard"]], "Examples using biolearn.imputation.impute_from_standard": [[35, "examples-using-biolearn-imputation-impute-from-standard"]], "biolearn.load.load_fhs": [[36, "biolearn-load-load-fhs"]], "biolearn.load.load_nhanes": [[37, "biolearn-load-load-nhanes"]], "Examples using biolearn.load.load_nhanes": [[37, "examples-using-biolearn-load-load-nhanes"]], "biolearn.imputation: Data Imputation Utilities": [[39, "biolearn-imputation-data-imputation-utilities"]], "API References": [[40, "api-references"]], "biolearn.load: Data Loading Utilities": [[41, "module-biolearn.load"]], "biolearn.model_gallery: ModelGallery": [[42, "module-biolearn.model_gallery"]], "Quickstart": [[43, "quickstart"]], "Important links": [[43, "important-links"]], "Requirements": [[43, "requirements"]], "Install": [[43, "install"]], "Discord server": [[43, "discord-server"]], "Issues": [[43, "issues"]], "Contributing": [[43, "contributing"]], "Biolearn and the Future of Biomarkers": [[44, "biolearn-and-the-future-of-biomarkers"]], "Join Us": [[44, "join-us"]], "biolearn.model_gallery.ModelGallery": [[38, "biolearn-model-gallery-modelgallery"]], "Examples using biolearn.model_gallery.ModelGallery": [[38, "examples-using-biolearn-model-gallery-modelgallery"]], "Building a competition submission using an existing model": [[13, "building-a-competition-submission-using-an-existing-model"]], "Use the Lin model to predict the age": [[13, "use-the-lin-model-to-predict-the-age"]], "Training an ElasticNet model": [[14, "training-an-elasticnet-model"]], "Load up some training data": [[14, "load-up-some-training-data"]], "Seperate data into training and test sets": [[14, "seperate-data-into-training-and-test-sets"]], "Train a model using elastic net": [[14, "train-a-model-using-elastic-net"]], "Plot the results to see how good our model is": [[14, "plot-the-results-to-see-how-good-our-model-is"]], "Run the challenge data through the model": [[14, "run-the-challenge-data-through-the-model"]], "Save the results as an output file for submission": [[14, "save-the-results-as-an-output-file-for-submission"]], "DNA Methylation Array Data Standard V-2410": [[24, "dna-methylation-array-data-standard-v-2410"]], "Why is this standard needed?": [[24, "why-is-this-standard-needed"]], "Preprocessing": [[24, "preprocessing"]], "Methylation Matrix File Standard": [[24, "methylation-matrix-file-standard"]], "File Format": [[24, "file-format"], [24, "id1"]], "Data Values": [[24, "data-values"]], "Metadata File Standard": [[24, "metadata-file-standard"]], "Standard Metadata Fields": [[24, "standard-metadata-fields"]], "Additional Metadata Fields": [[24, "additional-metadata-fields"]], "Example Files": [[24, "example-files"]], "References": [[24, "references"]], "biolearn.data_library: DataLibrary": [[27, "module-biolearn.data_library"]], "biolearn.data_library.DataLibrary": [[30, "biolearn-data-library-datalibrary"]], "Examples using biolearn.data_library.DataLibrary": [[30, "examples-using-biolearn-data-library-datalibrary"]], "biolearn.data_library.DataSource": [[31, "biolearn-data-library-datasource"]], "Examples using biolearn.data_library.DataSource": [[31, "examples-using-biolearn-data-library-datasource"]], "biolearn.data_library.GeoData": [[32, "biolearn-data-library-geodata"]], "Examples using biolearn.data_library.GeoData": [[32, "examples-using-biolearn-data-library-geodata"]], "Clocks and Other Models": [[21, "clocks-and-other-models"]], "GEO Data Sources": [[22, "geo-data-sources"]], "Model IP and Usage": [[25, "model-ip-and-usage"]]}, "indexentries": {"biolearn.cache": [[26, "module-biolearn.cache"]], "module": [[26, "module-biolearn.cache"], [27, "module-biolearn.data_library"], [39, "module-biolearn.imputation"], [41, "module-biolearn.load"], [42, "module-biolearn.model_gallery"]], "biolearn.data_library": [[27, "module-biolearn.data_library"]], "localfoldercache (class in biolearn.cache)": [[28, "biolearn.cache.LocalFolderCache"]], "__init__() (biolearn.cache.localfoldercache method)": [[28, "biolearn.cache.LocalFolderCache.__init__"]], "clear() (biolearn.cache.localfoldercache method)": [[28, "biolearn.cache.LocalFolderCache.clear"]], "get() (biolearn.cache.localfoldercache method)": [[28, "biolearn.cache.LocalFolderCache.get"]], "remove() (biolearn.cache.localfoldercache method)": [[28, "biolearn.cache.LocalFolderCache.remove"]], "store() (biolearn.cache.localfoldercache method)": [[28, "biolearn.cache.LocalFolderCache.store"]], "nocache (class in biolearn.cache)": [[29, "biolearn.cache.NoCache"]], "__init__() (biolearn.cache.nocache method)": [[29, "biolearn.cache.NoCache.__init__"]], "clear() (biolearn.cache.nocache method)": [[29, "biolearn.cache.NoCache.clear"]], "get() (biolearn.cache.nocache method)": [[29, "biolearn.cache.NoCache.get"]], "remove() (biolearn.cache.nocache method)": [[29, "biolearn.cache.NoCache.remove"]], "store() (biolearn.cache.nocache method)": [[29, "biolearn.cache.NoCache.store"]], "datalibrary (class in biolearn.data_library)": [[30, "biolearn.data_library.DataLibrary"]], "__init__() (biolearn.data_library.datalibrary method)": [[30, "biolearn.data_library.DataLibrary.__init__"]], "get() (biolearn.data_library.datalibrary method)": [[30, "biolearn.data_library.DataLibrary.get"]], "load_sources() (biolearn.data_library.datalibrary method)": [[30, "biolearn.data_library.DataLibrary.load_sources"]], "lookup_sources() (biolearn.data_library.datalibrary method)": [[30, "biolearn.data_library.DataLibrary.lookup_sources"]], "datasource (class in biolearn.data_library)": [[31, "biolearn.data_library.DataSource"]], "required_fields (biolearn.data_library.datasource attribute)": [[31, "biolearn.data_library.DataSource.REQUIRED_FIELDS"]], "__init__() (biolearn.data_library.datasource method)": [[31, "biolearn.data_library.DataSource.__init__"]], "load() (biolearn.data_library.datasource method)": [[31, "biolearn.data_library.DataSource.load"]], "geodata (class in biolearn.data_library)": [[32, "biolearn.data_library.GeoData"]], "__init__() (biolearn.data_library.geodata method)": [[32, "biolearn.data_library.GeoData.__init__"]], "copy() (biolearn.data_library.geodata method)": [[32, "biolearn.data_library.GeoData.copy"]], "dnam (biolearn.data_library.geodata attribute)": [[32, "biolearn.data_library.GeoData.dnam"]], "from_methylation_matrix() (biolearn.data_library.geodata class method)": [[32, "biolearn.data_library.GeoData.from_methylation_matrix"]], "metadata (biolearn.data_library.geodata attribute)": [[32, "biolearn.data_library.GeoData.metadata"]], "quality_report() (biolearn.data_library.geodata method)": [[32, "biolearn.data_library.GeoData.quality_report"]], "hybrid_impute() (in module biolearn.imputation)": [[33, "biolearn.imputation.hybrid_impute"]], "impute_from_average() (in module biolearn.imputation)": [[34, "biolearn.imputation.impute_from_average"]], "impute_from_standard() (in module biolearn.imputation)": [[35, "biolearn.imputation.impute_from_standard"]], "load_fhs() (in module biolearn.load)": [[36, "biolearn.load.load_fhs"]], "load_nhanes() (in module biolearn.load)": [[37, "biolearn.load.load_nhanes"]], "modelgallery (class in biolearn.model_gallery)": [[38, "biolearn.model_gallery.ModelGallery"]], "__init__() (biolearn.model_gallery.modelgallery method)": [[38, "biolearn.model_gallery.ModelGallery.__init__"]], "get() (biolearn.model_gallery.modelgallery method)": [[38, "biolearn.model_gallery.ModelGallery.get"]], "model_builders (biolearn.model_gallery.modelgallery attribute)": [[38, "biolearn.model_gallery.ModelGallery.model_builders"]], "search() (biolearn.model_gallery.modelgallery method)": [[38, "biolearn.model_gallery.ModelGallery.search"]], "biolearn.imputation": [[39, "module-biolearn.imputation"]], "biolearn.load": [[41, "module-biolearn.load"]], "biolearn.model_gallery": [[42, "module-biolearn.model_gallery"]]}})