Search.setIndex({"docnames": ["authors", "auto_examples/00_omics_biomarkers/index", "auto_examples/00_omics_biomarkers/plot_custom_imputation_implementation", "auto_examples/00_omics_biomarkers/plot_down_syndrome_model", "auto_examples/00_omics_biomarkers/plot_epigenetic_clocks_on_geo", "auto_examples/00_omics_biomarkers/plot_local_data_load", "auto_examples/00_omics_biomarkers/plot_transcriptomic_clock", "auto_examples/00_omics_biomarkers/sg_execution_times", "auto_examples/01_composite_biomarkers/index", "auto_examples/01_composite_biomarkers/plot_nhanes", "auto_examples/01_composite_biomarkers/sg_execution_times", "auto_examples/02_challenge_submissions/index", "auto_examples/02_challenge_submissions/plot_challenge_data", "auto_examples/02_challenge_submissions/sg_execution_times", "auto_examples/02_challenge_submissions/simple_competiton_submission", "auto_examples/02_challenge_submissions/training_simple_model", "auto_examples/03_deconvolution/index", "auto_examples/03_deconvolution/plot_blood_deconvolution", "auto_examples/03_deconvolution/sg_execution_times", "auto_examples/04_additional_visualizations/index", "auto_examples/04_additional_visualizations/plot_clock_model_relationships", "auto_examples/04_additional_visualizations/plot_dnam_relationships", "auto_examples/04_additional_visualizations/plot_quality_control", "auto_examples/04_additional_visualizations/sg_execution_times", "auto_examples/index", "auto_examples/sg_execution_times", "changes/names", "clocks", "data", "index", "methylation-standard", "model-usage", "modules/cache", "modules/data_library", "modules/generated/biolearn.cache.LocalFolderCache", "modules/generated/biolearn.cache.NoCache", "modules/generated/biolearn.data_library.DataLibrary", "modules/generated/biolearn.data_library.DataSource", "modules/generated/biolearn.data_library.GeoData", "modules/generated/biolearn.imputation.hybrid_impute", "modules/generated/biolearn.imputation.impute_from_average", "modules/generated/biolearn.imputation.impute_from_standard", "modules/generated/biolearn.load.load_fhs", "modules/generated/biolearn.load.load_nhanes", "modules/generated/biolearn.model_gallery.ModelGallery", "modules/generated/biolearn.mortality.calculate_age_adjusted_c_index", "modules/generated/biolearn.mortality.calculate_c_index", "modules/generated/biolearn.mortality.calculate_log_rank_test", "modules/generated/biolearn.mortality.calculate_mortality_hazard_ratios", "modules/generated/biolearn.mortality.plot_hazard_ratios", "modules/generated/biolearn.mortality.run_predictions", "modules/imputation", "modules/index", "modules/load", "modules/model_gallery", "modules/mortality", "quickstart", "roadmap", "sg_execution_times"], "filenames": ["authors.rst", "auto_examples/00_omics_biomarkers/index.rst", "auto_examples/00_omics_biomarkers/plot_custom_imputation_implementation.rst", "auto_examples/00_omics_biomarkers/plot_down_syndrome_model.rst", "auto_examples/00_omics_biomarkers/plot_epigenetic_clocks_on_geo.rst", "auto_examples/00_omics_biomarkers/plot_local_data_load.rst", "auto_examples/00_omics_biomarkers/plot_transcriptomic_clock.rst", "auto_examples/00_omics_biomarkers/sg_execution_times.rst", "auto_examples/01_composite_biomarkers/index.rst", "auto_examples/01_composite_biomarkers/plot_nhanes.rst", "auto_examples/01_composite_biomarkers/sg_execution_times.rst", "auto_examples/02_challenge_submissions/index.rst", "auto_examples/02_challenge_submissions/plot_challenge_data.rst", "auto_examples/02_challenge_submissions/sg_execution_times.rst", "auto_examples/02_challenge_submissions/simple_competiton_submission.rst", "auto_examples/02_challenge_submissions/training_simple_model.rst", "auto_examples/03_deconvolution/index.rst", "auto_examples/03_deconvolution/plot_blood_deconvolution.rst", "auto_examples/03_deconvolution/sg_execution_times.rst", "auto_examples/04_additional_visualizations/index.rst", "auto_examples/04_additional_visualizations/plot_clock_model_relationships.rst", "auto_examples/04_additional_visualizations/plot_dnam_relationships.rst", "auto_examples/04_additional_visualizations/plot_quality_control.rst", "auto_examples/04_additional_visualizations/sg_execution_times.rst", "auto_examples/index.rst", "auto_examples/sg_execution_times.rst", "changes/names.rst", "clocks.rst", "data.rst", "index.rst", "methylation-standard.rst", "model-usage.rst", "modules/cache.rst", "modules/data_library.rst", "modules/generated/biolearn.cache.LocalFolderCache.rst", "modules/generated/biolearn.cache.NoCache.rst", "modules/generated/biolearn.data_library.DataLibrary.rst", "modules/generated/biolearn.data_library.DataSource.rst", "modules/generated/biolearn.data_library.GeoData.rst", "modules/generated/biolearn.imputation.hybrid_impute.rst", "modules/generated/biolearn.imputation.impute_from_average.rst", "modules/generated/biolearn.imputation.impute_from_standard.rst", "modules/generated/biolearn.load.load_fhs.rst", "modules/generated/biolearn.load.load_nhanes.rst", "modules/generated/biolearn.model_gallery.ModelGallery.rst", "modules/generated/biolearn.mortality.calculate_age_adjusted_c_index.rst", "modules/generated/biolearn.mortality.calculate_c_index.rst", "modules/generated/biolearn.mortality.calculate_log_rank_test.rst", "modules/generated/biolearn.mortality.calculate_mortality_hazard_ratios.rst", "modules/generated/biolearn.mortality.plot_hazard_ratios.rst", "modules/generated/biolearn.mortality.run_predictions.rst", "modules/imputation.rst", "modules/index.rst", "modules/load.rst", "modules/model_gallery.rst", "modules/mortality.rst", "quickstart.md", "roadmap.rst", "sg_execution_times.rst"], "titles": ["Team", "Omics Biomarker Examples", "Performing custom imputations", "Down Syndrome Epigenetic Plotting", "\u201cEpigenetic Clocks\u201d in GEO Data", "Local Data Loading", "\u201cTranscriptomic Clock\u201d using GEO Data", "Computation times", "Composite Biomarkers Examples", "\u201cPhenotypic Ages\u201d in NHANES Data", "Computation times", "Challenge Submission Examples", "Exploring the Challenge Data", "Computation times", "Building a competition submission using an existing model", "Training an ElasticNet model", "Deconvolution Examples", "\u201cDeconvolution Example\u201d", "Computation times", "Additional Visualization Examples", "Clock/model visualizations using GEO datasets", "DNA methylation visualizations using GEO datasets", "Quality control visualization using GEO datasets", "Computation times", "Examples", "Computation times", "&lt;no title&gt;", "Clocks and Other Models", "GEO Data Sources", "Biolearn", "DNA Methylation Array Data Standard V-2410", "Model IP and Usage", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.cache</span></code>: Cache", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.data_library</span></code>: DataLibrary", "biolearn.cache.LocalFolderCache", "biolearn.cache.NoCache", "biolearn.data_library.DataLibrary", "biolearn.data_library.DataSource", "biolearn.data_library.GeoData", "biolearn.imputation.hybrid_impute", "biolearn.imputation.impute_from_average", "biolearn.imputation.impute_from_standard", "biolearn.load.load_fhs", "biolearn.load.load_nhanes", "biolearn.model_gallery.ModelGallery", "biolearn.mortality.calculate_age_adjusted_c_index", "biolearn.mortality.calculate_c_index", "biolearn.mortality.calculate_log_rank_test", "biolearn.mortality.calculate_mortality_hazard_ratios", "biolearn.mortality.plot_hazard_ratios", "biolearn.mortality.run_predictions", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.imputation</span></code>: Data Imputation Functions", "API References", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.load</span></code>: Data Loading Utilities", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.model_gallery</span></code>: ModelGallery", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">biolearn.mortality</span></code>: Mortality Predictor Evaluation", "Quickstart", "Biolearn and the Future of Biomarkers", "Computation times"], "terms": {"thi": [0, 2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 19, 20, 21, 22, 24, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 54, 55, 56, 57], "work": [0, 57], "i": [0, 22, 27, 28, 29, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 52, 56, 57], "made": [0, 57], "avail": [0, 29, 30, 44, 56, 57], "commun": [0, 14], "peopl": 0, "which": [0, 15, 34, 35, 38, 43], "origin": [0, 2], "from": [0, 5, 7, 8, 9, 10, 13, 14, 15, 17, 18, 20, 21, 23, 24, 25, 27, 28, 29, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 45, 46, 47, 48, 49, 50, 51, 56, 57, 58], "biomark": [0, 11, 28, 29, 36, 54, 56], "ag": [0, 5, 8, 10, 11, 12, 15, 19, 24, 27, 28, 29, 30, 43, 45, 46, 47, 48, 56, 57, 58], "consortium": [0, 29], "seth": 0, "paulson": [0, 29], "biolearn": [0, 2, 3, 4, 6, 9, 12, 14, 15, 17, 20, 21, 22, 27, 28, 52], "lead": 0, "mahdi": 0, "moqri": [0, 29], "research": [0, 29, 36, 57], "boa": 0, "execut": [0, 7, 10, 13, 18, 23, 24, 25, 56, 58], "albert": 0, "ying": [0, 29], "harvard": 0, "universii": 0, "martin": 0, "perez": [0, 29], "guevara": [0, 29], "machin": 0, "learn": [0, 5, 29, 56, 57], "engin": 0, "chancen": 0, "law": 0, "univers": 0, "alec": 0, "eam": 0, "dayoon": 0, "kwon": [0, 29], "california": 0, "mehrnoosh": 0, "emamifar": [0, 29], "northeastern": 0, "nureldin": 0, "moham": 0, "yasamin": 0, "jodat": 0, "_": [0, 6, 15], "gingko": 0, "biowork": 0, "aneesh": 0, "mazumd": 0, "ai4lif": 0, "institut": 0, "special": 0, "thank": 0, "scientif": 0, "member": [0, 12], "plot": [1, 2, 7, 8, 12, 16, 20, 22, 24, 29, 38, 49, 58], "result": [1, 3, 14, 16, 19, 24, 45, 46, 47, 48, 50], "run": [1, 5, 9, 14, 16, 17, 20, 21, 22, 24, 27, 29, 44, 50, 56], "base": [1, 8, 15, 24, 36, 38, 39, 44], "model": [1, 2, 6, 11, 13, 19, 23, 24, 36, 38, 44, 45, 47, 48, 50, 54, 57, 58], "function": [1, 19, 24, 37, 39, 40, 41, 42, 43, 45, 46, 47, 48, 49, 50, 52, 53, 55], "includ": [1, 5, 22, 24, 30, 37, 38, 56, 57], "methyl": [1, 5, 19, 22, 23, 24, 28, 36, 38, 39, 40, 41, 51, 57, 58], "transcriptom": [1, 7, 24, 38, 58], "data": [1, 7, 8, 10, 11, 13, 17, 20, 22, 24, 29, 32, 33, 34, 36, 37, 38, 39, 42, 43, 44, 45, 46, 47, 48, 49, 50, 56, 57, 58], "local": [1, 7, 24, 34, 38, 58], "load": [1, 7, 20, 22, 24, 29, 33, 36, 37, 38, 56, 57, 58], "clock": [1, 7, 8, 19, 23, 24, 29, 31, 37, 38, 54, 56, 58], "us": [1, 7, 8, 11, 13, 19, 23, 24, 28, 29, 30, 31, 34, 35, 39, 40, 42, 45, 46, 47, 48, 49, 56, 57, 58], "geo": [1, 7, 19, 23, 24, 30, 36, 37, 38, 57, 58], "epigenet": [1, 2, 7, 24, 28, 29, 37, 38, 57, 58], "down": [1, 7, 15, 24, 27, 38, 58], "syndrom": [1, 7, 24, 27, 38, 58], "perform": [1, 7, 9, 24, 38, 41, 58], "custom": [1, 7, 24, 38, 41, 58], "imput": [1, 4, 5, 7, 22, 24, 38, 44, 58], "go": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22], "end": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22], "download": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 24], "full": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22], "exampl": [2, 3, 4, 5, 6, 7, 9, 10, 12, 13, 14, 15, 18, 20, 21, 22, 23, 25, 56, 58], "code": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 24, 29, 56, 57], "browser": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22], "via": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 57], "binder": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22], "demonstr": [2, 20, 21, 22, 29], "eas": 2, "librari": [2, 21, 22, 28, 29, 32, 36, 44, 56, 57], "data_librari": [2, 3, 4, 5, 6, 12, 14, 15, 17, 20, 21, 22, 56], "import": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17], "datalibrari": [2, 3, 4, 6, 12, 15, 17, 20, 21, 22, 37, 56], "geodata": [2, 5, 14, 15, 21, 37, 48, 50], "comput": [2, 21, 29], "averag": [2, 30, 39, 40, 44, 51], "reference_dataset": 2, "gse40279": [2, 15, 28], "reference_averag": 2, "dnam": [2, 4, 5, 12, 15, 30, 38, 39, 40, 41], "mean": [2, 15], "axi": 2, "1": [2, 5, 10, 12, 15, 17, 18, 21, 23, 30, 38, 48, 57, 58], "second": [2, 3, 4, 5, 6, 9, 12, 17, 20, 21, 22], "target_dataset": 2, "gse51057": [2, 28], "impute_from_standard": 2, "imputed_data": 2, "imputed_dataset": 2, "metadata": [2, 3, 6, 15, 17, 37, 38, 45, 46, 47, 48], "galleri": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 24, 27, 50, 58], "compar": [2, 29], "model_galleri": [2, 3, 4, 6, 14, 17, 20], "modelgalleri": [2, 3, 4, 6, 14, 17, 20], "clock_model": 2, "horvathv1": [2, 4, 12, 20, 27, 31, 50], "original_age_predict": 2, "imputed_age_predict": 2, "seaborn": [2, 3, 9, 12, 17], "sn": [2, 3, 4, 6, 9, 12, 17], "panda": [2, 4, 12, 15, 38, 42, 43], "pd": [2, 4, 12, 15, 39, 40, 41, 45, 46, 47, 48, 49, 50], "matplotlib": [2, 3, 6, 9, 12, 15, 17], "pyplot": [2, 3, 6, 9, 12, 15, 17], "plt": [2, 3, 6, 9, 12, 15, 17], "prepar": [2, 57], "comparison_data": 2, "datafram": [2, 4, 12, 15, 38, 39, 40, 41, 42, 43, 45, 46, 47, 48, 49, 50], "creat": [2, 3, 12, 20, 38, 56], "scatter": [2, 4, 12, 15], "figur": [2, 3, 12, 15, 17], "figsiz": [2, 3, 12, 15, 17], "8": [2, 5, 7, 12, 15, 21, 39, 58], "6": [2, 3, 4, 12, 15, 21, 22, 28], "scatterplot": [2, 9, 17], "x": [2, 3, 6, 9, 12, 15, 17], "y": [2, 3, 6, 9, 12, 15, 17], "titl": [2, 3, 6, 12, 15, 17, 28, 37], "v": [2, 15, 21, 29, 38, 56], "xlabel": [2, 6, 9, 12, 15, 17], "ylabel": [2, 6, 9, 12, 15, 17], "grid": [2, 12, 15], "true": [2, 3, 12, 15], "show": [2, 5, 6, 12, 14, 15, 22, 29], "imputationdecor": 2, "defin": [2, 15, 36], "method": [2, 37, 44, 51], "def": 2, "custom_impute_method": 2, "dnam_data": 2, "needed_cpg": 2, "return": [2, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50], "wrap": 2, "decorated_clock": 2, "decorated_clock_predict": 2, "verifi": [2, 56], "ar": [2, 15, 28, 29, 30, 31, 36, 37, 38, 39, 43, 44, 45, 46, 47, 48, 50, 56, 57], "direct": 2, "approach": [2, 57], "same_result": 2, "all": [2, 5, 22, 24, 30, 31, 34, 35, 36, 38, 40, 41, 58], "print": [2, 6, 15, 21], "total": [2, 3, 4, 5, 6, 7, 9, 10, 12, 13, 17, 18, 20, 21, 22, 23, 25, 27, 29, 58], "time": [2, 3, 4, 5, 6, 9, 12, 15, 17, 20, 21, 22, 46, 57], "script": [2, 3, 4, 5, 6, 9, 12, 17, 20, 21, 22], "0": [2, 3, 4, 5, 6, 9, 12, 13, 14, 15, 17, 20, 21, 22, 23, 25, 30, 38, 39, 48, 58], "minut": [2, 3, 4, 5, 6, 9, 12, 17, 20, 21, 22, 57], "30": [2, 7, 58], "427": [2, 7, 58], "estim": [2, 3, 4, 5, 6, 9, 12, 14, 15, 20, 21, 22], "memori": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 28], "usag": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 27], "9838": 2, "mb": [2, 3, 4, 5, 6, 7, 9, 10, 12, 13, 14, 15, 17, 18, 20, 21, 22, 23, 25, 58], "jupyt": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 24, 56], "notebook": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 24, 56], "plot_custom_imputation_implement": [2, 7, 58], "ipynb": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22], "python": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 24, 56], "sourc": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 24, 27, 29, 36, 37, 39, 56, 57], "py": [2, 3, 4, 5, 6, 7, 9, 10, 12, 13, 14, 15, 17, 18, 19, 20, 21, 22, 23, 24, 58], "gener": [2, 4, 5, 6, 9, 11, 12, 14, 15, 17, 20, 21, 24, 38, 57], "sphinx": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 24], "your": [3, 4, 5, 6, 9, 12, 14, 15, 17, 20, 21, 22, 29, 57], "dna": [3, 4, 17, 19, 23, 24, 27, 28, 36, 38, 39, 40, 41, 51, 58], "how": [3, 12, 14, 29, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 56, 57], "samepl": 3, "can": [3, 14, 27, 29, 30, 31, 32, 36, 38, 44, 56, 57], "distiguish": 3, "test_data": 3, "gse52588": [3, 20, 21, 22, 28], "downsyndrom": [3, 20, 27, 31], "set": [3, 4, 5, 22, 28, 36, 37, 44, 51, 55], "index": [3, 4, 15, 45, 46], "merg": 3, "set_index": 3, "astyp": [3, 17], "str": [3, 34, 35, 36, 38, 39, 40, 41, 44], "inplac": 3, "merged_data": 3, "join": [3, 29], "category_ord": 3, "healthi": [3, 28], "palett": 3, "green": 3, "red": 3, "configur": [3, 37, 57], "10": [3, 12, 15, 29, 56], "ax": [3, 9], "disease_st": [3, 5, 20], "width": 3, "3": [3, 5, 12, 13, 15, 18, 23, 56, 58], "order": 3, "stripplot": 3, "jitter": 3, "hue": 3, "dodg": 3, "fals": 3, "legend": [3, 17], "set_titl": 3, "despin": 3, "4": [3, 5, 12, 17], "346": [3, 7, 58], "934": 3, "plot_down_syndrome_model": [3, 7, 58], "calcul": [4, 6, 15, 44, 45, 46, 47, 48], "multipl": [4, 28, 30], "them": [4, 17, 24, 36, 57], "against": 4, "chronolog": [4, 29], "gse41169": [4, 21, 22, 28], "blood": [4, 8, 9, 12, 24, 27, 28, 57], "data_sourc": 4, "get": [4, 6, 12, 14, 15, 17, 20, 21, 22, 29, 34, 35, 36, 44, 56], "note": [4, 5, 15, 21, 22], "default": [4, 5, 22, 36, 37, 39, 44], "miss": [4, 5, 22, 35, 37, 38, 39, 40, 41, 44, 51], "To": [4, 56], "chang": [4, 28, 56], "behavior": [4, 56], "paramet": [4, 15, 34, 35, 36, 37, 38, 39, 40, 41, 43, 44, 45, 46, 47, 48, 49, 50], "when": 4, "horvath_result": 4, "predict": [4, 12, 15, 17, 19, 24, 27, 29, 44, 50], "hannum_result": 4, "hannum": [4, 12, 20, 27, 31], "phenoage_result": 4, "phenoag": [4, 20, 27, 31], "actual_ag": 4, "plot_data": 4, "horvath": [4, 29, 56], "relplot": 4, "kind": 4, "axisgrid": 4, "facetgrid": 4, "object": [4, 15, 20, 34, 36, 37, 38, 44, 45, 46, 47, 48, 50], "0x7b97fdff41f0": 4, "464": [4, 7, 28, 58], "1555": 4, "plot_epigenetic_clocks_on_geo": [4, 7, 58], "util": [5, 12], "get_test_data_fil": 5, "format": [5, 14, 28, 33, 36, 37, 38, 57], "describ": [5, 30], "standard": [5, 8, 22, 24, 38, 44], "http": [5, 14, 15, 29, 56], "bio": [5, 56], "github": [5, 56], "io": [5, 56], "html": 5, "correctli": [5, 56], "search": [5, 44], "name": [5, 12, 15, 20, 27, 31, 38, 44, 45, 46, 47, 48, 50], "_metadata": 5, "csv": [5, 12, 14, 30, 38], "_methylation_part0": 5, "file_path": 5, "load_csv": [5, 38], "gsm3074480": [5, 30], "gsm3074481": [5, 30], "gsm3074482": [5, 30], "gsm3074483": [5, 30], "cpgsite": [5, 30], "cg00000029": [5, 12, 21, 30], "395": [5, 30], "499": [5, 30], "420": 5, "423": [5, 30], "cg00000109": [5, 12, 21, 30], "821": [5, 30], "868": [5, 30], "877": [5, 30], "841": [5, 30], "cg00000155": [5, 12, 21, 30], "878": [5, 30], "890": 5, "894": [5, 30], "nan": [5, 12, 30, 38], "sex": [5, 12, 27, 28, 30, 38], "sampleid": [5, 14, 15, 30], "25": [5, 12, 13, 30, 57, 58], "2": [5, 7, 9, 12, 15, 21, 28, 30, 38, 58], "none": [5, 6, 12, 22, 30, 34, 35, 36, 37, 38, 40, 41, 44, 49], "29": [5, 12, 30], "5": [5, 6, 7, 9, 13, 15, 17, 21, 22, 30, 58], "covid": [5, 30], "39": [5, 12, 30], "7": [5, 12, 15, 22, 23, 30, 58], "quality_report": [5, 22, 38], "qualiti": [5, 19, 23, 24, 30, 38, 58], "report": [5, 38], "summari": [5, 22, 37, 38], "sampl": [5, 15, 28, 30, 38, 39, 40, 41, 45, 47, 48, 57], "count": [5, 15, 22], "site": [5, 15, 22, 30, 38, 39, 40, 41], "33": [5, 12], "With": [5, 22], "high": [5, 22], "deviat": 5, "00": [5, 7, 10, 12, 13, 18, 22, 23, 25, 58], "over": [5, 22], "20": [5, 15, 22], "read": [5, 22, 29, 56], "have": [5, 12, 15, 22, 24, 30, 56, 57], "mai": [5, 22, 56], "replac": [5, 15, 22, 39], "valu": [5, 15, 21, 22, 34, 35, 38, 39, 40, 41, 44, 45, 46, 47, 48, 49, 50, 51], "gold": [5, 22, 44], "482": [5, 7, 58], "78": [5, 7, 58], "plot_local_data_load": [5, 7, 58], "gse134080": [6, 28], "rna_model": 6, "transcriptomicpredictionmodel": [6, 27, 31], "imputation_method": [6, 44], "scipi": 6, "stat": 6, "pearsonr": 6, "predicted_valu": 6, "true_valu": 6, "correl": [6, 15, 21], "f": [6, 15, 21], "pearson": [6, 21], "coeffici": [6, 15, 27], "set_them": 6, "style": 6, "whitegrid": 6, "regplot": [6, 12], "scatter_kw": 6, "alpha": [6, 15], "7025211774998293": 6, "081": [6, 7, 58], "393": 6, "plot_transcriptomic_clock": [6, 7, 58], "46": 7, "800": 7, "file": [7, 10, 13, 14, 18, 23, 25, 27, 36, 37, 38, 57, 58], "auto_exampl": [7, 10, 13, 18, 23, 25], "00_omics_biomark": [7, 58], "mem": [7, 10, 13, 18, 23, 25, 58], "9837": [7, 58], "06": [7, 22, 23, 58], "1554": [7, 58], "9": [7, 12, 58], "05": [7, 15, 58], "392": [7, 58], "04": [7, 18, 58], "933": [7, 58], "surviv": [8, 24, 29], "curv": [8, 24], "test": [8, 24, 47, 57], "phenotyp": [8, 10, 24, 43, 58], "nhane": [8, 10, 24, 29, 43, 58], "exam": 9, "analys": [9, 29, 56], "load_nhan": 9, "year": [9, 27, 30, 43, 45, 47, 48], "df": [9, 12, 15, 42, 43], "years_until_death": [9, 45, 46, 47, 48], "months_until_death": 9, "12": [9, 12, 28, 29], "hematologi": 9, "phenotypic_ag": 9, "": [9, 20, 27, 29, 37, 38, 44, 57], "lifelin": 9, "kaplanmeierfitt": 9, "biologically_old": 9, "subplot": 9, "group": [9, 21, 43], "ix": 9, "t": [9, 15, 17], "e": 9, "is_dead": 9, "kmf": 9, "fit": [9, 15], "label": 9, "plot_survival_funct": 9, "text": [9, 17], "23": 9, "52222222222222": 9, "593": [9, 10, 58], "19": [9, 10, 58], "plot_nhan": [9, 10, 58], "01_composite_biomark": [10, 58], "build": [11, 13, 24, 44, 57, 58], "competit": [11, 13, 24, 44, 58], "an": [11, 13, 20, 21, 22, 24, 28, 29, 34, 35, 36, 37, 38, 39, 41, 42, 43, 44, 57, 58], "exist": [11, 13, 24, 30, 34, 39, 44, 58], "explor": [11, 13, 24, 38, 50, 57, 58], "train": [11, 13, 24, 36, 58], "elasticnet": [11, 13, 24, 36, 58], "challenge_data": [12, 14, 15], "boachallengedata": [12, 28], "option": [12, 36, 37, 38, 39, 40, 41, 44], "displai": [12, 28], "max_column": 12, "gsm7866999": 12, "gsm7867305": 12, "gsm7867077": 12, "gsm7867431": 12, "gsm7867126": 12, "gsm7867000": 12, "720084": 12, "453491": 12, "535564": 12, "487361": 12, "337215": 12, "062426": 12, "938044": 12, "925149": 12, "949446": 12, "938523": 12, "940187": 12, "942090": 12, "957093": 12, "959030": 12, "948919": 12, "966912": 12, "968026": 12, "970274": 12, "cg00000158": [12, 21], "965191": 12, "957272": 12, "967278": 12, "971039": 12, "969745": 12, "969660": 12, "cg00000165": [12, 21], "144061": 12, "122099": 12, "167346": 12, "156199": 12, "133038": 12, "641788": 12, "rs9292570": 12, "523412": 12, "980810": 12, "986609": 12, "525876": 12, "537722": 12, "538440": 12, "rs9363764": 12, "960815": 12, "038638": 12, "451884": 12, "978343": 12, "971838": 12, "973633": 12, "rs951295": 12, "971034": 12, "558011": 12, "553397": 12, "485287": 12, "526686": 12, "499980": 12, "rs966367": 12, "455920": 12, "963994": 12, "453094": 12, "455700": 12, "029342": 12, "026907": 12, "rs9839873": 12, "599735": 12, "595571": 12, "630871": 12, "576566": 12, "579195": 12, "973815": 12, "930659": 12, "row": [12, 15, 30, 38, 39, 40, 41, 42, 43, 45, 46, 47, 48], "500": [12, 28, 58], "column": [12, 14, 15, 20, 30, 38, 39, 40, 41, 42, 43, 45, 46, 47, 48, 50], "gsm7867173": 12, "gsm7867127": 12, "gsm7867083": 12, "gsm7867095": 12, "gsm7867115": 12, "gsm7867385": 12, "alamartargetid": 12, "t10319": 12, "13": [12, 22, 58], "387778": 12, "948052": 12, "256551": 12, "062449": 12, "390582": 12, "587867": 12, "t10413": 12, "16": 12, "674015": 12, "15": [12, 22, 23, 58], "992939": 12, "547946": 12, "14": 12, "827118": 12, "872686": 12, "584386": 12, "t10466": 12, "890963": 12, "556167": 12, "528756": 12, "683530": 12, "667331": 12, "483932": 12, "t10563": 12, "933342": 12, "323690": 12, "271350": 12, "993500": 12, "11": 12, "699095": 12, "860206": 12, "t10876": 12, "063506": 12, "429302": 12, "582464": 12, "978461": 12, "654607": 12, "931614": 12, "t8333": 12, "725785": 12, "537733": 12, "848917": 12, "729721": 12, "787099": 12, "055526": 12, "t8334": 12, "503408": 12, "103376": 12, "381374": 12, "070087": 12, "317681": 12, "314627": 12, "t8358": 12, "734120": 12, "238780": 12, "673970": 12, "103683": 12, "949114": 12, "794795": 12, "t8367": 12, "062422": 12, "291734": 12, "604654": 12, "803189": 12, "313347": 12, "830717": 12, "t8399": 12, "599597": 12, "642498": 12, "590723": 12, "073775": 12, "184558": 12, "040475": 12, "374": 12, "503": 12, "get_data_fil": 12, "read_csv": 12, "alamar_refer": 12, "uniprotid": 12, "target": 12, "proteinnam": 12, "t10034": 12, "o75888": 12, "tnfsf13": 12, "tnf": 12, "superfamili": 12, "o95760": 12, "il33": 12, "interleukin": 12, "t10412": 12, "p01258": 12, "calca": 12, "calcitonin": 12, "cleav": 12, "katacalcin": 12, "t10417": 12, "p01903": 12, "hla": 12, "dra": 12, "major": 12, "histocompat": 12, "complex": [12, 51, 57], "class": [12, 32, 33, 34, 35, 36, 37, 38, 44, 52, 54], "ii": 12, "dr": 12, "t10433": 12, "p02745": 12, "c1qa": 12, "complement": 12, "c1q": 12, "A": [12, 25, 34, 35, 36, 37, 38, 42, 43, 44, 45, 46, 47, 48, 49, 50], "chain": 12, "317": 12, "t8244": 12, "p35247": 12, "sftpd": 12, "pulmonari": 12, "surfact": 12, "associ": [12, 15, 28, 30, 38], "d": [12, 28, 29], "318": 12, "t8246": 12, "p22303": 12, "ach": 12, "acetylcholinesteras": 12, "319": 12, "t8254": 12, "q8n474": 12, "sfrp1": 12, "secret": 12, "frizzl": 12, "relat": [12, 28, 56], "320": 12, "t8355": 12, "p0dji8": 12, "saa1": 12, "serum": 12, "amyloid": 12, "321": 12, "t9441": 12, "o15240": 12, "vgf": 12, "nerv": 12, "growth": 12, "factor": 12, "induc": 12, "322": 12, "ethnic": 12, "race1": 12, "subject_id": 12, "tissu": [12, 27, 28, 44], "gsm7866964": 12, "31": 12, "35": 12, "non": [12, 31], "hispan": 12, "white": 12, "boa1": 12, "gsm7866965": 12, "79": 12, "45": 12, "asian": 12, "boa2": 12, "gsm7866966": 12, "60": [12, 15], "42": [12, 15, 30], "boa3": 12, "gsm7866967": 12, "59": 12, "24": [12, 22], "boa4": 12, "gsm7866968": 12, "22": 12, "41": 12, "boa5": 12, "p499": 12, "80": 12, "98": [12, 22], "p500": 12, "89": [12, 30], "p501": 12, "92": [12, 28], "p502": 12, "69": 12, "p503": 12, "68": [12, 28], "18": 12, "651": 12, "mortal": [12, 27, 29], "run_predict": 12, "prediction_dict": 12, "21": [12, 20, 23, 58], "822423": 12, "17": 12, "303025": 12, "56": 12, "266752": 12, "52": 12, "803410": 12, "660930": 12, "36": 12, "211078": 12, "806215": 12, "32": [12, 28], "821235": 12, "gsm7867457": 12, "51": 12, "285108": 12, "49": [12, 22, 28], "308239": 12, "gsm7867136": 12, "37": 12, "539954": 12, "494127": 12, "gsm7867452": 12, "74": 12, "975617": 12, "639608": 12, "083565": 12, "906176": 12, "91": 12, "428559": 12, "870807": 12, "75": 12, "005273": 12, "77": 12, "630528": 12, "assum": [12, 30], "should": [12, 30, 43, 45, 47, 48], "regress": 12, "line": [12, 15, 21, 29, 56], "ci": 12, "779": [12, 13, 58], "15006": [12, 13, 58], "plot_challenge_data": [12, 13, 58], "02_challenge_submiss": [13, 58], "challeng": [13, 14, 28, 29, 38, 50, 57, 58], "submiss": [13, 44, 58], "simple_competiton_submiss": [13, 14, 58], "000": [13, 25, 58], "training_simple_model": [13, 15, 58], "you": [14, 15, 24, 29, 31, 56, 57], "warmup": [14, 15], "link": [14, 15], "here": [14, 15, 27], "www": [14, 15], "synaps": [14, 15, 29], "org": [14, 15, 29], "syn52966292": [14, 15], "wiki": [14, 15], "625231": [14, 15], "downloaded_data_file_path": [14, 15], "add": [14, 15], "path": [14, 15, 34, 36, 37, 38, 57], "from_methylation_matrix": [14, 15, 38], "score": 14, "best": 14, "so": 14, "let": 14, "save": [14, 38], "correct": [14, 30], "now": [14, 57], "we": [14, 29, 30, 56, 57], "expect": [14, 56], "rename_axi": 14, "renam": 14, "predictedag": [14, 15], "to_csv": 14, "lin_submiss": 14, "submit": [14, 15], "follow": [14, 24, 30, 48], "instruct": [14, 24, 56], "below": [14, 39], "simpl": [15, 51], "narrow": 15, "what": [15, 31], "take": [15, 57], "long": 15, "numpi": 15, "np": 15, "sklearn": 15, "linear_model": 15, "linearregress": 15, "extract": [15, 57], "transpos": 15, "cpg": [15, 30, 39, 40, 41], "bootstrap": 15, "featur": [15, 56], "select": [15, 44], "n_bootstrap": 15, "threshold": [15, 21, 39], "store": [15, 28, 34, 35, 36], "each": [15, 30, 38, 42, 43, 45, 46, 47, 48, 49, 50], "deem": 15, "signific": 15, "cpg_count": 15, "zero": 15, "shape": [15, 28], "begin": 15, "iter": 15, "rang": [15, 30], "sample_idx": 15, "random": 15, "choic": 15, "size": [15, 34], "x_sampl": 15, "y_sampl": 15, "identifi": [15, 36, 38, 57], "magnitud": [15, 57], "significant_cpg": 15, "where": [15, 24, 29, 38, 42, 43, 50, 57], "ab": 15, "coef_": 15, "determin": [15, 31, 39], "stabl": 15, "stable_cpg_sit": [15, 21], "stable_cpg_nam": 15, "tolist": 15, "more": [15, 29, 30, 51, 56, 57], "than": [15, 30], "model_select": 15, "train_test_split": 15, "top_sites_df": 15, "x_train": 15, "x_test": 15, "y_train": 15, "y_test": 15, "test_siz": 15, "random_st": 15, "regular": 15, "strength": 15, "l1_ratio": 15, "mix": 15, "between": [15, 29, 30, 45, 47, 48], "l1": 15, "l2": 15, "lasso": 15, "ridg": [15, 22], "metric": 15, "mean_squared_error": 15, "01": [15, 21, 58], "max_it": 15, "10000": 15, "evalu": 15, "y_pred": 15, "mse": 15, "squar": 15, "error": [15, 34, 56], "edgecolor": 15, "w": [15, 28], "linewidth": 15, "min": 15, "max": 15, "k": [15, 29], "lw": 15, "refer": [15, 28, 29, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 56], "actual": [15, 56], "both": [15, 28, 38, 39, 57], "linestyl": 15, "4f": 15, "absolut": 15, "mae": 15, "pruned_data": 15, "fillna": 15, "challenge_result": 15, "predicted_age_df": 15, "dataset": [16, 19, 23, 24, 28, 38, 39, 40, 41, 44, 45, 46, 47, 51, 57, 58], "known": [16, 24, 45, 47, 48], "cell": [16, 24, 27, 28], "proport": [16, 24, 27, 48], "type": [17, 28, 34, 35, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 50], "epic_facs_salas_18": 17, "gse112618": [17, 21, 22, 28], "epic_facs_cell_prop": 17, "deconvolutebloodep": [17, 27, 31], "set_styl": 17, "tick": 17, "neutrophils_proport": 17, "float": [17, 34, 39], "loc": 17, "neutrophil": 17, "monocytes_proport": 17, "monocyt": 17, "nk_proport": 17, "nk_cell": 17, "bcell_proport": 17, "b_cell": 17, "cd4t_proport": 17, "cd4_t_cell": 17, "cd8t_proport": 17, "cd8_t_cell": 17, "nk": 17, "b": 17, "cd4": 17, "cd8": 17, "valid": [17, 28], "265": [17, 18, 58], "73": [17, 18, 58], "plot_blood_deconvolut": [17, 18, 58], "03_deconvolut": [18, 58], "deconvolut": [18, 38, 58], "document": [19, 24, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 56], "showcas": [19, 24], "control": [19, 23, 24, 30, 38, 58], "built": [20, 21, 22, 57], "differ": [20, 28, 38, 54, 57], "plot_clock_correlation_matrix": 20, "plot_clock_deviation_heatmap": 20, "plot_age_predict": 20, "plot_health_outcom": 20, "appropri": [20, 21], "gse120307": [20, 28], "list": [20, 21, 36, 38, 39, 40, 41, 50], "analyz": 20, "modelnam": 20, "dunedinpac": [20, 27, 29, 31, 44, 56], "lin": [20, 27, 31], "zhang_10": [20, 27, 31], "age_prediction_model": 20, "down_syndrome_data": 20, "provid": [20, 27, 29, 32, 33, 37, 49, 51, 54, 55, 56, 57], "health_outcome_col": 20, "721": [20, 23, 58], "1483": [20, 23, 58], "plot_clock_model_relationship": [20, 23, 58], "across": 21, "identify_stable_cpg_sit": 21, "compute_age_group_stat": 21, "compute_methylation_stat": 21, "plot_methylation_by_age_sex": 21, "plot_methylation_vs_ag": 21, "id": [21, 22, 28, 30, 37, 45, 46, 47, 48], "dataset_id": [21, 22], "gse110554": [21, 22, 28], "dictionari": [21, 37, 44, 50], "stable_sit": 21, "dataset_nam": 21, "cg00000103": 21, "cg00000221": 21, "cg00000236": 21, "cg00000289": 21, "cg00000292": 21, "cg00000321": 21, "cg00000363": 21, "cg00000108": 21, "cg00000622": 21, "initi": [21, 34, 36, 37, 38, 44], "cpg_site": 21, "statist": [21, 47], "specifi": [21, 34, 36, 38, 44, 50], "stats_df": 21, "combined_data1": 21, "age_bin": 21, "In": 21, "repres": [21, 30, 37, 38, 42, 43], "femal": [21, 30, 38], "male": [21, 30, 38], "combined_data2": 21, "r": [21, 29], "0311": 21, "1764": 21, "p": [21, 47, 48, 49], "2813e": 21, "03": 21, "degre": 21, "0327": 21, "720": [21, 23, 28, 58], "1814": 21, "plot_dnam_relationship": [21, 23, 58], "densiti": 22, "plot_sample_devi": 22, "866091": 22, "No": [22, 28, 30, 44], "point": [22, 29], "impli": 22, "ha": [22, 30, 56, 57], "alreadi": 22, "gone": 22, "through": 22, "process": [22, 30, 57], "low": 22, "were": 22, "48": [22, 28], "It": [22, 29, 30, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 56], "like": [22, 29, 30, 37, 56, 57], "been": [22, 30], "distort": 22, "due": 22, "technic": 22, "issu": [22, 57], "95": [22, 28], "485577": 22, "60482": 22, "87": [22, 28], "27449": 22, "621": [22, 23, 58], "3082": [22, 23, 58], "plot_quality_control": [22, 23, 58], "44": 23, "062": 23, "04_additional_visu": [23, 58], "visual": [23, 38, 58], "1813": [23, 58], "If": [24, 29, 30, 31, 36, 37, 38, 39, 44, 56, 57], "want": [24, 57], "make": 24, "sure": 24, "directori": [24, 34, 38], "write": [24, 57], "permiss": 24, "copi": [24, 38], "instal": 24, "biolean": 24, "manual": 24, "auto_examples_python": 24, "zip": 24, "auto_examples_jupyt": 24, "n": [25, 29], "support": [27, 28, 29, 36, 43, 44, 57], "number": [27, 28, 30, 38, 43], "easili": [27, 29, 36, 56], "ip": 27, "inform": 27, "speci": [27, 44], "2013": 27, "human": [27, 28, 57], "multi": [27, 29, 57], "paper": [27, 29, 31, 56], "2016": 27, "2018": 27, "yingcausag": [27, 31], "2022": 27, "yingdamag": [27, 31], "yingadaptag": [27, 31], "horvathv2": [27, 31], "skin": 27, "pedb": [27, 31], "2019": 27, "buccal": 27, "risk": 27, "dunedinpoam38": [27, 31], "2020": 27, "rate": 27, "grimagev1": [27, 31], "adjust": [27, 45, 46, 47], "grimagev2": [27, 31], "alcoholmccartnei": [27, 31], "alcohol": 27, "consumpt": 27, "dnamtl": [27, 31], "adipos": 27, "telomer": 27, "length": [27, 30], "hrsinchphenoag": [27, 31], "knight": [27, 31], "cord": [27, 28], "gestat": 27, "leecontrol": [27, 31], "placenta": 27, "sexestim": [27, 31], "2021": 27, "leerefinedrobust": [27, 31], "leerobust": [27, 31], "smokingmccartnei": [27, 31], "smoke": [27, 28], "statu": [27, 31], "deconvoluteblood450k": [27, 31], "2024": [27, 28, 29], "twelvecelldeconvolutebloodep": [27, 31], "bmi_mccartnei": [27, 31], "bmi": 27, "educationmccartnei": [27, 31], "educ": 27, "attain": 27, "totalcholesterolmccartnei": [27, 31], "cholesterol": 27, "hdlcholesterolmccartnei": [27, 31], "hdl": 27, "ldlcholesterolmccartnei": [27, 31], "ldl": 27, "remnant": 27, "bodyfatmccartnei": [27, 31], "percentag": [27, 38], "bodi": 27, "fat": 27, "bmi_re": [27, 31], "prostatecancerkirbi": [27, 31], "2017": 27, "prostat": 27, "cancer": 27, "hepatoxu": [27, 31], "circul": 27, "hepatocellular": 27, "carcinoma": 27, "cvd_westerman": [27, 31], "coronari": 27, "heart": [27, 29, 42, 56], "diseas": [27, 28, 30, 57], "ad_bahado": [27, 31], "singh": [27, 31], "alzheim": 27, "depressionbarbu": [27, 31], "depress": 27, "2015": 27, "present": 28, "fac": 28, "optim": [28, 57], "illuminaep": 28, "ye": 28, "flowsort": 28, "epic": 28, "ba": 28, "genom": [28, 38], "wide": 28, "profil": 28, "reveal": 28, "quantit": 28, "view": 28, "o": 28, "illumina450k": 28, "656": 28, "gse19711": 28, "unit": 28, "kingdom": 28, "ovar": 28, "illumina27k": 28, "540": 28, "methylom": 28, "analysi": [28, 30], "me": 28, "329": 28, "gse42861": 28, "differenti": 28, "rheumatoid": 28, "arthriti": 28, "689": 28, "dutch": 28, "popul": 28, "gse51032": 28, "itali": 28, "hugef": 28, "845": 28, "gse73103": 28, "mani": [28, 29, 56, 57], "obes": 28, "snp": 28, "strongli": 28, "met": 28, "355": 28, "gse69270": 28, "middl": 28, "indi": 28, "184": 28, "gse36054": 28, "children": 28, "192": 28, "gse64495": 28, "sever": [28, 29], "113": 28, "gse30870": 28, "newborn": 28, "nonagenarian": 28, "40": 28, "identif": 28, "signatur": [28, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50], "gse157131": 28, "peripher": 28, "leukocyt": 28, "fro": 28, "946": 28, "gse132203": 28, "gradi": 28, "trauma": 28, "project": 28, "795": 28, "rnaseq": 28, "whole": 28, "500fg": 28, "cohort": 28, "illuminahiseq2500": 28, "100": 28, "jenag": 28, "deep": [28, 38], "sequenc": 28, "endogen": 28, "mrna": 28, "homo": 28, "sapien": 28, "fo": 28, "62": 28, "gse167998": 28, "bloodextend": 28, "enhanc": [28, 44], "l": 28, "gse180683": 28, "cd4t": 28, "cd8t": 28, "naiv": 28, "76": 28, "gse180970": 28, "umbil": 28, "artifici": 28, "mixtur": 28, "gse182379": 28, "enha": 28, "gse106648": 28, "sclerosi": 28, "279": 28, "34": 28, "gse134429": 28, "inflammatori": 28, "cytokin": 28, "mon": 28, "64": 28, "gse151278": 28, "sam": 28, "70": 28, "gse164056": 28, "social": 28, "anxieti": 28, "143": 28, "gse166611": 28, "trait": 28, "gse20236": 28, "hypermethyl": 28, "occur": [28, 34], "prefer": 28, "93": 28, "gse32148": 28, "individu": [28, 29, 42, 43], "cr": 28, "gse41037": 28, "effect": 28, "modul": [28, 32, 33, 36, 37, 51, 54, 55], "gse49904": 28, "common": [28, 29, 30, 56], "71": 28, "gse50660": 28, "cigarett": 28, "reduc": 28, "level": [28, 30, 57], "gse53740": 28, "384": 28, "gse53840": 28, "hiv": 28, "posit": 28, "men": 28, "120": 28, "gse67705": 28, "chronic": 28, "infect": 28, "patient": 28, "284": 28, "gse67751": 28, "illumina": [28, 30], "inf": 28, "450": 28, "gse77445": 28, "alter": 28, "cortisol": 28, "stre": 28, "85": 28, "gse85311": 28, "pattern": 28, "exercis": 28, "38": 28, "gse96879": 28, "epigenom": 28, "sustem": 28, "lupu": 28, "erythem": 28, "90": 28, "gse99624": 28, "primari": 28, "osteoporosi": 28, "reflect": 28, "specif": [28, 30, 34, 35], "dn": 28, "enabl": [29, 56], "easi": [29, 56, 57], "versatil": [29, 56], "tool": [29, 56, 57], "publicli": [29, 56], "gene": [29, 56], "express": [29, 56], "omnibu": [29, 56], "nation": [29, 43, 56], "health": [29, 43, 56, 57], "nutrit": [29, 43, 56], "examin": [29, 43], "survei": [29, 43, 56], "framingham": [29, 42, 56], "studi": [29, 42, 56], "also": [29, 56, 57], "contain": [29, 30, 37, 38, 39, 41, 45, 46, 47, 48, 49, 50, 56], "implement": [29, 32, 33, 34, 35, 54, 57], "other": [29, 30, 31, 38, 39, 56], "onli": [29, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 56], "few": [29, 56], "about": [29, 37, 42, 43, 56, 57], "our": [29, 56, 57], "develop": [29, 56, 57], "organ": [29, 36, 37, 38], "especi": 29, "methuselah": 29, "foundat": [29, 31], "volo": 29, "host": 29, "2025": 29, "seri": [29, 30, 39, 41], "platform": 29, "particip": 29, "ask": 29, "morbid": 29, "award": 29, "200k": 29, "pleas": [29, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 56, 57], "cite": 29, "preprint": 29, "m": 29, "mart\u00ed": 29, "nez": 29, "c": [29, 45, 46], "poganik": 29, "j": 29, "gladyshev": 29, "2023": 29, "open": [29, 56, 57], "biorxiv": 29, "doi": 29, "1101": 29, "02": [29, 58], "569722": 29, "quickstart": 29, "start": [29, 56], "roadmap": [29, 57], "see": [29, 57], "head": 29, "contribut": [29, 57], "biolog": [29, 44], "relationship": 29, "variou": [29, 44, 51], "consist": 30, "focus": 30, "storag": [30, 34], "transmiss": 30, "without": [30, 46], "mandat": 30, "ani": [30, 34, 35, 37, 38, 56, 57], "form": 30, "veri": 30, "commonli": 30, "howev": [30, 57], "thei": 30, "two": [30, 43], "problem": 30, "while": [30, 39], "inclus": 30, "doe": [30, 34, 35], "even": 30, "As": 30, "grown": 30, "larger": 30, "longer": 30, "instead": 30, "ad": [30, 36, 57], "supplementari": 30, "do": [30, 31], "pre": 30, "batch": 30, "normal": 30, "recommend": 30, "sesam": [30, 44], "must": [30, 37, 45, 47, 48], "comma": 30, "separ": [30, 57], "same": 30, "entri": [30, 34, 35], "correspond": 30, "measur": [30, 42, 43], "first": 30, "header": 30, "The": [30, 34, 35, 36, 38, 44, 45, 47, 48, 56, 57], "publish": 30, "match": [30, 36, 44], "exactli": 30, "1000": 30, "singl": [30, 37, 57], "beta": 30, "decim": 30, "precis": 30, "round": 30, "three": 30, "place": 30, "handl": [30, 37, 44], "integ": [30, 38], "permit": 30, "unknown": [30, 38], "822": 30, "state": 30, "encod": 30, "soft": 30, "content": [30, 34], "ncbi": 30, "tip": 30, "interest": [31, 57], "commerci": 31, "tabl": 31, "help": [31, 57], "contact": [31, 45, 47, 48], "author": 31, "free": 31, "For": 31, "cosmet": 31, "life": 31, "insur": 31, "applic": 31, "ucla": 31, "tdg": 31, "regard": 31, "licens": 31, "page": [34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50], "explain": [34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50], "user": [34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50], "guid": [34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50], "big": [34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 57], "pictur": [34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50], "max_size_gb": 34, "serial": 34, "basic": 34, "lru": 34, "least": 34, "recent": 34, "cleanup": 34, "maintain": [34, 57], "within": [34, 44], "maximum": 34, "limit": 34, "__init__": [34, 35, 36, 37, 38, 44], "instanc": [34, 36, 37, 38, 44], "filesystem": 34, "gigabyt": 34, "kei": [34, 35, 37, 50], "retriev": [34, 35, 36, 44], "item": [34, 35, 37, 57], "its": [34, 36, 44], "under": [34, 35], "manag": [34, 36, 44], "stai": 34, "clear": [34, 35], "remov": [34, 35], "noth": 35, "alwai": 35, "simul": 35, "library_fil": 36, "cach": [36, 37, 57], "collect": [36, 44, 45, 47, 48, 50, 57], "respons": 36, "new": [36, 38], "runtim": 36, "current": [36, 57], "mechan": [36, 37], "adher": [36, 37], "interfac": [36, 37, 56], "load_sourc": 36, "given": 36, "append": 36, "source_id": 36, "found": [36, 44], "otherwis": [36, 38, 45, 47, 48], "lookup_sourc": 36, "look": 36, "up": [36, 37, 57], "filter": [36, 38, 44], "criteria": [36, 44], "source_definit": 37, "encapsul": 37, "detail": [37, 38, 56], "rais": [37, 39, 44], "valueerror": [37, 39, 44], "requir": [37, 39, 57], "field": [37, 38], "parser": 37, "dure": 37, "pars": [37, 57], "typic": 37, "yaml": 37, "check": [37, 56], "essenti": 37, "attribut": 37, "dict": [37, 44, 50], "properti": 37, "nocach": 37, "input": [37, 38, 43, 44], "required_field": 37, "geograph": 37, "rtype": 37, "rna": 38, "protein": 38, "focu": 38, "facilit": 38, "access": [38, 57], "section": 38, "per": 38, "qualityreport": 38, "classmethod": 38, "matrix": 38, "either": 38, "directli": [38, 57], "union": 38, "static": 38, "convert_biolearn_to_standard_sex": 38, "convert": 38, "intern": 38, "represent": 38, "int": 38, "convert_standard_to_biolearn_sex": 38, "val": 38, "back": 38, "save_csv": 38, "folder_path": 38, "accord": 38, "arrai": 38, "2410": 38, "series_part": 38, "locat": 38, "part": 38, "concaten": 38, "cpg_sourc": [39, 41], "required_cpg": 39, "extern": [39, 41], "strategi": [39, 44, 51], "fill": [39, 40, 41], "certain": 39, "cpgs_to_imput": [40, 41], "itself": 40, "Not": 43, "come": 43, "pass": 43, "later": 43, "2010": 43, "2012": 43, "model_definit": 44, "produc": 44, "definit": 44, "model_build": 44, "deconvolutionmodel": 44, "bound": 44, "from_definit": 44, "grimagemodel": 44, "linearmethylationmodel": 44, "linearmodel": 44, "lineartranscriptomicmodel": 44, "sexestimationmodel": 44, "request": [44, 56], "possibli": 44, "keyerror": 44, "invalid": 44, "appli": 44, "dunedin": 44, "hybrid_imput": 44, "sesame_450k": 44, "450k": 44, "predictor_result": [45, 46, 47, 48], "predictor": [45, 46, 47, 48, 49], "dead": [45, 46, 47, 48], "boolean": [45, 46, 47], "indic": [45, 46, 47], "subject": [45, 46, 47, 48, 56], "death": [45, 46, 47, 48], "last": [45, 47, 48], "live": [45, 47, 48], "until": 46, "censor": 46, "log": 47, "rank": 47, "hazard": [48, 49], "ratio": [48, 49], "cox": 48, "aliv": 48, "confid": [48, 49], "interv": [48, 49], "hazard_ratio_data": 49, "forest": 49, "predictors_dict": 50, "output": 50, "ex": 50, "These": 51, "allow": 51, "hybrid": 51, "examim": 56, "implemen": 56, "prereleas": 56, "version": 56, "There": 56, "bug": 56, "com": 56, "homepag": 56, "pip": 56, "wa": 56, "out": [56, 57], "some": [56, 57], "team": 56, "answer": 56, "question": 56, "discuss": 56, "find": 56, "replic": 56, "setup": 56, "repo": 56, "omic": 57, "revolut": 57, "transform": 57, "biologi": 57, "wealth": 57, "potenti": 57, "care": 57, "could": 57, "offer": 57, "comprehens": 57, "insight": 57, "one": 57, "surrog": 57, "endpoint": 57, "revers": 57, "therapeut": 57, "obstruct": 57, "address": 57, "modern": 57, "recogn": 57, "ve": 57, "adopt": 57, "scienc": 57, "invit": 57, "mission": 57, "primarili": 57, "compris": 57, "thousand": 57, "ambit": 57, "need": 57, "achiev": 57, "larg": 57, "small": 57, "task": 57, "vari": 57, "skill": 57, "knowledg": 57, "right": 57, "improv": 57, "better": 57, "environ": 57, "speed": 57, "10x": 57, "integr": 57, "unifi": 57, "structur": 57, "know": 57, "partial": 57, "autom": 57, "review": 57, "genet": 57, "share": 57, "reach": 57, "proprietari": 57, "would": 57, "api": 57, "talk": 57, "public": 57, "hear": 57, "privat": 57, "think": 57, "wai": 57, "fund": 57, "financi": 57, "monei": 57, "mainten": 57, "prize": 57, "pot": 57, "further": 57, "expans": 57, "contriubt": 57, "discord": 57, "server": 57}, "objects": {"biolearn": [[32, 0, 0, "-", "cache"], [33, 0, 0, "-", "data_library"], [51, 0, 0, "-", "imputation"], [53, 0, 0, "-", "load"], [54, 0, 0, "-", "model_gallery"], [55, 0, 0, "-", "mortality"]], "biolearn.cache": [[34, 1, 1, "", "LocalFolderCache"], [35, 1, 1, "", "NoCache"]], "biolearn.cache.LocalFolderCache": [[34, 2, 1, "", "__init__"], [34, 2, 1, "", "clear"], [34, 2, 1, "", "get"], [34, 2, 1, "", "remove"], [34, 2, 1, "", "store"]], "biolearn.cache.NoCache": [[35, 2, 1, "", "__init__"], [35, 2, 1, "", "clear"], [35, 2, 1, "", "get"], [35, 2, 1, "", "remove"], [35, 2, 1, "", "store"]], "biolearn.data_library": [[36, 1, 1, "", "DataLibrary"], [37, 1, 1, "", "DataSource"], [38, 1, 1, "", "GeoData"]], "biolearn.data_library.DataLibrary": [[36, 2, 1, "", "__init__"], [36, 2, 1, "", "get"], [36, 2, 1, "", "load_sources"], [36, 2, 1, "", "lookup_sources"]], "biolearn.data_library.DataSource": [[37, 3, 1, "", "REQUIRED_FIELDS"], [37, 2, 1, "", "__init__"], [37, 2, 1, "", "load"]], "biolearn.data_library.GeoData": [[38, 2, 1, "", "__init__"], [38, 2, 1, "", "convert_biolearn_to_standard_sex"], [38, 2, 1, "", "convert_standard_to_biolearn_sex"], [38, 2, 1, "", "copy"], [38, 3, 1, "", "dnam"], [38, 2, 1, "", "from_methylation_matrix"], [38, 2, 1, "", "load_csv"], [38, 3, 1, "", "metadata"], [38, 2, 1, "", "quality_report"], [38, 2, 1, "", "save_csv"]], "biolearn.imputation": [[39, 4, 1, "", "hybrid_impute"], [40, 4, 1, "", "impute_from_average"], [41, 4, 1, "", "impute_from_standard"]], "biolearn.load": [[42, 4, 1, "", "load_fhs"], [43, 4, 1, "", "load_nhanes"]], "biolearn.model_gallery": [[44, 1, 1, "", "ModelGallery"]], "biolearn.model_gallery.ModelGallery": [[44, 2, 1, "", "__init__"], [44, 2, 1, "", "get"], [44, 3, 1, "", "model_builders"], [44, 2, 1, "", "search"]], "biolearn.mortality": [[45, 4, 1, "", "calculate_age_adjusted_c_index"], [46, 4, 1, "", "calculate_c_index"], [47, 4, 1, "", "calculate_log_rank_test"], [48, 4, 1, "", "calculate_mortality_hazard_ratios"], [49, 4, 1, "", "plot_hazard_ratios"], [50, 4, 1, "", "run_predictions"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:attribute", "4": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "attribute", "Python attribute"], "4": ["py", "function", "Python function"]}, "titleterms": {"team": 0, "core": 0, "other": [0, 5, 27], "contributor": 0, "partner": 0, "organ": 0, "omic": [1, 24], "biomark": [1, 8, 24, 57], "exampl": [1, 8, 11, 16, 17, 19, 24, 29, 30, 36, 37, 38, 41, 43, 44, 50], "perform": 2, "custom": 2, "imput": [2, 39, 40, 41, 51], "load": [2, 3, 4, 5, 6, 9, 12, 14, 15, 17, 21, 42, 43, 53], "methyl": [2, 3, 4, 12, 17, 21, 30], "data": [2, 3, 4, 5, 6, 9, 12, 14, 15, 28, 30, 51, 53], "from": [2, 3, 4, 6, 12, 22], "geo": [2, 3, 4, 6, 17, 20, 21, 22, 28], "dataset": [2, 4, 5, 17, 20, 21, 22], "us": [2, 3, 4, 5, 6, 14, 15, 17, 20, 21, 22, 36, 37, 38, 41, 43, 44, 50], "refer": [2, 12, 30, 52], "up": [2, 3, 4, 5, 6, 12, 14, 15, 21], "target": 2, "run": [2, 3, 4, 6, 12, 15], "now": [2, 3, 4, 5, 6], "clock": [2, 4, 6, 20, 27], "predict": [2, 3, 6, 14, 20], "befor": 2, "after": 2, "visual": [2, 19, 20, 21, 22, 24], "comparison": 2, "ag": [2, 4, 6, 9, 14, 20, 21], "you": [2, 5, 12], "can": [2, 5, 12], "also": [2, 12], "build": [2, 14], "an": [2, 14, 15], "decor": 2, "bundl": 2, "get": [2, 3], "same": 2, "result": [2, 4, 6, 15], "function": [2, 20, 21, 22, 51], "directli": 2, "your": 2, "down": 3, "syndrom": 3, "epigenet": [3, 4], "plot": [3, 4, 6, 9, 15, 21], "first": [3, 4, 6], "some": [3, 4, 6, 12, 15], "librari": [3, 4, 6], "model": [3, 12, 14, 15, 17, 20, 27, 31], "score": 3, "final": [3, 4], "gener": [3, 22], "boxplot": 3, "show": [3, 9], "power": 3, "three": 4, "differ": 4, "produc": 4, "extract": 4, "metadata": [4, 5, 12, 30], "seaborn": [4, 6], "local": 5, "file": [5, 15, 30], "i": [5, 12, 15, 30], "avail": 5, "like": 5, "ani": 5, "biolearn": [5, 29, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 53, 54, 55, 56, 57], "transcriptom": 6, "against": [6, 20, 21], "chronolog": [6, 9, 20], "comput": [7, 10, 13, 18, 23, 25, 58], "time": [7, 10, 13, 18, 23, 25, 58], "composit": [8, 24], "phenotyp": 9, "nhane": 9, "2010": 9, "calcul": 9, "biolog": 9, "base": 9, "phenotypicag": 9, "relat": 9, "between": 9, "surviv": 9, "curv": 9, "peopl": 9, "acceler": 9, "older": 9, "v": [9, 30], "deceler": 9, "younger": 9, "challeng": [11, 12, 15, 24], "submiss": [11, 14, 15, 24], "explor": 12, "competit": [12, 14, 15], "The": 12, "ha": 12, "proteom": 12, "learn": 12, "more": 12, "about": 12, "what": 12, "protein": 12, "identifi": [12, 21], "our": [12, 15], "overlap": 12, "while": 12, "doe": 12, "all": 12, "combin": 12, "easili": 12, "sever": 12, "them": 12, "we": 12, "compar": [12, 17], "output": [12, 15], "two": 12, "exist": 14, "lin": 14, "train": 15, "elasticnet": 15, "seper": 15, "test": 15, "set": [15, 21], "elast": 15, "net": 15, "see": 15, "how": 15, "good": 15, "through": 15, "save": 15, "deconvolut": [16, 17, 24], "blood": 17, "measur": 17, "epic": 17, "platform": 17, "estim": 17, "cell": 17, "proport": 17, "known": 17, "fac": 17, "fluoresc": 17, "activ": 17, "sort": 17, "addit": [19, 24, 30], "import": [20, 21, 22, 56], "requir": [20, 21, 22, 56], "class": [20, 21, 22], "correl": 20, "matrix": [20, 30], "across": 20, "deviat": [20, 22], "sampl": [20, 22], "heatmap": 20, "its": 20, "correspond": 20, "health": 20, "outcom": 20, "dna": [21, 30], "": 21, "object": [21, 22], "top": 21, "10": 21, "stabl": 21, "cpg": 21, "site": 21, "low": 21, "varianc": 21, "each": [21, 22], "level": 21, "sex": 21, "violin": 21, "linear": 21, "polynomi": 21, "regress": 21, "qualiti": 22, "control": 22, "creat": 22, "dictionari": 22, "displai": 22, "name": 22, "geodata": [22, 38], "report": 22, "distribut": 22, "popul": 22, "mean": 22, "sourc": 28, "featur": 29, "arrai": 30, "standard": 30, "2410": 30, "why": 30, "thi": 30, "need": 30, "preprocess": 30, "format": 30, "valu": 30, "field": 30, "ip": 31, "usag": 31, "cach": [32, 34, 35], "data_librari": [33, 36, 37, 38], "datalibrari": [33, 36], "localfoldercach": 34, "nocach": 35, "datasourc": 37, "hybrid_imput": 39, "impute_from_averag": 40, "impute_from_standard": 41, "load_fh": 42, "load_nhan": 43, "model_galleri": [44, 54], "modelgalleri": [44, 54], "mortal": [45, 46, 47, 48, 49, 50, 55], "calculate_age_adjusted_c_index": 45, "calculate_c_index": 46, "calculate_log_rank_test": 47, "calculate_mortality_hazard_ratio": 48, "plot_hazard_ratio": 49, "run_predict": 50, "api": 52, "util": 53, "predictor": 55, "evalu": 55, "quickstart": 56, "link": 56, "instal": 56, "discord": 56, "server": 56, "issu": 56, "contribut": 56, "futur": 57, "join": 57, "u": 57}, "envversion": {"sphinx.domains.c": 3, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 9, "sphinx.domains.index": 1, "sphinx.domains.javascript": 3, "sphinx.domains.math": 2, "sphinx.domains.python": 4, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinxcontrib.bibtex": 9, "sphinx": 60}, "alltitles": {"Team": [[0, "team"]], "Core Team": [[0, "core-team"]], "Other contributors": [[0, "other-contributors"]], "Partner Organizations": [[0, "partner-organizations"]], "Omics Biomarker Examples": [[1, "omics-biomarker-examples"], [24, "omics-biomarker-examples"]], "Performing custom imputations": [[2, "performing-custom-imputations"]], "Load methylation data from a GEO dataset to use as a reference for imputation": [[2, "load-methylation-data-from-a-geo-dataset-to-use-as-a-reference-for-imputation"]], "Load up a target dataset and run the imputation": [[2, "load-up-a-target-dataset-and-run-the-imputation"]], "Now run clock predictions on the dataset before and after": [[2, "now-run-clock-predictions-on-the-dataset-before-and-after"]], "Visualize the comparison of age predictions": [[2, "visualize-the-comparison-of-age-predictions"]], "You can also build an imputation decorator to bundle with the clock": [[2, "you-can-also-build-an-imputation-decorator-to-bundle-with-the-clock"]], "You get the same results as running the function directly on your dataset": [[2, "you-get-the-same-results-as-running-the-function-directly-on-your-dataset"]], "Down Syndrome Epigenetic Plotting": [[3, "down-syndrome-epigenetic-plotting"]], "First load up some methylation data from GEO using the data library": [[3, "first-load-up-some-methylation-data-from-geo-using-the-data-library"], [4, "first-load-up-some-methylation-data-from-geo-using-the-data-library"]], "Now run the down syndrome model on the data to get a score": [[3, "now-run-the-down-syndrome-model-on-the-data-to-get-a-score"]], "Finally generate a boxplot to show the predictive power": [[3, "finally-generate-a-boxplot-to-show-the-predictive-power"]], "\u201cEpigenetic Clocks\u201d in GEO Data": [[4, "epigenetic-clocks-in-geo-data"]], "Now run three different clocks on the dataset to produce epigenetic clock ages": [[4, "now-run-three-different-clocks-on-the-dataset-to-produce-epigenetic-clock-ages"]], "Finally extract the age data from the metadata from GEO and plot the results using seaborn": [[4, "finally-extract-the-age-data-from-the-metadata-from-geo-and-plot-the-results-using-seaborn"]], "Local Data Loading": [[5, "local-data-loading"]], "Load up a local data file": [[5, "load-up-a-local-data-file"]], "Metadata is loaded if available": [[5, "metadata-is-loaded-if-available"]], "You can now use it like any other Biolearn dataset": [[5, "you-can-now-use-it-like-any-other-biolearn-dataset"]], "\u201cTranscriptomic Clock\u201d using GEO Data": [[6, "transcriptomic-clock-using-geo-data"]], "First load up some transcriptomic data from GEO using the data library": [[6, "first-load-up-some-transcriptomic-data-from-geo-using-the-data-library"]], "Now run the transcriptomic clock to predict age": [[6, "now-run-the-transcriptomic-clock-to-predict-age"]], "Plot the results against the chronological age using seaborn": [[6, "plot-the-results-against-the-chronological-age-using-seaborn"]], "Computation times": [[7, "computation-times"], [10, "computation-times"], [13, "computation-times"], [18, "computation-times"], [23, "computation-times"], [25, "computation-times"], [58, "computation-times"]], "Composite Biomarkers Examples": [[8, "composite-biomarkers-examples"], [24, "composite-biomarkers-examples"]], "\u201cPhenotypic Ages\u201d in NHANES Data": [[9, "phenotypic-ages-in-nhanes-data"]], "Loading NHANES 2010 data": [[9, "loading-nhanes-2010-data"]], "Calculate \u201cbiological age\u201d based on PhenotypicAge": [[9, "calculate-biological-age-based-on-phenotypicage"]], "Show relation between biological age and chronological age": [[9, "show-relation-between-biological-age-and-chronological-age"]], "Plot survival curve for people with accelerated aging (older biological age) vs decelerated aging (younger biological age)": [[9, "plot-survival-curve-for-people-with-accelerated-aging-older-biological-age-vs-decelerated-aging-younger-biological-age"]], "Challenge Submission Examples": [[11, "challenge-submission-examples"], [24, "challenge-submission-examples"]], "Exploring the Challenge Data": [[12, "exploring-the-challenge-data"]], "Loading up the data for the competition": [[12, "loading-up-the-data-for-the-competition"], [14, "loading-up-the-data-for-the-competition"], [15, "loading-up-the-data-for-the-competition"]], "The challenge data has methylation data": [[12, "the-challenge-data-has-methylation-data"]], "The challenge data also has proteomic data": [[12, "the-challenge-data-also-has-proteomic-data"]], "You can learn more about what the protein identifies in our reference": [[12, "you-can-learn-more-about-what-the-protein-identifies-in-our-reference"]], "Some of the data overlaps while some does not but all the metadata is combined": [[12, "some-of-the-data-overlaps-while-some-does-not-but-all-the-metadata-is-combined"]], "You can easily run several models on them": [[12, "you-can-easily-run-several-models-on-them"]], "We can then compare the output from the two models": [[12, "we-can-then-compare-the-output-from-the-two-models"]], "Deconvolution Examples": [[16, "deconvolution-examples"], [24, "deconvolution-examples"]], "\u201cDeconvolution Example\u201d": [[17, "deconvolution-example"]], "Load GEO blood methylation dataset measured on EPIC platform": [[17, "load-geo-blood-methylation-dataset-measured-on-epic-platform"]], "Estimate cell proportions using EPIC deconvolution model": [[17, "estimate-cell-proportions-using-epic-deconvolution-model"]], "Compare to known cell proportions measured with FACS (fluorescence-activated cell sorting)": [[17, "compare-to-known-cell-proportions-measured-with-facs-fluorescence-activated-cell-sorting"]], "Additional Visualization Examples": [[19, "additional-visualization-examples"], [24, "additional-visualization-examples"]], "Clock/model visualizations using GEO datasets": [[20, "clock-model-visualizations-using-geo-datasets"]], "Import required classes and functions": [[20, "import-required-classes-and-functions"], [22, "import-required-classes-and-functions"]], "Visualize a correlation matrix across aging clocks/models": [[20, "visualize-a-correlation-matrix-across-aging-clocks-models"]], "Visualize clock/model chronological age deviations across samples in a heatmap": [[20, "visualize-clock-model-chronological-age-deviations-across-samples-in-a-heatmap"]], "Visualize aging clock/model predictions against chronological age": [[20, "visualize-aging-clock-model-predictions-against-chronological-age"]], "Visualize model predictions against its corresponding health outcome": [[20, "visualize-model-predictions-against-its-corresponding-health-outcome"]], "DNA methylation visualizations using GEO datasets": [[21, "dna-methylation-visualizations-using-geo-datasets"]], "Import required classes, and functions": [[21, "import-required-classes-and-functions"]], "Load GEO methylation dataset(s) and set up objects for visualization function use": [[21, "load-geo-methylation-dataset-s-and-set-up-objects-for-visualization-function-use"]], "Identify the top 10 stable CpG sites with low variance for each dataset": [[21, "identify-the-top-10-stable-cpg-sites-with-low-variance-for-each-dataset"]], "Visualize DNA methylation levels by age and sex in a violin plot": [[21, "visualize-dna-methylation-levels-by-age-and-sex-in-a-violin-plot"]], "Visualize DNA methylation against age with linear and polynomial regression": [[21, "visualize-dna-methylation-against-age-with-linear-and-polynomial-regression"]], "Quality control visualization using GEO datasets": [[22, "quality-control-visualization-using-geo-datasets"]], "Create a dictionary of dataset display names to GeoData objects": [[22, "create-a-dictionary-of-dataset-display-names-to-geodata-objects"]], "Generate a quality control report for each dataset": [[22, "generate-a-quality-control-report-for-each-dataset"]], "Visualize the distribution of sample deviations from the population mean": [[22, "visualize-the-distribution-of-sample-deviations-from-the-population-mean"]], "Examples": [[24, "examples"]], "Biolearn": [[29, "biolearn"], [56, "biolearn"]], "Featured examples": [[29, "featured-examples"]], "DNA Methylation Array Data Standard V-2410": [[30, "dna-methylation-array-data-standard-v-2410"]], "Why is this standard needed?": [[30, "why-is-this-standard-needed"]], "Preprocessing": [[30, "preprocessing"]], "Methylation Matrix File Standard": [[30, "methylation-matrix-file-standard"]], "File Format": [[30, "file-format"], [30, "id1"]], "Data Values": [[30, "data-values"]], "Metadata File Standard": [[30, "metadata-file-standard"]], "Standard Metadata Fields": [[30, "standard-metadata-fields"]], "Additional Metadata Fields": [[30, "additional-metadata-fields"]], "Example Files": [[30, "example-files"]], "References": [[30, "references"]], "biolearn.cache: Cache": [[32, "module-biolearn.cache"]], "biolearn.cache.LocalFolderCache": [[34, "biolearn-cache-localfoldercache"]], "biolearn.cache.NoCache": [[35, "biolearn-cache-nocache"]], "biolearn.imputation.hybrid_impute": [[39, "biolearn-imputation-hybrid-impute"]], "biolearn.imputation.impute_from_average": [[40, "biolearn-imputation-impute-from-average"]], "biolearn.imputation.impute_from_standard": [[41, "biolearn-imputation-impute-from-standard"]], "Examples using biolearn.imputation.impute_from_standard": [[41, "examples-using-biolearn-imputation-impute-from-standard"]], "biolearn.load.load_fhs": [[42, "biolearn-load-load-fhs"]], "biolearn.load.load_nhanes": [[43, "biolearn-load-load-nhanes"]], "Examples using biolearn.load.load_nhanes": [[43, "examples-using-biolearn-load-load-nhanes"]], "biolearn.mortality.calculate_age_adjusted_c_index": [[45, "biolearn-mortality-calculate-age-adjusted-c-index"]], "biolearn.mortality.calculate_c_index": [[46, "biolearn-mortality-calculate-c-index"]], "biolearn.mortality.calculate_log_rank_test": [[47, "biolearn-mortality-calculate-log-rank-test"]], "biolearn.mortality.calculate_mortality_hazard_ratios": [[48, "biolearn-mortality-calculate-mortality-hazard-ratios"]], "biolearn.mortality.plot_hazard_ratios": [[49, "biolearn-mortality-plot-hazard-ratios"]], "biolearn.mortality.run_predictions": [[50, "biolearn-mortality-run-predictions"]], "Examples using biolearn.mortality.run_predictions": [[50, "examples-using-biolearn-mortality-run-predictions"]], "biolearn.imputation: Data Imputation Functions": [[51, "biolearn-imputation-data-imputation-functions"]], "API References": [[52, "api-references"]], "biolearn.load: Data Loading Utilities": [[53, "module-biolearn.load"]], "biolearn.model_gallery: ModelGallery": [[54, "module-biolearn.model_gallery"]], "biolearn.mortality: Mortality Predictor Evaluation": [[55, "biolearn-mortality-mortality-predictor-evaluation"]], "Quickstart": [[56, "quickstart"]], "Important links": [[56, "important-links"]], "Requirements": [[56, "requirements"]], "Install": [[56, "install"]], "Discord server": [[56, "discord-server"]], "Issues": [[56, "issues"]], "Contributing": [[56, "contributing"]], "Biolearn and the Future of Biomarkers": [[57, "biolearn-and-the-future-of-biomarkers"]], "Join Us": [[57, "join-us"]], "Building a competition submission using an existing model": [[14, "building-a-competition-submission-using-an-existing-model"]], "Use the Lin model to predict the age": [[14, "use-the-lin-model-to-predict-the-age"]], "Training an ElasticNet model": [[15, "training-an-elasticnet-model"]], "Load up some training data": [[15, "load-up-some-training-data"]], "Seperate data into training and test sets": [[15, "seperate-data-into-training-and-test-sets"]], "Train a model using elastic net": [[15, "train-a-model-using-elastic-net"]], "Plot the results to see how good our model is": [[15, "plot-the-results-to-see-how-good-our-model-is"]], "Run the challenge data through the model": [[15, "run-the-challenge-data-through-the-model"]], "Save the results as an output file for submission": [[15, "save-the-results-as-an-output-file-for-submission"]], "Clocks and Other Models": [[27, "clocks-and-other-models"]], "GEO Data Sources": [[28, "geo-data-sources"]], "Model IP and Usage": [[31, "model-ip-and-usage"]], "biolearn.data_library: DataLibrary": [[33, "module-biolearn.data_library"]], "biolearn.data_library.DataLibrary": [[36, "biolearn-data-library-datalibrary"]], "Examples using biolearn.data_library.DataLibrary": [[36, "examples-using-biolearn-data-library-datalibrary"]], "biolearn.data_library.DataSource": [[37, "biolearn-data-library-datasource"]], "Examples using biolearn.data_library.DataSource": [[37, "examples-using-biolearn-data-library-datasource"]], "biolearn.data_library.GeoData": [[38, "biolearn-data-library-geodata"]], "Examples using biolearn.data_library.GeoData": [[38, "examples-using-biolearn-data-library-geodata"]], "biolearn.model_gallery.ModelGallery": [[44, "biolearn-model-gallery-modelgallery"]], "Examples using biolearn.model_gallery.ModelGallery": [[44, "examples-using-biolearn-model-gallery-modelgallery"]]}, "indexentries": {"biolearn.cache": [[32, "module-biolearn.cache"]], "module": [[32, "module-biolearn.cache"], [33, "module-biolearn.data_library"], [51, "module-biolearn.imputation"], [53, "module-biolearn.load"], [54, "module-biolearn.model_gallery"], [55, "module-biolearn.mortality"]], "biolearn.data_library": [[33, "module-biolearn.data_library"]], "localfoldercache (class in biolearn.cache)": [[34, "biolearn.cache.LocalFolderCache"]], "__init__() (biolearn.cache.localfoldercache method)": [[34, "biolearn.cache.LocalFolderCache.__init__"]], "clear() (biolearn.cache.localfoldercache method)": [[34, "biolearn.cache.LocalFolderCache.clear"]], "get() (biolearn.cache.localfoldercache method)": [[34, "biolearn.cache.LocalFolderCache.get"]], "remove() (biolearn.cache.localfoldercache method)": [[34, "biolearn.cache.LocalFolderCache.remove"]], "store() (biolearn.cache.localfoldercache method)": [[34, "biolearn.cache.LocalFolderCache.store"]], "nocache (class in biolearn.cache)": [[35, "biolearn.cache.NoCache"]], "__init__() (biolearn.cache.nocache method)": [[35, "biolearn.cache.NoCache.__init__"]], "clear() (biolearn.cache.nocache method)": [[35, "biolearn.cache.NoCache.clear"]], "get() (biolearn.cache.nocache method)": [[35, "biolearn.cache.NoCache.get"]], "remove() (biolearn.cache.nocache method)": [[35, "biolearn.cache.NoCache.remove"]], "store() (biolearn.cache.nocache method)": [[35, "biolearn.cache.NoCache.store"]], "datalibrary (class in biolearn.data_library)": [[36, "biolearn.data_library.DataLibrary"]], "__init__() (biolearn.data_library.datalibrary method)": [[36, "biolearn.data_library.DataLibrary.__init__"]], "get() (biolearn.data_library.datalibrary method)": [[36, "biolearn.data_library.DataLibrary.get"]], "load_sources() (biolearn.data_library.datalibrary method)": [[36, "biolearn.data_library.DataLibrary.load_sources"]], "lookup_sources() (biolearn.data_library.datalibrary method)": [[36, "biolearn.data_library.DataLibrary.lookup_sources"]], "datasource (class in biolearn.data_library)": [[37, "biolearn.data_library.DataSource"]], "required_fields (biolearn.data_library.datasource attribute)": [[37, "biolearn.data_library.DataSource.REQUIRED_FIELDS"]], "__init__() (biolearn.data_library.datasource method)": [[37, "biolearn.data_library.DataSource.__init__"]], "load() (biolearn.data_library.datasource method)": [[37, "biolearn.data_library.DataSource.load"]], "geodata (class in biolearn.data_library)": [[38, "biolearn.data_library.GeoData"]], "__init__() (biolearn.data_library.geodata method)": [[38, "biolearn.data_library.GeoData.__init__"]], "convert_biolearn_to_standard_sex() (biolearn.data_library.geodata static method)": [[38, "biolearn.data_library.GeoData.convert_biolearn_to_standard_sex"]], "convert_standard_to_biolearn_sex() (biolearn.data_library.geodata static method)": [[38, "biolearn.data_library.GeoData.convert_standard_to_biolearn_sex"]], "copy() (biolearn.data_library.geodata method)": [[38, "biolearn.data_library.GeoData.copy"]], "dnam (biolearn.data_library.geodata attribute)": [[38, "biolearn.data_library.GeoData.dnam"]], "from_methylation_matrix() (biolearn.data_library.geodata class method)": [[38, "biolearn.data_library.GeoData.from_methylation_matrix"]], "load_csv() (biolearn.data_library.geodata class method)": [[38, "biolearn.data_library.GeoData.load_csv"]], "metadata (biolearn.data_library.geodata attribute)": [[38, "biolearn.data_library.GeoData.metadata"]], "quality_report() (biolearn.data_library.geodata method)": [[38, "biolearn.data_library.GeoData.quality_report"]], "save_csv() (biolearn.data_library.geodata method)": [[38, "biolearn.data_library.GeoData.save_csv"]], "hybrid_impute() (in module biolearn.imputation)": [[39, "biolearn.imputation.hybrid_impute"]], "impute_from_average() (in module biolearn.imputation)": [[40, "biolearn.imputation.impute_from_average"]], "impute_from_standard() (in module biolearn.imputation)": [[41, "biolearn.imputation.impute_from_standard"]], "load_fhs() (in module biolearn.load)": [[42, "biolearn.load.load_fhs"]], "load_nhanes() (in module biolearn.load)": [[43, "biolearn.load.load_nhanes"]], "modelgallery (class in biolearn.model_gallery)": [[44, "biolearn.model_gallery.ModelGallery"]], "__init__() (biolearn.model_gallery.modelgallery method)": [[44, "biolearn.model_gallery.ModelGallery.__init__"]], "get() (biolearn.model_gallery.modelgallery method)": [[44, "biolearn.model_gallery.ModelGallery.get"]], "model_builders (biolearn.model_gallery.modelgallery attribute)": [[44, "biolearn.model_gallery.ModelGallery.model_builders"]], "search() (biolearn.model_gallery.modelgallery method)": [[44, "biolearn.model_gallery.ModelGallery.search"]], "calculate_age_adjusted_c_index() (in module biolearn.mortality)": [[45, "biolearn.mortality.calculate_age_adjusted_c_index"]], "calculate_c_index() (in module biolearn.mortality)": [[46, "biolearn.mortality.calculate_c_index"]], "calculate_log_rank_test() (in module biolearn.mortality)": [[47, "biolearn.mortality.calculate_log_rank_test"]], "calculate_mortality_hazard_ratios() (in module biolearn.mortality)": [[48, "biolearn.mortality.calculate_mortality_hazard_ratios"]], "plot_hazard_ratios() (in module biolearn.mortality)": [[49, "biolearn.mortality.plot_hazard_ratios"]], "run_predictions() (in module biolearn.mortality)": [[50, "biolearn.mortality.run_predictions"]], "biolearn.imputation": [[51, "module-biolearn.imputation"]], "biolearn.load": [[53, "module-biolearn.load"]], "biolearn.model_gallery": [[54, "module-biolearn.model_gallery"]], "biolearn.mortality": [[55, "module-biolearn.mortality"]]}})